
<!DOCTYPE html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      <meta http-equiv="x-ua-compatible" content="ie=edge">
      
        <meta name="description" content="rlx input variables">
      
      
        <link rel="canonical" href="http://www.abinit.org/variables/rlx/">
      
      
        <meta name="author" content="The Abinit group">
      
      
        <link rel="shortcut icon" href="../../images/favicon.ico">
      
      <meta name="generator" content="mkdocs-0.16.3, mkdocs-material-1.10.1">
    
    
      
        <title>Relaxation - abinit</title>
      
    
    
    
      <script src="../../assets/javascripts/modernizr-e826f8942a.js"></script>
    

    <!-- Jquery from Google CDN -->
    <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.2.1/jquery.min.js"></script>

    <!-- Load Jquery UI -->
    <script src="https://code.jquery.com/ui/1.12.1/jquery-ui.min.js"></script>
    <!-- Handle jQuery plugin naming conflict between jQuery UI and Bootstrap 
    https://stackoverflow.com/questions/13731400/jqueryui-tooltips-are-competing-with-twitter-bootstrap
    -->
    <script>
      $.widget.bridge('uibutton', $.ui.button);
      $.widget.bridge('uitooltip', $.ui.tooltip);
    </script>

    <!--
    <link rel="stylesheet" href="https://code.jquery.com/ui/1.12.1/themes/smoothness/jquery-ui.css">
    <link rel="stylesheet" href="https://code.jquery.com/ui/1.12.1/themes/pepper-grinder/jquery-ui.css">
    -->
    <link rel="stylesheet" href="https://code.jquery.com/ui/1.12.1/themes/base/jquery-ui.css">


    <!-- JavaScript -->
    <script src="//cdn.jsdelivr.net/npm/alertifyjs@1.11.0/build/alertify.min.js"></script>

    <!-- CSS -->
    <link rel="stylesheet" href="//cdn.jsdelivr.net/npm/alertifyjs@1.11.0/build/css/alertify.min.css"/>
    <!-- Default theme
    <link rel="stylesheet" href="//cdn.jsdelivr.net/npm/alertifyjs@1.11.0/build/css/themes/default.min.css"/>
    -->
    <!-- Semantic UI theme
    <link rel="stylesheet" href="//cdn.jsdelivr.net/npm/alertifyjs@1.11.0/build/css/themes/semantic.min.css"/>
    -->
    <!-- Bootstrap theme -->
    <link rel="stylesheet" href="//cdn.jsdelivr.net/npm/alertifyjs@1.11.0/build/css/themes/bootstrap.min.css"/>
    

    <!-- Bootstrap http://getbootstrap.com/getting-started/ -->
    <!-- Latest compiled and minified CSS -->
    <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" 
    integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">

    <!-- Optional theme -->
    <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap-theme.min.css" 
    integrity="sha384-rHyoN1iRsVXV4nD0JutlnGaslCJuC7uwjduW9SVrLvRYooPp2bWYgmgJQIXwl/Sp" crossorigin="anonymous">

    <!-- Latest compiled and minified JavaScript -->
    <script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" 
    integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous">
    </script>

    

    <!--Pace support. http://github.hubspot.com/pace/ -->
    <script>
        paceOptions = {
            ajax: true,      // disabled
            document: true,  // activated
            eventLag: false  // disabled
            //target: 'article'
            //elements: {selectors: ['.my-page']}
        };
    </script>

    <script src="https://cdnjs.cloudflare.com/ajax/libs/pace/1.0.2/pace.min.js"></script>
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/pace/1.0.2/themes/white/pace-theme-big-counter.min.css" />
    <!--
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/pace/1.0.2/themes/white/pace-theme-bounce.min.css" />
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/pace/1.0.2/themes/blue/pace-theme-corner-indicator.min.css" />
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/pace/1.0.2/themes/blue/pace-theme-flat-top.min.css" />
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/pace/1.0.2/themes/blue/pace-theme-loading-bar.min.css" />
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/pace/1.0.2/themes/blue/pace-theme-center-atom.css" />
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/pace/1.0.2/themes/blue/pace-theme-flash.min.css" />
    -->

    <!-- MathJax support
    <script type="text/javascript" async 
        src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-MML-AM_CHTML">
    </script> 
    -->
    <!--
    This is compatible with plotly.
    -->
    <script type="text/javascript" async 
        src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_SVG">
    </script> 

    <!--
    Configure MathJax to produce automatic equation numbers http://docs.mathjax.org/en/latest/tex.html
    -->
    <script type="text/x-mathjax-config">
      MathJax.Hub.Config({
          TeX: {
              equationNumbers: { autoNumber: "all" },
              extensions: ["AMSmath.js"],
              Macros: {
                  rr: "{\\bf r}",
                  GG: "{\\bf G}",
                  kk: "{\\bf k}",
                  qq: "{\\bf q}",
                  kq: "{\\kk + \\qq}"
              }
          }
      });
    </script>

    <!-- See http://docs.mathjax.org/en/latest/start.html 
    <script type="text/x-mathjax-config">
    MathJax.Hub.Config({
        extensions: ["tex2jax.js", "MathMenu.js", "MathZoom.js"],
        jax: ["input/TeX", "output/HTML-CSS", "output/NativeMML],
        tex2jax: {
            inlineMath: [ ['$','$'], ["\\(","\\)"] ],
            displayMath: [ ['$$','$$'], ["\\[","\\]"] ],
            processEscapes: true
        },
        "HTML-CSS": { availableFonts: ["TeX"] }
      });
    </script>
    -->

    
    
    

    <!-- Load fonts used for input variables -->
    <!--
    <link href="https://fonts.googleapis.com/css?family=Ubuntu+Mono" rel="stylesheet">
    -->
    <link href="https://fonts.googleapis.com/css?family=Inconsolata" rel="stylesheet">


    
      <link rel="stylesheet" href="../../assets/stylesheets/application-a20f419c8e.css">
      
        <link rel="stylesheet" href="../../assets/stylesheets/application-23f75ab9c7.palette.css">
      
    
    
      
        
        
        
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,400,400i,700|Roboto+Mono">
        <style>body,input{font-family:"Roboto","Helvetica Neue",Helvetica,Arial,sans-serif}code,kbd,pre{font-family:"Roboto Mono","Courier New",Courier,monospace}</style>
      
      <link rel="stylesheet" href="https://fonts.googleapis.com/icon?family=Material+Icons">
    
    
      <link rel="stylesheet" href="../../css/extra.css">
    
      <link rel="stylesheet" href="../../css/gallery.css">
    
      <link rel="stylesheet" href="../../css/my_codehilite.css">
    
    
  </head>
  
  
  
  
    <body data-md-color-primary="deep-orange" data-md-color-accent="deep-orange">
  
    <svg class="md-svg">
      <defs>
        
        
          <svg xmlns="http://www.w3.org/2000/svg" width="416" height="448" viewBox="0 0 416 448" id="github"><path fill="currentColor" d="M160 304q0 10-3.125 20.5t-10.75 19T128 352t-18.125-8.5-10.75-19T96 304t3.125-20.5 10.75-19T128 256t18.125 8.5 10.75 19T160 304zm160 0q0 10-3.125 20.5t-10.75 19T288 352t-18.125-8.5-10.75-19T256 304t3.125-20.5 10.75-19T288 256t18.125 8.5 10.75 19T320 304zm40 0q0-30-17.25-51T296 232q-10.25 0-48.75 5.25Q229.5 240 208 240t-39.25-2.75Q130.75 232 120 232q-29.5 0-46.75 21T56 304q0 22 8 38.375t20.25 25.75 30.5 15 35 7.375 37.25 1.75h42q20.5 0 37.25-1.75t35-7.375 30.5-15 20.25-25.75T360 304zm56-44q0 51.75-15.25 82.75-9.5 19.25-26.375 33.25t-35.25 21.5-42.5 11.875-42.875 5.5T212 416q-19.5 0-35.5-.75t-36.875-3.125-38.125-7.5-34.25-12.875T37 371.5t-21.5-28.75Q0 312 0 260q0-59.25 34-99-6.75-20.5-6.75-42.5 0-29 12.75-54.5 27 0 47.5 9.875t47.25 30.875Q171.5 96 212 96q37 0 70 8 26.25-20.5 46.75-30.25T376 64q12.75 25.5 12.75 54.5 0 21.75-6.75 42 34 40 34 99.5z"/></svg>
        
      </defs>
    </svg>
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="drawer">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="search">
    <label class="md-overlay" data-md-component="overlay" for="drawer"></label>
    
      <header class="md-header" data-md-component="header">
  <nav class="md-header-nav md-grid">
    <div class="md-flex">

      <div class="md-flex__cell md-flex__cell--shrink">
        
          <a href="http://www.abinit.org" title="abinit" class="md-logo md-header-nav__button">
            <img src="../../images/logo-abinit-2015.svg" width="24" height="24">
          </a>
        
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        <label class="md-icon md-icon--menu md-header-nav__button" for="drawer"></label>
      </div>
      <div class="md-flex__cell md-flex__cell--stretch">
        <span class="md-flex__ellipsis md-header-nav__title">
          
            
              
                <span class="md-header-nav__parent">
                  Variables
                </span>
              
                <span class="md-header-nav__parent">
                  Abinit
                </span>
              
            
            Relaxation
          
        </span>
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        
          <label class="md-icon md-icon--search md-header-nav__button" for="search"></label>
          
<div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="search"></label>
  <div class="md-search__inner">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" required placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="query">
      <label class="md-icon md-search__icon" for="search"></label>
      <button type="reset" class="md-icon md-search__icon" data-md-component="reset">&#xE5CD;</button>
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="result" data-md-lang-search="" data-md-lang-tokenizer="[\s\-]+">
          <div class="md-search-result__meta" data-md-lang-result-none="No matching documents" data-md-lang-result-one="1 matching document" data-md-lang-result-other="# matching documents">
            Type to start searching
          </div>
          <ol class="md-search-result__list"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
        
      </div>
      
        <div class="md-flex__cell md-flex__cell--shrink">
          <div class="md-header-nav__source">
            


  


  <a href="https://github.com/abinit/abinit/" title="Go to repository" class="md-source" data-md-source="github">
    
      <div class="md-source__icon">
        <svg viewBox="0 0 24 24" width="24" height="24">
          <use xlink:href="#github" width="24" height="24"></use>
        </svg>
      </div>
    
    <div class="md-source__repository">
      github-abinit
    </div>
  </a>

          </div>
        </div>
      

      <div class="md-flex__cell md-flex__cell--shrink">
        <div class="dropdown">
          <button class="btn btn-basic btn-sm dropdown-toggle" type="button" data-toggle="dropdown">
            8.5.4
          <span class="caret"></span></button>
          <ul class="dropdown-menu">
            
            <li><a href="#">8.2.0</a></li>
            
            <li><a href="#">8.1.0</a></li>
            
          </ul>
        </div>
      </div>

    </div>
  </nav>
</header>
    
    <div class="md-container">
      
      
        

  

<nav class="md-tabs md-tabs--active" data-md-component="tabs">
  <div class="md-tabs__inner md-grid">
    <ul class="md-tabs__list">
      
        
  
  
    <li class="md-tabs__item">
      
        <a href="../.." title="User Guide" class="md-tabs__link">
          User Guide
        </a>
      
    </li>
  

      
        
  
  
    <li class="md-tabs__item">
      
        <a href="../../user-guide/features/" title="Topics" class="md-tabs__link">
          Topics
        </a>
      
    </li>
  

      
        
  
  
    <li class="md-tabs__item">
      
        <a href="../" title="Variables" class="md-tabs__link md-tabs__link--active">
          Variables
        </a>
      
    </li>
  

      
        
  
  
    <li class="md-tabs__item">
      
        <a href="../../tutorials/welcome/" title="Lessons" class="md-tabs__link">
          Lessons
        </a>
      
    </li>
  

      
        
  
  
    <li class="md-tabs__item">
      
        <a href="../../theory/wavefunctions/" title="Theory" class="md-tabs__link">
          Theory
        </a>
      
    </li>
  

      
        
  
  
    <li class="md-tabs__item">
      
        <a href="../../developers/abimkdocs/" title="Developers" class="md-tabs__link">
          Developers
        </a>
      
    </li>
  

      
        
  
  
    <li class="md-tabs__item">
      
        <a href="../../about/release-notes/" title="About" class="md-tabs__link">
          About
        </a>
      
    </li>
  

      
    </ul>
  </div>
</nav>
      
      <main class="md-main">
        <div class="md-main__inner md-grid" data-md-component="container">
          
            
              <div class="md-sidebar md-sidebar--primary" data-md-component="navigation">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    <nav class="md-nav md-nav--primary" data-md-level="0">
  <label class="md-nav__title md-nav__title--site" for="drawer">
    <div class="md-nav__button md-logo">
      
        
          <img src="../../images/logo-abinit-2015.svg">
        
      
    </div>
    abinit
  </label>
  
    <div class="md-nav__source">
      


  


  <a href="https://github.com/abinit/abinit/" title="Go to repository" class="md-source" data-md-source="github">
    
      <div class="md-source__icon">
        <svg viewBox="0 0 24 24" width="24" height="24">
          <use xlink:href="#github" width="24" height="24"></use>
        </svg>
      </div>
    
    <div class="md-source__repository">
      github-abinit
    </div>
  </a>

    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-1" type="checkbox" id="nav-1">
    
    <label class="md-nav__link" for="nav-1">
      User Guide
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-1">
        User Guide
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../.." title="Introduction" class="md-nav__link">
      Introduction
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../installation/" title="Installation" class="md-nav__link">
      Installation
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../user-guide/help_what_ABINIT_does/" title="Overview" class="md-nav__link">
      Overview
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../user-guide/abinit/" title="Abinit" class="md-nav__link">
      Abinit
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../user-guide/aim/" title="Aim" class="md-nav__link">
      Aim
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../user-guide/anaddb/" title="Anaddb" class="md-nav__link">
      Anaddb
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../user-guide/cut3d/" title="Cut3d" class="md-nav__link">
      Cut3d
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../user-guide/features/" title="Features" class="md-nav__link">
      Features
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../user-guide/fold2bloch/" title="Fold2Bloch" class="md-nav__link">
      Fold2Bloch
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../user-guide/mrgddb/" title="Mrgddb" class="md-nav__link">
      Mrgddb
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../user-guide/mrgscr/" title="Mrgscr" class="md-nav__link">
      Mrgscr
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../user-guide/new_user/" title="NewUser" class="md-nav__link">
      NewUser
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../user-guide/optic/" title="Optic" class="md-nav__link">
      Optic
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../user-guide/respfn/" title="Respfn" class="md-nav__link">
      Respfn
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-2" type="checkbox" id="nav-2">
    
    <label class="md-nav__link" for="nav-2">
      Topics
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-2">
        Topics
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../user-guide/features/" title="Features" class="md-nav__link">
      Features
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../topics/Abipy/" title="Abipy" class="md-nav__link">
      Abipy
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../topics/APPA/" title="APPA" class="md-nav__link">
      APPA
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../topics/Artificial/" title="Artificial" class="md-nav__link">
      Artificial
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../topics/AtomManipulator/" title="AtomManipulator" class="md-nav__link">
      AtomManipulator
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../topics/AtomTypes/" title="AtomTypes" class="md-nav__link">
      AtomTypes
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../topics/Bader/" title="Bader" class="md-nav__link">
      Bader
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../topics/Band2eps/" title="Band2eps" class="md-nav__link">
      Band2eps
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../topics/Berry/" title="Berry" class="md-nav__link">
      Berry
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../topics/BandOcc/" title="BandOcc" class="md-nav__link">
      BandOcc
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../topics/BSE/" title="BSE" class="md-nav__link">
      BSE
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../topics/ConstrainedPol/" title="ConstrainedPol" class="md-nav__link">
      ConstrainedPol
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../topics/Control/" title="Control" class="md-nav__link">
      Control
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../topics/Coulomb/" title="Coulomb" class="md-nav__link">
      Coulomb
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../topics/CRPA/" title="CRPA" class="md-nav__link">
      CRPA
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../topics/crystal/" title="Crystal" class="md-nav__link">
      Crystal
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../topics/DFT+U/" title="DFT+U" class="md-nav__link">
      DFT+U
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../topics/DeltaSCF/" title="DeltaSCF" class="md-nav__link">
      DeltaSCF
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../topics/DensityPotential/" title="DensityPotential" class="md-nav__link">
      DensityPotential
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../topics/Dev/" title="Dev" class="md-nav__link">
      Dev
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../topics/DFPT/" title="DFPT" class="md-nav__link">
      DFPT
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../topics/DMFT/" title="DMFT" class="md-nav__link">
      DMFT
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../topics/EffMass/" title="EffMass" class="md-nav__link">
      EffMass
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../topics/EFG/" title="EFG" class="md-nav__link">
      EFG
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../topics/Elastic/" title="Elastic" class="md-nav__link">
      Elastic
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../topics/ElPhonInt/" title="ElPhonInt" class="md-nav__link">
      ElPhonInt
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../topics/ElPhonTransport/" title="ElPhonTransport" class="md-nav__link">
      ElPhonTransport
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../topics/ElecDOS/" title="ElecDOS" class="md-nav__link">
      ElecDOS
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../topics/ElecBandStructure/" title="ElecBandStructure" class="md-nav__link">
      ElecBandStructure
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../topics/ForcesStresses/" title="ForcesStresses" class="md-nav__link">
      ForcesStresses
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../topics/FrequencyMeshMBPT/" title="FrequencyMeshMBPT" class="md-nav__link">
      FrequencyMeshMBPT
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../topics/GeoConstraints/" title="GeoConstraints" class="md-nav__link">
      GeoConstraints
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../topics/GeoOpt/" title="GeoOpt" class="md-nav__link">
      GeoOpt
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../topics/Git/" title="Git" class="md-nav__link">
      Git
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../topics/GSintroduction/" title="GSintroduction" class="md-nav__link">
      GSintroduction
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../topics/GW/" title="GW" class="md-nav__link">
      GW
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../topics/GWls/" title="GWls" class="md-nav__link">
      GWls
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../topics/Hybrids/" title="Hybrids" class="md-nav__link">
      Hybrids
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../topics/k-points/" title="K points" class="md-nav__link">
      K points
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../topics/LOTF/" title="LOTF" class="md-nav__link">
      LOTF
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../topics/MagField/" title="MagField" class="md-nav__link">
      MagField
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../topics/MagMom/" title="MagMom" class="md-nav__link">
      MagMom
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../topics/MolecularDynamics/" title="MolecularDynamics" class="md-nav__link">
      MolecularDynamics
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../topics/multidtset/" title="Multidtset" class="md-nav__link">
      Multidtset
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../topics/nonlinear/" title="Nonlinear" class="md-nav__link">
      Nonlinear
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../topics/Optic/" title="Optic" class="md-nav__link">
      Optic
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../topics/Output/" title="Output" class="md-nav__link">
      Output
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../topics/parallelism/" title="Parallelism" class="md-nav__link">
      Parallelism
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../topics/PAW/" title="PAW" class="md-nav__link">
      PAW
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../topics/PIMD/" title="PIMD" class="md-nav__link">
      PIMD
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../topics/Planewaves/" title="Planewaves" class="md-nav__link">
      Planewaves
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../topics/Phonons/" title="Phonons" class="md-nav__link">
      Phonons
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../topics/PhononBands/" title="PhononBands" class="md-nav__link">
      PhononBands
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../topics/PhononWidth/" title="PhononWidth" class="md-nav__link">
      PhononWidth
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../topics/PortabilityNonRegression/" title="PortabilityNonRegression" class="md-nav__link">
      PortabilityNonRegression
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../topics/positron/" title="Positron" class="md-nav__link">
      Positron
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../topics/printing/" title="Printing" class="md-nav__link">
      Printing
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../topics/PseudosPAW/" title="PseudosPAW" class="md-nav__link">
      PseudosPAW
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../topics/q-points/" title="Q points" class="md-nav__link">
      Q points
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../topics/RandStopPow/" title="RandStopPow" class="md-nav__link">
      RandStopPow
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../topics/Recursion/" title="Recursion" class="md-nav__link">
      Recursion
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../topics/RPACorrEn/" title="RPACorrEn" class="md-nav__link">
      RPACorrEn
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../topics/SCFControl/" title="SCFControl" class="md-nav__link">
      SCFControl
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../topics/SCFAlgorithms/" title="SCFAlgorithms" class="md-nav__link">
      SCFAlgorithms
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../topics/SelfEnergy/" title="SelfEnergy" class="md-nav__link">
      SelfEnergy
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../topics/SmartSymm/" title="SmartSymm" class="md-nav__link">
      SmartSymm
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../topics/spinpolarisation/" title="Spinpolarisation" class="md-nav__link">
      Spinpolarisation
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../topics/STM/" title="STM" class="md-nav__link">
      STM
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../topics/Susceptibility/" title="Susceptibility" class="md-nav__link">
      Susceptibility
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../topics/TDDFT/" title="TDDFT" class="md-nav__link">
      TDDFT
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../topics/TDepES/" title="TDepES" class="md-nav__link">
      TDepES
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../topics/Temperature/" title="Temperature" class="md-nav__link">
      Temperature
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../topics/TransPath/" title="TransPath" class="md-nav__link">
      TransPath
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../topics/TuningSpeed/" title="TuningSpeed" class="md-nav__link">
      TuningSpeed
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../topics/Unfolding/" title="Unfolding" class="md-nav__link">
      Unfolding
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../topics/UnitCell/" title="UnitCell" class="md-nav__link">
      UnitCell
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../topics/vdw/" title="Vdw" class="md-nav__link">
      Vdw
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../topics/Verification/" title="Verification" class="md-nav__link">
      Verification
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../topics/Wannier/" title="Wannier" class="md-nav__link">
      Wannier
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../topics/Wavelets/" title="Wavelets" class="md-nav__link">
      Wavelets
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../topics/xc/" title="Xc" class="md-nav__link">
      Xc
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      

  


  <li class="md-nav__item md-nav__item--active md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-3" type="checkbox" id="nav-3" checked>
    
    <label class="md-nav__link" for="nav-3">
      Variables
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-3">
        Variables
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../" title="All Variables" class="md-nav__link">
      All Variables
    </a>
  </li>

        
          
          
          

  


  <li class="md-nav__item md-nav__item--active md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-3-2" type="checkbox" id="nav-3-2" checked>
    
    <label class="md-nav__link" for="nav-3-2">
      Abinit
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="2">
      <label class="md-nav__title" for="nav-3-2">
        Abinit
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../basic/" title="Basic" class="md-nav__link">
      Basic
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../bse/" title="Bethe-Salpeter" class="md-nav__link">
      Bethe-Salpeter
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../dev/" title="Developers" class="md-nav__link">
      Developers
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../dmft/" title="DMFT" class="md-nav__link">
      DMFT
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../eph/" title="Electron-Phonon" class="md-nav__link">
      Electron-Phonon
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../ffield/" title="Varff" class="md-nav__link">
      Varff
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../files/" title="File-handling" class="md-nav__link">
      File-handling
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../geo/" title="Geometry" class="md-nav__link">
      Geometry
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../gstate/" title="Ground-State" class="md-nav__link">
      Ground-State
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../gw/" title="GW" class="md-nav__link">
      GW
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../internal/" title="Internal" class="md-nav__link">
      Internal
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../paral/" title="Parallelism" class="md-nav__link">
      Parallelism
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../paw/" title="PAW" class="md-nav__link">
      PAW
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../dfpt/" title="DFPT" class="md-nav__link">
      DFPT
    </a>
  </li>

        
          
          
          

  


  <li class="md-nav__item md-nav__item--active">
    
    <input class="md-toggle md-nav__toggle" data-md-toggle="toc" type="checkbox" id="toc">
    
      
    
    
      <label class="md-nav__link md-nav__link--active" for="toc">
        Relaxation
      </label>
    
    <a href="./" title="Relaxation" class="md-nav__link md-nav__link--active">
      Relaxation
    </a>
    
      
<nav class="md-nav md-nav--secondary">
  
  
    
  
  
    <label class="md-nav__title" for="toc">Table of contents</label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#adpimd" title="adpimd" class="md-nav__link">
    adpimd
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#adpimd_gamma" title="adpimd_gamma" class="md-nav__link">
    adpimd_gamma
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#amu" title="amu" class="md-nav__link">
    amu
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#bmass" title="bmass" class="md-nav__link">
    bmass
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#cineb_start" title="cineb_start" class="md-nav__link">
    cineb_start
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#delayperm" title="delayperm" class="md-nav__link">
    delayperm
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#diismemory" title="diismemory" class="md-nav__link">
    diismemory
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#dilatmx" title="dilatmx" class="md-nav__link">
    dilatmx
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#dtion" title="dtion" class="md-nav__link">
    dtion
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#dynimage" title="dynimage" class="md-nav__link">
    dynimage
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#ecutsm" title="ecutsm" class="md-nav__link">
    ecutsm
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#friction" title="friction" class="md-nav__link">
    friction
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#fxcartfactor" title="fxcartfactor" class="md-nav__link">
    fxcartfactor
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#ga_algor" title="ga_algor" class="md-nav__link">
    ga_algor
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#ga_fitness" title="ga_fitness" class="md-nav__link">
    ga_fitness
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#ga_n_rules" title="ga_n_rules" class="md-nav__link">
    ga_n_rules
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#ga_opt_percent" title="ga_opt_percent" class="md-nav__link">
    ga_opt_percent
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#ga_rules" title="ga_rules" class="md-nav__link">
    ga_rules
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#getcell" title="getcell" class="md-nav__link">
    getcell
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#getvel" title="getvel" class="md-nav__link">
    getvel
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#getxcart" title="getxcart" class="md-nav__link">
    getxcart
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#getxred" title="getxred" class="md-nav__link">
    getxred
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#goprecon" title="goprecon" class="md-nav__link">
    goprecon
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#goprecprm" title="goprecprm" class="md-nav__link">
    goprecprm
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#iatcon" title="iatcon" class="md-nav__link">
    iatcon
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#iatfix" title="iatfix" class="md-nav__link">
    iatfix
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#iatfixx" title="iatfixx" class="md-nav__link">
    iatfixx
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#iatfixy" title="iatfixy" class="md-nav__link">
    iatfixy
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#iatfixz" title="iatfixz" class="md-nav__link">
    iatfixz
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#imgmov" title="imgmov" class="md-nav__link">
    imgmov
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#ionmov" title="ionmov" class="md-nav__link">
    ionmov
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#istatimg" title="istatimg" class="md-nav__link">
    istatimg
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#mdtemp" title="mdtemp" class="md-nav__link">
    mdtemp
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#mdwall" title="mdwall" class="md-nav__link">
    mdwall
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#mep_mxstep" title="mep_mxstep" class="md-nav__link">
    mep_mxstep
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#mep_solver" title="mep_solver" class="md-nav__link">
    mep_solver
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#natcon" title="natcon" class="md-nav__link">
    natcon
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#natfix" title="natfix" class="md-nav__link">
    natfix
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#natfixx" title="natfixx" class="md-nav__link">
    natfixx
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#natfixy" title="natfixy" class="md-nav__link">
    natfixy
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#natfixz" title="natfixz" class="md-nav__link">
    natfixz
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#nconeq" title="nconeq" class="md-nav__link">
    nconeq
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#neb_algo" title="neb_algo" class="md-nav__link">
    neb_algo
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#neb_spring" title="neb_spring" class="md-nav__link">
    neb_spring
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#nimage" title="nimage" class="md-nav__link">
    nimage
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#nnos" title="nnos" class="md-nav__link">
    nnos
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#noseinert" title="noseinert" class="md-nav__link">
    noseinert
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#ntime" title="ntime" class="md-nav__link">
    ntime
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#ntimimage" title="ntimimage" class="md-nav__link">
    ntimimage
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#optcell" title="optcell" class="md-nav__link">
    optcell
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#pimass" title="pimass" class="md-nav__link">
    pimass
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#pimd_constraint" title="pimd_constraint" class="md-nav__link">
    pimd_constraint
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#pitransform" title="pitransform" class="md-nav__link">
    pitransform
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#prtatlist" title="prtatlist" class="md-nav__link">
    prtatlist
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#qmass" title="qmass" class="md-nav__link">
    qmass
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#random_atpos" title="random_atpos" class="md-nav__link">
    random_atpos
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#restartxf" title="restartxf" class="md-nav__link">
    restartxf
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#signperm" title="signperm" class="md-nav__link">
    signperm
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#strfact" title="strfact" class="md-nav__link">
    strfact
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#string_algo" title="string_algo" class="md-nav__link">
    string_algo
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#strprecon" title="strprecon" class="md-nav__link">
    strprecon
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#strtarget" title="strtarget" class="md-nav__link">
    strtarget
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#tolimg" title="tolimg" class="md-nav__link">
    tolimg
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#tolmxde" title="tolmxde" class="md-nav__link">
    tolmxde
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#tolmxf" title="tolmxf" class="md-nav__link">
    tolmxf
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#vel" title="vel" class="md-nav__link">
    vel
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#vel_cell" title="vel_cell" class="md-nav__link">
    vel_cell
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#vis" title="vis" class="md-nav__link">
    vis
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#wtatcon" title="wtatcon" class="md-nav__link">
    wtatcon
  </a>
  
</li>
      
      
      
    </ul>
  
</nav>
    
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../vdw/" title="VdW" class="md-nav__link">
      VdW
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../w90/" title="Wannier90" class="md-nav__link">
      Wannier90
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../anaddb/" title="Anaddb" class="md-nav__link">
      Anaddb
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../optic/" title="Optic" class="md-nav__link">
      Optic
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../aim/" title="Aim" class="md-nav__link">
      Aim
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../external_parameters/" title="External Parameters" class="md-nav__link">
      External Parameters
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../varset_stats/" title="Statistics" class="md-nav__link">
      Statistics
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../connections/" title="Connections" class="md-nav__link">
      Connections
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-4" type="checkbox" id="nav-4">
    
    <label class="md-nav__link" for="nav-4">
      Lessons
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-4">
        Lessons
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../tutorials/welcome/" title="Overview" class="md-nav__link">
      Overview
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-4-2" type="checkbox" id="nav-4-2">
    
    <label class="md-nav__link" for="nav-4-2">
      Base Tutorials
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="2">
      <label class="md-nav__title" for="nav-4-2">
        Base Tutorials
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../tutorials/base1/" title="Base1" class="md-nav__link">
      Base1
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../tutorials/base2/" title="Base2" class="md-nav__link">
      Base2
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../tutorials/base3/" title="Base3" class="md-nav__link">
      Base3
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../tutorials/base4/" title="Base4" class="md-nav__link">
      Base4
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../tutorials/spin/" title="Spin" class="md-nav__link">
      Spin
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

        
          
          
          


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-4-3" type="checkbox" id="nav-4-3">
    
    <label class="md-nav__link" for="nav-4-3">
      PAW
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="2">
      <label class="md-nav__title" for="nav-4-3">
        PAW
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../tutorials/paw1/" title="PAW1" class="md-nav__link">
      PAW1
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../tutorials/paw2/" title="PAW2" class="md-nav__link">
      PAW2
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../tutorials/paw3/" title="PAW3" class="md-nav__link">
      PAW3
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

        
          
          
          


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-4-4" type="checkbox" id="nav-4-4">
    
    <label class="md-nav__link" for="nav-4-4">
      DFT_PLUS_U
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="2">
      <label class="md-nav__title" for="nav-4-4">
        DFT_PLUS_U
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../tutorials/dftu/" title="Dftu" class="md-nav__link">
      Dftu
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../tutorials/udet/" title="Udet" class="md-nav__link">
      Udet
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../tutorials/ucalc_crpa/" title="Ucalc crpa" class="md-nav__link">
      Ucalc crpa
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

        
          
          
          


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-4-5" type="checkbox" id="nav-4-5">
    
    <label class="md-nav__link" for="nav-4-5">
      DFPT
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="2">
      <label class="md-nav__title" for="nav-4-5">
        DFPT
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../tutorials/rf1/" title="RF1" class="md-nav__link">
      RF1
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../tutorials/rf2/" title="RF2" class="md-nav__link">
      RF2
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../tutorials/elastic/" title="Elastic" class="md-nav__link">
      Elastic
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../tutorials/ffield/" title="Ffield" class="md-nav__link">
      Ffield
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../tutorials/nlo/" title="Nlo" class="md-nav__link">
      Nlo
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../tutorials/dmft/" title="DMFT" class="md-nav__link">
      DMFT
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-4-7" type="checkbox" id="nav-4-7">
    
    <label class="md-nav__link" for="nav-4-7">
      EPH
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="2">
      <label class="md-nav__title" for="nav-4-7">
        EPH
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../tutorials/eph/" title="Eph" class="md-nav__link">
      Eph
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../tutorials/tdepes/" title="Tdepes" class="md-nav__link">
      Tdepes
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

        
          
          
          


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-4-8" type="checkbox" id="nav-4-8">
    
    <label class="md-nav__link" for="nav-4-8">
      Many-Body
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="2">
      <label class="md-nav__title" for="nav-4-8">
        Many-Body
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../tutorials/gw1/" title="GW1" class="md-nav__link">
      GW1
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../tutorials/gw2/" title="GW2" class="md-nav__link">
      GW2
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../tutorials/bse/" title="Bethe-Salpeter" class="md-nav__link">
      Bethe-Salpeter
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

        
          
          
          


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-4-9" type="checkbox" id="nav-4-9">
    
    <label class="md-nav__link" for="nav-4-9">
      Parallelism
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="2">
      <label class="md-nav__title" for="nav-4-9">
        Parallelism
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../tutorials/basepar/" title="Base" class="md-nav__link">
      Base
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../tutorials/paral_dfpt/" title="DFPT" class="md-nav__link">
      DFPT
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../tutorials/paral_gspw/" title="GSPW" class="md-nav__link">
      GSPW
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../tutorials/paral_gswvl/" title="GSWVL" class="md-nav__link">
      GSWVL
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../tutorials/paral_images/" title="Images" class="md-nav__link">
      Images
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../tutorials/paral_mbt/" title="Many-Body" class="md-nav__link">
      Many-Body
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../tutorials/paral_moldyn/" title="Moldyn" class="md-nav__link">
      Moldyn
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../tutorials/analysis_tools/" title="Analysis tools" class="md-nav__link">
      Analysis tools
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../tutorials/fold2bloch/" title="Fold2bloch" class="md-nav__link">
      Fold2bloch
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../tutorials/nuc/" title="Nuc" class="md-nav__link">
      Nuc
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../tutorials/optic/" title="Optic" class="md-nav__link">
      Optic
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../tutorials/positron/" title="Positron" class="md-nav__link">
      Positron
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../tutorials/source_code/" title="Source code" class="md-nav__link">
      Source code
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../tutorials/tddft/" title="Tddft" class="md-nav__link">
      Tddft
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../tutorials/wannier90/" title="Wannier90" class="md-nav__link">
      Wannier90
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-5" type="checkbox" id="nav-5">
    
    <label class="md-nav__link" for="nav-5">
      Theory
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-5">
        Theory
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../theory/wavefunctions/" title="Wavefunctions" class="md-nav__link">
      Wavefunctions
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../theory/mbt/" title="Many-Body Perturbation Theory" class="md-nav__link">
      Many-Body Perturbation Theory
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../theory/bse/" title="Bethe-Salpeter" class="md-nav__link">
      Bethe-Salpeter
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../theory/documents/" title="Documents" class="md-nav__link">
      Documents
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../theory/acknowledgments/" title="Acknowledgments" class="md-nav__link">
      Acknowledgments
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../theory/bibliography/" title="Bibliography" class="md-nav__link">
      Bibliography
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-6" type="checkbox" id="nav-6">
    
    <label class="md-nav__link" for="nav-6">
      Developers
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-6">
        Developers
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../developers/abimkdocs/" title="Abimkdocs" class="md-nav__link">
      Abimkdocs
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../developers/markdown/" title="Markdown" class="md-nav__link">
      Markdown
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../developers/testsuite/" title="Test Suite" class="md-nav__link">
      Test Suite
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../developers/codestats/" title="Code stats" class="md-nav__link">
      Code stats
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../developers/external_resources/" title="External Resources" class="md-nav__link">
      External Resources
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-7" type="checkbox" id="nav-7">
    
    <label class="md-nav__link" for="nav-7">
      About
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-7">
        About
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../about/release-notes/" title="Release Notes" class="md-nav__link">
      Release Notes
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../about/contributing/" title="Contributing" class="md-nav__link">
      Contributing
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../about/license/" title="License" class="md-nav__link">
      License
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              <div class="md-sidebar md-sidebar--secondary" data-md-component="toc">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    
<nav class="md-nav md-nav--secondary">
  
  
    
  
  
    <label class="md-nav__title" for="toc">Table of contents</label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#adpimd" title="adpimd" class="md-nav__link">
    adpimd
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#adpimd_gamma" title="adpimd_gamma" class="md-nav__link">
    adpimd_gamma
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#amu" title="amu" class="md-nav__link">
    amu
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#bmass" title="bmass" class="md-nav__link">
    bmass
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#cineb_start" title="cineb_start" class="md-nav__link">
    cineb_start
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#delayperm" title="delayperm" class="md-nav__link">
    delayperm
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#diismemory" title="diismemory" class="md-nav__link">
    diismemory
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#dilatmx" title="dilatmx" class="md-nav__link">
    dilatmx
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#dtion" title="dtion" class="md-nav__link">
    dtion
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#dynimage" title="dynimage" class="md-nav__link">
    dynimage
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#ecutsm" title="ecutsm" class="md-nav__link">
    ecutsm
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#friction" title="friction" class="md-nav__link">
    friction
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#fxcartfactor" title="fxcartfactor" class="md-nav__link">
    fxcartfactor
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#ga_algor" title="ga_algor" class="md-nav__link">
    ga_algor
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#ga_fitness" title="ga_fitness" class="md-nav__link">
    ga_fitness
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#ga_n_rules" title="ga_n_rules" class="md-nav__link">
    ga_n_rules
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#ga_opt_percent" title="ga_opt_percent" class="md-nav__link">
    ga_opt_percent
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#ga_rules" title="ga_rules" class="md-nav__link">
    ga_rules
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#getcell" title="getcell" class="md-nav__link">
    getcell
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#getvel" title="getvel" class="md-nav__link">
    getvel
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#getxcart" title="getxcart" class="md-nav__link">
    getxcart
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#getxred" title="getxred" class="md-nav__link">
    getxred
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#goprecon" title="goprecon" class="md-nav__link">
    goprecon
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#goprecprm" title="goprecprm" class="md-nav__link">
    goprecprm
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#iatcon" title="iatcon" class="md-nav__link">
    iatcon
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#iatfix" title="iatfix" class="md-nav__link">
    iatfix
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#iatfixx" title="iatfixx" class="md-nav__link">
    iatfixx
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#iatfixy" title="iatfixy" class="md-nav__link">
    iatfixy
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#iatfixz" title="iatfixz" class="md-nav__link">
    iatfixz
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#imgmov" title="imgmov" class="md-nav__link">
    imgmov
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#ionmov" title="ionmov" class="md-nav__link">
    ionmov
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#istatimg" title="istatimg" class="md-nav__link">
    istatimg
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#mdtemp" title="mdtemp" class="md-nav__link">
    mdtemp
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#mdwall" title="mdwall" class="md-nav__link">
    mdwall
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#mep_mxstep" title="mep_mxstep" class="md-nav__link">
    mep_mxstep
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#mep_solver" title="mep_solver" class="md-nav__link">
    mep_solver
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#natcon" title="natcon" class="md-nav__link">
    natcon
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#natfix" title="natfix" class="md-nav__link">
    natfix
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#natfixx" title="natfixx" class="md-nav__link">
    natfixx
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#natfixy" title="natfixy" class="md-nav__link">
    natfixy
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#natfixz" title="natfixz" class="md-nav__link">
    natfixz
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#nconeq" title="nconeq" class="md-nav__link">
    nconeq
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#neb_algo" title="neb_algo" class="md-nav__link">
    neb_algo
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#neb_spring" title="neb_spring" class="md-nav__link">
    neb_spring
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#nimage" title="nimage" class="md-nav__link">
    nimage
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#nnos" title="nnos" class="md-nav__link">
    nnos
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#noseinert" title="noseinert" class="md-nav__link">
    noseinert
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#ntime" title="ntime" class="md-nav__link">
    ntime
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#ntimimage" title="ntimimage" class="md-nav__link">
    ntimimage
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#optcell" title="optcell" class="md-nav__link">
    optcell
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#pimass" title="pimass" class="md-nav__link">
    pimass
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#pimd_constraint" title="pimd_constraint" class="md-nav__link">
    pimd_constraint
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#pitransform" title="pitransform" class="md-nav__link">
    pitransform
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#prtatlist" title="prtatlist" class="md-nav__link">
    prtatlist
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#qmass" title="qmass" class="md-nav__link">
    qmass
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#random_atpos" title="random_atpos" class="md-nav__link">
    random_atpos
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#restartxf" title="restartxf" class="md-nav__link">
    restartxf
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#signperm" title="signperm" class="md-nav__link">
    signperm
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#strfact" title="strfact" class="md-nav__link">
    strfact
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#string_algo" title="string_algo" class="md-nav__link">
    string_algo
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#strprecon" title="strprecon" class="md-nav__link">
    strprecon
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#strtarget" title="strtarget" class="md-nav__link">
    strtarget
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#tolimg" title="tolimg" class="md-nav__link">
    tolimg
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#tolmxde" title="tolmxde" class="md-nav__link">
    tolmxde
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#tolmxf" title="tolmxf" class="md-nav__link">
    tolmxf
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#vel" title="vel" class="md-nav__link">
    vel
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#vel_cell" title="vel_cell" class="md-nav__link">
    vel_cell
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#vis" title="vis" class="md-nav__link">
    vis
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#wtatcon" title="wtatcon" class="md-nav__link">
    wtatcon
  </a>
  
</li>
      
      
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          <div class="md-content">
            <article class="md-content__inner md-typeset">
              

                
                
                <!--
This file is automatically generated by mksite.py. All changes will be lost.
Change the input yaml files or the python code
-->

<h1 id="rlx-input-variables">rlx input variables<a class="headerlink" href="#rlx-input-variables" title="Permanent link">&para;</a></h1>
<p>This document lists and provides the description of the name (keywords) of the
rlx input variables to be used in the input file for the abinit executable.</p>
<h2 id="adpimd"><strong>adpimd</strong><a class="headerlink" href="#adpimd" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> ADiabatic Path-Integral Molecular Dynamics<br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-content="How to perform a PIMD calculation" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../topics/PIMD">topic_PIMD</a><br />
<em>Variable type:</em> integer<br />
<em>Dimensions:</em> scalar<br />
<em>Default value:</em> 0<br />
<em>Only relevant if:</em> <a class="wikilink codevar-wikilink" href="../rlx#imgmov">imgmov</a>==9 or <a class="wikilink codevar-wikilink" href="../rlx#imgmov">imgmov</a>==13  </p>
<details class="note"><summary>Test list (click to open). Rarely used, [1/920] in all abinit tests, [0/116] in abinit tutorials</summary><ul><li>v7:  <a class="wikilink abifile-wikilink" data-content="Hydrogen molecule: Test the Path-Integral Molecular Dynamics (PIMD) implementation in (N, V, T) ensemble. THe two thermostats (Nose-Hoover chains and Langevin) are tested together with the three coordinate systems (primitive,  normal mode,  staging). NOTE: MPI run with 2 procs is ok but fldiff comparison fails due to an extra section with --- Pseudopotential description --" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v7/Input/t08.in" target="_blank">t08.in</a></li></ul></details><p>Controls whether adiabatic Path-Integral Molecular Dynamics is performed or
not.<br />
The corresponding adiabaticity parameter is given by <a class="wikilink codevar-wikilink" href="../rlx#adpimd_gamma">adpimd_gamma</a>.  </p>
<p>If equal to 0, no adiabatic Path-Integral Molecular Dynamics (standard PIMD)
is performed.<br />
If equal to 1, adiabatic Path-Integral Molecular Dynamics is activated.<br />
Only relevant with <a class="wikilink codevar-wikilink" href="../rlx#pitransform">pitransform</a>=1 (normal mode transformation). In that
case,<br />
- the mass associated with to the zero-frequency mode is the true mass
<a class="wikilink codevar-wikilink" href="../rlx#amu">amu</a>,<br />
- the mass associated to the other higher frequency modes of the polymer
chains is equal to the normal mode mass divided by <a class="wikilink codevar-wikilink" href="../rlx#adpimd_gamma">adpimd_gamma</a>
(adiabaticity parameter),<br />
- the equation of motion on the zero-frequency mode is not thermostated.<br />
NOT YET USABLE</p>
<h2 id="adpimd_gamma"><strong>adpimd_gamma</strong><a class="headerlink" href="#adpimd_gamma" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> ADiabatic Path-Integral Molecular Dynamics: GAMMA factor<br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-content="How to perform a PIMD calculation" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../topics/PIMD">topic_PIMD</a><br />
<em>Variable type:</em> real<br />
<em>Dimensions:</em> scalar<br />
<em>Default value:</em> 1<br />
<em>Only relevant if:</em> <a class="wikilink codevar-wikilink" href="../rlx#adpimd">adpimd</a>==1 and <a class="wikilink codevar-wikilink" href="../rlx#imgmov">imgmov</a> in [9,13]  </p>
<details class="note"><summary>Test list (click to open). Rarely used, [1/920] in all abinit tests, [0/116] in abinit tutorials</summary><ul><li>v7:  <a class="wikilink abifile-wikilink" data-content="Hydrogen molecule: Test the Path-Integral Molecular Dynamics (PIMD) implementation in (N, V, T) ensemble. THe two thermostats (Nose-Hoover chains and Langevin) are tested together with the three coordinate systems (primitive,  normal mode,  staging). NOTE: MPI run with 2 procs is ok but fldiff comparison fails due to an extra section with --- Pseudopotential description --" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v7/Input/t08.in" target="_blank">t08.in</a></li></ul></details><p>Adiabaticity parameter to be used in adiabatic Path-Integral Molecular
Dynamics.<br />
NOT YET USABLE</p>
<h2 id="amu"><strong>amu</strong><a class="headerlink" href="#amu" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> Atomic Mass Units<br />
<em>Characteristics:</em> <a class="wikilink" href="../../user-guide/abinit#32-more-about-abinit-input-variables">EVOLVING</a><br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-content="How to perform a PIMD calculation" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../topics/PIMD">topic_PIMD</a>, <a class="wikilink topic-wikilink" data-content="How to compute phonon frequencies and modes, IR and Raman spectra, Born effective
charges, IR reflectivity ..." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../topics/Phonons">topic_Phonons</a>, <a class="wikilink topic-wikilink" data-content="How to specify the types of atoms that form the system" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../topics/AtomTypes">topic_AtomTypes</a>, <a class="wikilink topic-wikilink" data-content="How to perform some artificial modifications of the physics" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../topics/Artificial">topic_Artificial</a><br />
<em>Variable type:</em> real<br />
<em>Dimensions:</em> (<a class="wikilink codevar-wikilink" href="../basic#ntypat">ntypat</a>)<br />
<em>Default value:</em> None<br />
<em>Comment:</em> provided by a database of atomic masses.  </p>
<details class="note"><summary>Test list (click to open). Moderately used, [81/920] in all abinit tests, [2/116] in abinit tutorials</summary><ul><li>bigdft:  <a class="wikilink abifile-wikilink" data-content="H in isolated boundary conditions (plane waves,  PAW). Tests free boundary conditions Poisson solver for PAW." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/bigdft/Input/t23.in" target="_blank">t23.in</a>, <a class="wikilink abifile-wikilink" data-content="H in a box (PAW). PAW projectors are fitted to Gaussians." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/bigdft/Input/t31.in" target="_blank">t31.in</a>, <a class="wikilink abifile-wikilink" data-content="H in a box (PAW). PAW projectors are fitted to Gaussians." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/bigdft/Input/t32.in" target="_blank">t32.in</a> &hellip;</li><li>bigdft_paral:  <a class="wikilink abifile-wikilink" data-content="O2 in a box (NC). ABINIT routines are used (wvl_bigdft_comp=0) Test: DIIS,  potential/density mixing and parallelism" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/bigdft_paral/Input/t01.in" target="_blank">t01.in</a>, <a class="wikilink abifile-wikilink" data-content="O2 in a box (NC). ABINIT routines are used (wvl_bigdft_comp=0) Test: DIIS,  potential/density mixing and parallelism" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/bigdft_paral/Input/t01.in" target="_blank">t01.in</a> &hellip;</li><li>built-in:  <a class="wikilink abifile-wikilink" data-content="GaAs crystal with few bands." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/built-in/Input/testin_wannier90.in" target="_blank">testin_wannier90.in</a> &hellip;</li><li>etsf_io:  <a class="wikilink abifile-wikilink" data-content="Si2 molecule,  static,  spin-polarized. Same system as test 49 of v2,  except that nctime is non-zero. Test ionmov=6 (Verlet) as well as ionmov=7." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/etsf_io/Input/t21.in" target="_blank">t21.in</a> &hellip;</li><li>fast:  <a class="wikilink abifile-wikilink" data-content="21. Conduct same relaxation but use ionmov=1 (molecular dynamics)." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/fast/Input/t21.in" target="_blank">t21.in</a>, <a class="wikilink abifile-wikilink" data-content="move with natfix. Output also GEO files" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/fast/Input/t29.in" target="_blank">t29.in</a> &hellip;</li><li>gpu:  <a class="wikilink abifile-wikilink" data-content="GaAs phonons calculation - norm-conserving psp Test the automatic detection of the GPU and the automatic setting of use_gpu_cuda keyword according to the requested calculation. Ground-states calculations should be run with use_gpu_cuda activated (except if set to 0 in input file) while response fonction calculations should be run with use_gpu_cuda deactivated (at present : ABINIT v7.0)." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/gpu/Input/t01.in" target="_blank">t01.in</a> &hellip;</li><li>mpiio:  <a class="wikilink abifile-wikilink" data-content="GaAs linear response with NC. Test the parallelisation on perturbations" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/mpiio/Input/t69.in" target="_blank">t69.in</a>, <a class="wikilink abifile-wikilink" data-content="GaAs linear response with NC. Test the parallelisation on perturbations" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/mpiio/Input/t69.in" target="_blank">t69.in</a> &hellip;</li><li>paral:  <a class="wikilink abifile-wikilink" data-content="Gold,  8 atom supercell,  with PAW. Test lotf_nitex,  lotf_classic,  lotf_version,  lotf_nneigx. These variables,  concerning LOTF method,  are taken into account only if the enable_lotf=&amp;quot;yes&amp;quot; is used in configuration." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/paral/Input/t41.in" target="_blank">t41.in</a>, <a class="wikilink abifile-wikilink" data-content="FCC Al metallic;  10 phonon at 1/4 1/8 1/8" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/paral/Input/t53.in" target="_blank">t53.in</a>, <a class="wikilink abifile-wikilink" data-content="FCC Al metallic;  10 phonon at 1/4 1/8 1/8" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/paral/Input/t53.in" target="_blank">t53.in</a> &hellip;</li><li>seq:  <a class="wikilink abifile-wikilink" data-content="GaAs,  zinc-blende structure. Compute Berry phase along 0 0 1,  using a 4x4x10 sampling." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/seq/Input/tsv2_82.in" target="_blank">tsv2_82.in</a> &hellip;</li><li>tutorespfn:  <a class="wikilink abifile-wikilink" data-content="Prepare the computation of linear and non-linear optic properties of GaAs crystal : ground-state with few bands, then non-SCF with a larger number of bands,  then ddk for different directions Note that the k point sampling shoud be finer for significant results. The cut-off energy is also too low." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/tutorespfn/Input/toptic_1.in" target="_blank">toptic_1.in</a>, <a class="wikilink abifile-wikilink" data-content="Prepare the computation of linear optic properties (for the imaginary spectrum only) of GaAs crystal : ground-state with few bands, then non-SCF with a larger number of bands,  then ddk for different directions Note that the k point sampling shoud be finer for significant results. The cut-off energy is also too low." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/tutorespfn/Input/toptic_3.in" target="_blank">toptic_3.in</a> &hellip;</li><li>v1:  <a class="wikilink abifile-wikilink" data-content="SiH4 : molecular dynamics with partially constrained atomic positions. 1 special k-point,  in a box 8x8x8 (too small),  and 8Ha cut-off. Test ionmov=1" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v1/Input/t86.in" target="_blank">t86.in</a> &hellip;</li><li>v2:  <a class="wikilink abifile-wikilink" data-content="Linear chain of Ge2 molecules (2 atoms per unit cell), using a local (Starkloff-Joannopoulos) pseudopotential, and no exchange-correlation (ixc=0). Uses 4 k-points. Computation of the second derivative of the total energy with respect to a atomic displacement along the chain, with q(0 0 0) wavevector. (see test.ge.xc0 of RESPFN)" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v2/Input/t01.in" target="_blank">t01.in</a>, <a class="wikilink abifile-wikilink" data-content="Linear chain of Ge2 molecules (2 atoms per unit cell), using a local (Starkloff-Joannopoulos) pseudopotential, with exchange-correlation (ixc=3). Uses 2 k-points. Computation of the second derivative of the total energy with respect to a atomic displacement along the chain, with q(0 0 0) wavevector. Same as test 1,  except ixc and the k-points,  and except that it doesn't uses multi-data mode,  so GS and RF calculations are separated. (see test.ge.xc5 of RESPFN)" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v2/Input/t03.in" target="_blank">t03.in</a>, <a class="wikilink abifile-wikilink" data-content="Linear chain of Si2 molecules (2 atoms per unit cell), using a separable pseudopotential,  with ixc=3. Computation of the second derivative of the total energy with respect to a atomic displacement perpendicular to the chain,  with q(0 0 1/2) wavevector. (see test.si.chain of RESPFN)" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v2/Input/t04.in" target="_blank">t04.in</a> &hellip;</li><li>v3:  <a class="wikilink abifile-wikilink" data-content="Linear chain of Ge2 molecules (2 atoms per unit cell), using a local (Starkloff-Joannopoulos) pseudopotential, and no exchange-correlation (ixc=0). Uses 4 k-points. Computation of the second derivative of the total energy with respect to a atomic displacement along the chain, with q(0 0 0) wavevector. Similar to test v2 #1,  except that it uses dilatmx=1.1 ." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v3/Input/t06.in" target="_blank">t06.in</a>, <a class="wikilink abifile-wikilink" data-content="H2 molecule,  using a local pseudopotential,  and GGA. Similar to test 33 of v2. Use the x-only PBE functional. Compute the interatomic force constant for the displacement of the second atom along x. RF calculation of the 2DTE gives 189.10691 Ha while finite differences gives : with delta(xred)=0.0002,  189.10790 Ha with delta(xred)=0.0001,  189.10703 Ha . Combining the results gives 189.10675 Ha. The agreement is quite good,  and might likely be improved by increasing tolvrs." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v3/Input/t07.in" target="_blank">t07.in</a>, <a class="wikilink abifile-wikilink" data-content="H2 molecule,  using a local pseudopotential,  and GGA. Similar to test 33 of v2,  and preceeding test, but use the full XC PBE functional. Also,  tolvrs is better. Compute the interatomic force constant for the displacement of the second atom along x. RF calculation of the 2DTE gives 188.81162533 Ha while finite differences gives : with delta(xred)=0.0002,  188.81213804 Ha with delta(xred)=0.0001,  188.81175352 Ha . Combining the results gives 188.81162525 Ha." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v3/Input/t08.in" target="_blank">t08.in</a> &hellip;</li><li>v4:  <a class="wikilink abifile-wikilink" data-content="2. Generate first-order responses for GaAs (zinc-blende) (from test v2 #98). Compute responses at Gamma: 1) Ground state calculation,  with a k-point sampling in the IBZ only; 2) Computation of the ddk response; 3) Computation of the dynamical matrix at Gamma,  as well as the Born effective charges,  and the dielectric tensor; Perform five datasets: * First dataset is the ground state calculation, * Second dataset compute ddk perturbation with the use of Legendre Polynomials for the nl operator, * Third dataset compute phonon and elec. field perturbation with the use of Legendre Polynomials for the nl operator, * Fourth dataset compute ddk perturbation with the use of spherical harmonics for the nl operator, * Fifth dataset compute phonon and elec. field perturbation with the use of spherical harmonics for the nl operator. Datasets 2 and 3 must give the same results as 4 and 5." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v4/Input/t02.in" target="_blank">t02.in</a>, <a class="wikilink abifile-wikilink" data-content="GaAs,  zinc-blende structure. Generate the data for optic (linear and non-linear coefficients). to be analysed in the test 57." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v4/Input/t56.in" target="_blank">t56.in</a> &hellip;</li><li>v5:  <a class="wikilink abifile-wikilink" data-content="Test to prepare a conductivity calculation (test v5#t10) Ground state calculation for 4 Al atoms The number of bands is unrealistic. This test produces files that are inputs for test 10. It tests the following parts of the code: optic_paw,  optic_paw_core Note that,  in addition to the psp file,  the core wave functions file is needed." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v5/Input/t09.in" target="_blank">t09.in</a>, <a class="wikilink abifile-wikilink" data-content="H2 molecule,  using a local pseudopotential,  and Wu-Cohen GGA. Similar to test v3#08 Compute the interatomic force constant for the displacement of the second atom along x. RF calculation of the 2DTE gives 189.49410411 Ha while finite differences gives : with delta(xred)=0.0002,  189.4945677  Ha with delta(xred)=0.0001,  189.49422003  Ha . Combining the results gives 189.4941041 Ha. Perfect !" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v5/Input/t29.in" target="_blank">t29.in</a>, <a class="wikilink abifile-wikilink" data-content="H2 molecule,  using a local pseudopotential, and the Wu-Cohen GGA. Similar to Test v3#48 Compute the total energy and force as a displacement of the second atom along x. Direct calculation of the gradient with respect to reduced coordinates gives -3.469435369425Ha while finite differences gives : with delta(xred)=0.0002,  -3.4694479488 Ha with delta(xred)=0.0001,  -3.4694385143 Ha . Combining the results gives -3.4694353695 Ha. Excellent." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v5/Input/t48.in" target="_blank">t48.in</a> &hellip;</li><li>v6:  <a class="wikilink abifile-wikilink" data-content="Tests series for the mGGA implementation (from 31 to 33) Silicon bulk. Two datasets are run. One (classic) LDA calculation with ixc 7. One with a fake native mGGA (fake1,  i.e. ixc 31) and using effmass_free=1.01. Both calculation should give the same total energy (change in kinetic energy due to effmass_free is compensated by the fake mGGA using kinetic energy density: taur). (Two other datasets are also run for spin polarized case)" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v6/Input/t31.in" target="_blank">t31.in</a>, <a class="wikilink abifile-wikilink" data-content="Test the mGGA implementation. Silicon bulk. Two datasets are run. One with a fake native mGGA (fake2,  i.e. ixc 32) involving laplacian of the density. One with a fake native mGGA (fake3,  i.e. ixc 33) involving gradient of the density. Both calculation should give the same total and exchange-correlation energies. (Two other datasets are also run for spin polarized case)" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v6/Input/t32.in" target="_blank">t32.in</a> &hellip;</li><li>v7:  <a class="wikilink abifile-wikilink" data-content="Hydrogen molecule: Test the Path-Integral Molecular Dynamics (PIMD) implementation in (N, V, T) ensemble. THe two thermostats (Nose-Hoover chains and Langevin) are tested together with the three coordinate systems (primitive,  normal mode,  staging). NOTE: MPI run with 2 procs is ok but fldiff comparison fails due to an extra section with --- Pseudopotential description --" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v7/Input/t08.in" target="_blank">t08.in</a>, <a class="wikilink abifile-wikilink" data-content="Boron,  32 atoms - Thomas-Fermi method (without orbitals) Test the temperature-dependent XC functional (ixc=50) (Ichimaru et al Phys. Rep. 149,  91-205 (1987) Test the von Weizsacker gradient correction (tfkinfunc=11 or 12)" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v7/Input/t11.in" target="_blank">t11.in</a> &hellip;</li><li>v8:  <a class="wikilink abifile-wikilink" data-content="H2 molecule: PIMD simulation with Langevin thermostat. Test of restart with HIST.nc file. Test of linear constraint." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v8/Input/t05.in" target="_blank">t05.in</a> &hellip;</li><li>wannier90:  <a class="wikilink abifile-wikilink" data-content="GaAs crystal with few bands. Very quick built-in test,  to check that Wannier90 is installed." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/wannier90/Input/t00.in" target="_blank">t00.in</a>, <a class="wikilink abifile-wikilink" data-content="Gallium Arsenide,  zinc-blende. 8 bands. prtwant=2 Should still be completed by the call to wannier90,  and subsequent fldiff use..." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/wannier90/Input/t01.in" target="_blank">t01.in</a>, <a class="wikilink abifile-wikilink" data-content="The same as in 01 but using PAW." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/wannier90/Input/t02.in" target="_blank">t02.in</a> &hellip;</li></ul></details><p>Gives the masses in atomic mass units for each kind of atom in cell. These
masses are used in performing molecular dynamical atomic motion if
<a class="wikilink codevar-wikilink" href="../rlx#ionmov">ionmov</a>=1, 6, 7 or 8. They are also used in phonon calculations, in the
diagonalization of the dynamical matrix. Note that one may set all masses to 1
for certain cases in which merely structural relaxation is desired and not
actual molecular dynamics.</p>
<p>Using 1986 recommended values, 1 atomic mass unit = 1.6605402e-27 kg. In this
unit the mass of Carbon 12 is exactly 12.</p>
<p>A database of atomic masses is provided, giving default values. Note that the
default database uses mixed isotope masses (for Carbon the natural occurrence
of Carbon 13 is taken into account). The values are those recommended by the
commission on Atomic Weights and Isotopic Abundances, Inorganic Chemistry
Division, IUPAC, in _ Pure Appl. Chem. _ ** 60 ** , 841 (1988). For Tc, Pm, Po
to Ac, Pa and beyond U, none of the isotopes has a half-life greater than
3.0d10 years, and the values provided in the database do not come from that
source.</p>
<p>For alchemical pseudoatoms, the masses of the constituents atoms are mixed,
according to the alchemical mixing coefficients <a class="wikilink codevar-wikilink" href="../gstate#mixalch">mixalch</a></p>
<p>In most cases, the use of <a class="wikilink codevar-wikilink" href="../rlx#amu">amu</a> will be as a static (non-evolving) variable.
However, the possibility to have different values of <a class="wikilink codevar-wikilink" href="../rlx#amu">amu</a> for different
images has been coded. A population of cells with different atomic
characteristics can thus be considered, and can be made to evolve, e.g. with a
genetic algorithm (not coded in v7.0.0 though).</p>
<h2 id="bmass"><strong>bmass</strong><a class="headerlink" href="#bmass" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> Barostat MASS<br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-content="How to perform a molecular dynamics calculation" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../topics/MolecularDynamics">topic_MolecularDynamics</a><br />
<em>Variable type:</em> real<br />
<em>Dimensions:</em> scalar<br />
<em>Default value:</em> 10  </p>
<details class="note"><summary>Test list (click to open). Rarely used, [1/920] in all abinit tests, [0/116] in abinit tutorials</summary><ul><li>v5:  <a class="wikilink abifile-wikilink" data-content="This is a test,  in the norm conserving case,  of the Molecular Dynamics flag ionmov==13. With the values of optcell=0,  1,  2 (isothermal (0),  isothermal/isenthalpic with homogeneous cell deformations (1) or full cell deformations (2).) This is the reversible integrator of Martyna,  Tuckerman,  Tobias and Klein. (Mol. Phys.,  1996,  p 1117). XG20090912 : Why is the result of dataset 1 and dataset 4 equivalent,  while the bmass varies by a factor of 2 ?" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v5/Input/t01.in" target="_blank">t01.in</a></li></ul></details><p>bmass is the mass of the barostat when <a class="wikilink codevar-wikilink" href="../rlx#ionmov">ionmov</a>=13 (constant pressure
molecular dynamics)</p>
<h2 id="cineb_start"><strong>cineb_start</strong><a class="headerlink" href="#cineb_start" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> Climbing-Image Nudged Elastic Band: STARTing iteration<br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-content="How to calculate transition paths" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../topics/TransPath">topic_TransPath</a><br />
<em>Variable type:</em> integer<br />
<em>Dimensions:</em> scalar<br />
<em>Default value:</em> 7<br />
<em>Only relevant if:</em> <a class="wikilink codevar-wikilink" href="../rlx#imgmov">imgmov</a>== 5 and <a class="wikilink codevar-wikilink" href="../rlx#neb_algo">neb_algo</a>==2  </p>
<details class="note"><summary>Test list (click to open). Rarely used, [1/920] in all abinit tests, [0/116] in abinit tutorials</summary><ul><li>v6:  <a class="wikilink abifile-wikilink" data-content="Hydrogen diatomic molecule in a cell,  close to BCC Test the string method : 6 images,  exploring the transition path. Similar to test 22 (but with string)." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v6/Input/t24.in" target="_blank">t24.in</a></li></ul></details><p>Gives the index of the first CI-NEB iteration..<br />
The CI-NEB method constitutes a small modification to the NEB method allowing
a rigorous convergence to the saddle point. As the image with the highest
energy has to be identified, the calculation begins with several iterations of
the standard NEB algorithm. The effective CI-NEB begins at the <a class="wikilink codevar-wikilink" href="../rlx#cineb_start">cineb_start</a>
iteration.<br />
_ See: J. Chem. Phys. 113, 9901 (2000). _</p>
<h2 id="delayperm"><strong>delayperm</strong><a class="headerlink" href="#delayperm" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> DELAY between trials to PERMUTE atoms<br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-content="How to perform a molecular dynamics calculation" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../topics/MolecularDynamics">topic_MolecularDynamics</a><br />
<em>Variable type:</em> integer<br />
<em>Dimensions:</em> scalar<br />
<em>Default value:</em> 0  </p>
<details class="note"><summary>Test list (click to open). Rarely used, [2/920] in all abinit tests, [0/116] in abinit tutorials</summary><ul><li>v2:  <a class="wikilink abifile-wikilink" data-content="Ge liquid. Test of Nose and Langevin dynamics. 2 atoms in a cell. Allows 10 time steps." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v2/Input/t87.in" target="_blank">t87.in</a>, <a class="wikilink abifile-wikilink" data-content="Basic test of geometry optimization for water First computing the ground state of a close configuration Uses 5 different methods to find the relaxed positions" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v2/Input/t88.in" target="_blank">t88.in</a></li></ul></details><p>Delay (number of time steps) between trials to permute two atoms, in view of
accelerated search of minima. Still in development. See the routine
moldyn.F90. See also <a class="wikilink codevar-wikilink" href="../rlx#signperm">signperm</a>. When <a class="wikilink codevar-wikilink" href="../rlx#delayperm">delayperm</a> is zero, there is not
permutation trials.</p>
<h2 id="diismemory"><strong>diismemory</strong><a class="headerlink" href="#diismemory" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> Direct Inversion in the Iterative Subspace MEMORY<br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-content="How to perform a molecular dynamics calculation" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../topics/MolecularDynamics">topic_MolecularDynamics</a><br />
<em>Variable type:</em> integer<br />
<em>Dimensions:</em> scalar<br />
<em>Default value:</em> 8  </p>
<details class="note"><summary>Test list (click to open). Rarely used, [0/920] in all abinit tests, [0/116] in abinit tutorials</summary></details><p>Gives the maximum number of &ldquo;time&rdquo; steps for which the forces and stresses are
stored, and taken into account in the DIIS algorithm (<a class="wikilink codevar-wikilink" href="../rlx#ionmov">ionmov</a>=20) to find
zero-force and stress configurations.</p>
<h2 id="dilatmx"><strong>dilatmx</strong><a class="headerlink" href="#dilatmx" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> lattice DILATation : MaXimal value<br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-content="How to perform a geometry optimization" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../topics/GeoOpt">topic_GeoOpt</a><br />
<em>Variable type:</em> real<br />
<em>Dimensions:</em> scalar<br />
<em>Default value:</em> 1.0  </p>
<details class="note"><summary>Test list (click to open). Moderately used, [66/920] in all abinit tests, [20/116] in abinit tutorials</summary><ul><li>gpu:  <a class="wikilink abifile-wikilink" data-content="Silicon - diamond structure - PAW - volume relaxation Test the use of GPU within Projector Augmented-Wave formalism. Test structure relaxation (multiple calls to gstate routine)." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/gpu/Input/t04.in" target="_blank">t04.in</a> &hellip;</li><li>mpiio:  <a class="wikilink abifile-wikilink" data-content="C-diamond,  Bulk,  2 atoms,  paral_kgb,  with PAW. Test of ground state with different occs(7 and 0), and also ionmov 2. Only with 0 and 4 procs." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/mpiio/Input/t22.in" target="_blank">t22.in</a>, <a class="wikilink abifile-wikilink" data-content="C-diamond,  Bulk,  2 atoms,  paral_kgb,  with PAW. Test the triple parallelisation. We cannot check the distribution npband*npfft*npkpt=2*2*2=8 processors, this number being not allowed in the test procedure. Here,  we only test the parallelisation over FFT and kpoints: npband*npfft*npkpt=1*2*2=4 processors. In test tY.in we check other distributions for guarantee." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/mpiio/Input/t24.in" target="_blank">t24.in</a> &hellip;</li><li>paral:  <a class="wikilink abifile-wikilink" data-content="C-diamond,  Bulk,  2 atoms,  paral_kgb,  with PAW. Test of ground state with different occs(7 and 0), and also ionmov 2. Only with 0 and 4 procs." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/paral/Input/t22.in" target="_blank">t22.in</a>, <a class="wikilink abifile-wikilink" data-content="C-diamond,  Bulk,  2 atoms,  paral_kgb,  with PAW. Test the triple parallelisation. We cannot check the distribution npband*npfft*npkpt=2*2*2=8 processors, this number being not allowed in the test procedure. Here,  we only test the parallelisation over FFT and kpoints: npband*npfft*npkpt=1*2*2=4 processors. In test tY.in we check other distributions for guarantee." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/paral/Input/t24.in" target="_blank">t24.in</a>, <a class="wikilink abifile-wikilink" data-content="C-diamond,  Bulk,  2 atoms,  paral_kgb,  with PAW. Test of ground state with different occs(7 and 0), and also ionmov 2. Only with 4 procs. Same of test R:test the automatic parallelisation" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/paral/Input/t26.in" target="_blank">t26.in</a> &hellip;</li><li>seq:  <a class="wikilink abifile-wikilink" data-content="PbTiO3 in the tetragonal geometry. Test berryopt=14 (finite reduced electric field calculation, relaxing cell parameters,  e.g. optcell=2) (XG120616 : tranferred to tests/seq)" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/seq/Input/tsv6_121.in" target="_blank">tsv6_121.in</a>, <a class="wikilink abifile-wikilink" data-content="PbTiO3 in the tetragonal geometry. Test berryopt=16 (finite reduced electric displacement field calculation, relaxing cell parameters,  e.g. optcell=2) (XG120616 : tranferred to tests/seq)" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/seq/Input/tsv6_122.in" target="_blank">tsv6_122.in</a>, <a class="wikilink abifile-wikilink" data-content="PbTiO3 in the tetragonal geometry. Test berryopt=4 (finite electric field calculation,  relaxing cell parameters,  e.g. optcell=2) (XG120616 : tranferred to tests/seq)" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/seq/Input/tsv6_123.in" target="_blank">tsv6_123.in</a> &hellip;</li><li>tutorespfn:  <a class="wikilink abifile-wikilink" data-content="AlAs in hypothetical wurzite (hexagonal) structure Structural optimization run" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/tutorespfn/Input/telast_1.in" target="_blank">telast_1.in</a>, <a class="wikilink abifile-wikilink" data-content="Al fcc metal - elastic constant calculation" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/tutorespfn/Input/telast_6.in" target="_blank">telast_6.in</a>, <a class="wikilink abifile-wikilink" data-content="Finite difference calculation of the Born effective charges of AlAs" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/tutorespfn/Input/tffield_1.in" target="_blank">tffield_1.in</a> &hellip;</li><li>tutorial:  <a class="wikilink abifile-wikilink" data-content="Crystalline silicon : computation of the optimal lattice parameter Convergence with respect to the number of k points." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/tutorial/Input/tbase3_4.in" target="_blank">tbase3_4.in</a>, <a class="wikilink abifile-wikilink" data-content="Crystalline aluminum : optimization of the lattice parameter at fixed number of k points and broadening." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/tutorial/Input/tbase4_1.in" target="_blank">tbase4_1.in</a>, <a class="wikilink abifile-wikilink" data-content="Crystalline aluminum : optimization of the lattice parameter Convergence with respect to k points." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/tutorial/Input/tbase4_2.in" target="_blank">tbase4_2.in</a> &hellip;</li><li>v1:  <a class="wikilink abifile-wikilink" data-content="2 special k point Si in 2-atom diamond unit cell. Test dilatmx. Increase it,  the number of planewaves increases,  but the total energy stays practically the same. The small fluctuations (-8.7967199088907, -8.7967198524822, -8.7967198520954 ) come from varying the FFT grid size,  with effect on the xc functional. It has been checked that using exactly the same FFT grid for the three cases gives the same energy. It has been checked also that larger fluctuations arise with intxc=0." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v1/Input/t77.in" target="_blank">t77.in</a>, <a class="wikilink abifile-wikilink" data-content="2 special k point Si in 2-atom diamond unit cell. Optimization of unit cell volume (optcell=1). Use 2 datasets,  and the variable getcell." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v1/Input/t78.in" target="_blank">t78.in</a>, <a class="wikilink abifile-wikilink" data-content="H2 molecule in a cell of variable size along the H2 chain =&amp;gt; H polymer. Optimization of cell size and atomic distances (optcell=4)." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v1/Input/t79.in" target="_blank">t79.in</a> &hellip;</li><li>v2:  <a class="wikilink abifile-wikilink" data-content="Arsenic in rhombohedral structure. Optimization of unit cell size and shape. With 4x4x4 k point grid,  ecut 3.0,  occopt 4 and tsmear 0.06d0, one get acell 3*7.325  angdeg 3*57.36 and xred 0.2277 These input parameters are too small,  but still give a realistic geometry,  since with the much better parameters 12x12x12 k point grid,  ecut 12.0,  occopt 4 and tsmear 0.04d0, one gets acell 3*7.633  angdeg 3*54.95 and xred 0.2296,  while the experimental values are acell 3*7.751   angdeg 3*54.554 and xred 0.2276" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v2/Input/t44.in" target="_blank">t44.in</a>, <a class="wikilink abifile-wikilink" data-content="Li,  simple cubic,  to test accurately RF in the metallic case. With non-linear XC core correction and non-zero ecutsm and dilatmx. Similar to test 92,  except ecutsm and dilatmx,  and no q(1/4 0 0). The total energy of the 1-atom cell is -.29337493523588 Ha The 2DTE with respect to a q(0 0 0) phonon is 5.5619d-8 . The 2DTE with respect to a transverse q(1/2 0 0) phonon is -0.12068188927 Ha." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v2/Input/t96.in" target="_blank">t96.in</a>, <a class="wikilink abifile-wikilink" data-content="Li,  simple cubic,  frozen-phonon calculations corresponding to test 96 (with non-zero ecutsm and dilatmx). The total energy of the double cell is -0.58674987047187 Ha, in excellent agreement with the result of test 96. The frozen-phonon (from energy) calculation of the 2DTE for the q(1/2 0 0) case gives -0.1206807 Ha . The frozen-phonon (from forces) calculation of the 2DTE for the q(1/2 0 0) case gives -0.1206794 Ha . Both are in rather good agreement with the result of test 96." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v2/Input/t97.in" target="_blank">t97.in</a> &hellip;</li><li>v3:  <a class="wikilink abifile-wikilink" data-content="Linear chain of Ge2 molecules (2 atoms per unit cell), using a local (Starkloff-Joannopoulos) pseudopotential, and no exchange-correlation (ixc=0). Uses 4 k-points. Computation of the second derivative of the total energy with respect to a atomic displacement along the chain, with q(0 0 0) wavevector. Similar to test v2 #1,  except that it uses dilatmx=1.1 ." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v3/Input/t06.in" target="_blank">t06.in</a>, <a class="wikilink abifile-wikilink" data-content="Magnesium HCP. Optimize the cell volume and shape using different constraints. Checks whether the stresses are symmetric,  and whether the optimisation algorithm does not break the symmetry." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v3/Input/t42.in" target="_blank">t42.in</a> &hellip;</li><li>v4:  <a class="wikilink abifile-wikilink" data-content="2. Generate first-order responses for GaAs (zinc-blende) (from test v2 #98). Compute responses at Gamma: 1) Ground state calculation,  with a k-point sampling in the IBZ only; 2) Computation of the ddk response; 3) Computation of the dynamical matrix at Gamma,  as well as the Born effective charges,  and the dielectric tensor; Perform five datasets: * First dataset is the ground state calculation, * Second dataset compute ddk perturbation with the use of Legendre Polynomials for the nl operator, * Third dataset compute phonon and elec. field perturbation with the use of Legendre Polynomials for the nl operator, * Fourth dataset compute ddk perturbation with the use of spherical harmonics for the nl operator, * Fifth dataset compute phonon and elec. field perturbation with the use of spherical harmonics for the nl operator. Datasets 2 and 3 must give the same results as 4 and 5." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v4/Input/t02.in" target="_blank">t02.in</a>, <a class="wikilink abifile-wikilink" data-content="Test of the strain perturbation with GGA  and spin polarization This is the same as t79,  bcc Fe, but with the PBE GGA.  This calculation is restricted to one unaxial strain,  which adequately tests the polarization-dependent parts of the strain GGA code." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v4/Input/t59.in" target="_blank">t59.in</a>, <a class="wikilink abifile-wikilink" data-content="Test of the strain perturbation with spin polarization and metallic occupation of states Fe in the bcc structure.  This tests both the introduction of spin polarization into strain response function calculations and an improved treatment of the first-order fermi energy for Q=0 metallic calculations.  Similar data (with much larger cutoffs,  tighter convergence tolerances,  and larger k sample) was used test the elastic tensor in comparison with numerical derivatives of the stress from GS calculations." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v4/Input/t79.in" target="_blank">t79.in</a> &hellip;</li><li>v5:  <a class="wikilink abifile-wikilink" data-content="This is a test,  in the norm conserving case,  of the Molecular Dynamics flag ionmov==13. With the values of optcell=0,  1,  2 (isothermal (0),  isothermal/isenthalpic with homogeneous cell deformations (1) or full cell deformations (2).) This is the reversible integrator of Martyna,  Tuckerman,  Tobias and Klein. (Mol. Phys.,  1996,  p 1117). XG20090912 : Why is the result of dataset 1 and dataset 4 equivalent,  while the bmass varies by a factor of 2 ?" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v5/Input/t01.in" target="_blank">t01.in</a>, <a class="wikilink abifile-wikilink" data-content="White tin,  body-centered tetragonal. This test checks that in an optcell 2 run,  the symmetry of the body-centered case is not broken. This test is run for two values of strprecon also, to test the strprecon variable. Coded and contributed by J. Zwanziger." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v5/Input/t77.in" target="_blank">t77.in</a>, <a class="wikilink abifile-wikilink" data-content="Make a simple geometry relaxation and validate the XML output (testing prtxml)." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v5/Input/t79.in" target="_blank">t79.in</a> &hellip;</li><li>v6:  <a class="wikilink abifile-wikilink" data-content="Bismuth,  rhombohedral,  two atoms per unit cell Scalar relativistic. Relax four different images,  using the steepest descent algorithm." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v6/Input/t21.in" target="_blank">t21.in</a>, <a class="wikilink abifile-wikilink" data-content="Genetic algorithm structure random search. Hydrogen has been used as an example. No physical meaning. Number of images is only 10 but experience shows that at least 20 are fine." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v6/Input/t27.in" target="_blank">t27.in</a>, <a class="wikilink abifile-wikilink" data-content="Hydrure of Lithium-Boron,  in a face-centered monoclinic cell. Test the recognition of the space group (C2/c (# 15)). Was lacking a case before Dec 2011." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v6/Input/t30.in" target="_blank">t30.in</a> &hellip;</li><li>v7:  <a class="wikilink abifile-wikilink" data-content="Nickel ferromagnetic fcc structure with an ONCVPSP pseudo containing valence density. We test several improvements that are available when the NC file contains the PS valence charge. More specifically,  we test the initialization of the density from atomic quantities (initro.F90), the correction to the forces due to the residuals and the extrapolation of the density for structural relaxations (well,  xred is fixed by symmetry). We also test the treatment of the core charge in G-space in the case of GS calculations By default,  abinit handles XCCC in real space,  here we use nc_xccc_gspace==1 to treat it in G-space using the same approach as the one used in PAW." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v7/Input/t10.in" target="_blank">t10.in</a>, <a class="wikilink abifile-wikilink" data-content="Relaxation of FCC Argon crystal with DFT-D3 dispersion correction Both the pair-wise term and the 3-body term are included In this test the following properties are thus tested: contribution to the energy and to the stress of DFT-D3 including the 3-body term" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v7/Input/t97.in" target="_blank">t97.in</a>, <a class="wikilink abifile-wikilink" data-content="Computation of the dynamical matrix at different q-vector including vdW-DFT-D3 corrections with Becke-Jonhson damping (DFT-D3(BJ)) Only the pair-wise is computed (3-body term not implemented) In this test the following properties are thus tested: contribution to the energy and to the stress of DFT-D3(BJ) as well as the contribution to the dynamical matrix" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v7/Input/t98.in" target="_blank">t98.in</a> &hellip;</li><li>vdwxc:  <a class="wikilink abifile-wikilink" data-content="Argon FCC. Test the vdW-DFT-D2 (Grimme) semi-empirical potential." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/vdwxc/Input/t10.in" target="_blank">t10.in</a> &hellip;</li></ul></details><p>Gives the maximal permitted scaling of the lattice parameters when the cell
shape and dimension is varied (see variable <a class="wikilink codevar-wikilink" href="../rlx#optcell">optcell</a>). It is used to define
the sphere of plane waves and FFT box coherent with the possible modifications
of the cell (<a class="wikilink codevar-wikilink" href="../rlx#ionmov">ionmov</a>==2 and [[optcell] /=0). For these definitions, it is
equivalent to changing <a class="wikilink codevar-wikilink" href="../basic#ecut">ecut</a> by multiplying it by <a class="wikilink codevar-wikilink" href="../rlx#dilatmx">dilatmx</a>  2  (the
result is an &ldquo;effective ecut&rdquo;, called internally &ldquo;ecut_eff&rdquo;, other uses of
<a class="wikilink codevar-wikilink" href="../basic#ecut">ecut</a> being not modified when <a class="wikilink codevar-wikilink" href="../rlx#dilatmx">dilatmx</a>&gt;1.0 .<br />
Using <a class="wikilink codevar-wikilink" href="../rlx#dilatmx">dilatmx</a>&lt;1.0 is equivalent to changing <a class="wikilink codevar-wikilink" href="../basic#ecut">ecut</a> in all its uses.
This is allowed, although its meaning is no longer related to a maximal
expected scaling.<br />
Setting <a class="wikilink codevar-wikilink" href="../rlx#dilatmx">dilatmx</a> to a large value leads to waste of CPU time and memory.
Supposing you think that the optimized <a class="wikilink codevar-wikilink" href="../basic#acell">acell</a> values might be 10% larger
than your input values, use simply <a class="wikilink codevar-wikilink" href="../rlx#dilatmx">dilatmx</a> 1.1 . This will already lead to
an increase of the number of planewaves by a factor (1.1)  3  =1.331 , and a
corresponding increase in CPU time and memory.<br />
It is possible to use <a class="wikilink codevar-wikilink" href="../rlx#dilatmx">dilatmx</a> when <a class="wikilink codevar-wikilink" href="../rlx#optcell">optcell</a> =0, but a value larger than
1.0 will be a waste.</p>
<h2 id="dtion"><strong>dtion</strong><a class="headerlink" href="#dtion" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> Delta Time for IONs<br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-content="How to perform a PIMD calculation" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../topics/PIMD">topic_PIMD</a>, <a class="wikilink topic-wikilink" data-content="How to perform a molecular dynamics calculation" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../topics/MolecularDynamics">topic_MolecularDynamics</a><br />
<em>Variable type:</em> real<br />
<em>Dimensions:</em> scalar<br />
<em>Default value:</em> 100  </p>
<details class="note"><summary>Test list (click to open). Moderately used, [18/920] in all abinit tests, [0/116] in abinit tutorials</summary><ul><li>etsf_io:  <a class="wikilink abifile-wikilink" data-content="Si2 molecule,  static,  spin-polarized. Same system as test 49 of v2,  except that nctime is non-zero. Test ionmov=6 (Verlet) as well as ionmov=7." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/etsf_io/Input/t21.in" target="_blank">t21.in</a></li><li>fast:  <a class="wikilink abifile-wikilink" data-content="21. Conduct same relaxation but use ionmov=1 (molecular dynamics)." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/fast/Input/t21.in" target="_blank">t21.in</a>, <a class="wikilink abifile-wikilink" data-content="move with natfix. Output also GEO files" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/fast/Input/t29.in" target="_blank">t29.in</a></li><li>mpiio:  <a class="wikilink abifile-wikilink" data-content="He FCC solid in conventional cell (4 atoms). Test the recursion algorithm (for high-temperature calculations). Only with 0 and 4 procs." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/mpiio/Input/t42.in" target="_blank">t42.in</a></li><li>paral:  <a class="wikilink abifile-wikilink" data-content="Gold,  8 atom supercell,  with PAW. Test lotf_nitex,  lotf_classic,  lotf_version,  lotf_nneigx. These variables,  concerning LOTF method,  are taken into account only if the enable_lotf=&amp;quot;yes&amp;quot; is used in configuration." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/paral/Input/t41.in" target="_blank">t41.in</a></li><li>v2:  <a class="wikilink abifile-wikilink" data-content="Si2 molecule,  static,  spin-polarized. Same system as test 17 of fast,  except lower ecut. Test ionmov=6 (Verlet) as well as ionmov=7. Cannote be executed in parallel: --- !ERROR message: | the number of bands in the spin up case must be equal to the number of bands in the spin down case. This is not the case for the k point number : 1 The number of bands spin up and down are : 5 3 Action : change nband,  or use the sequential version of ABINIT. src_file: chkinp.F90 src_line: 1181" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v2/Input/t49.in" target="_blank">t49.in</a>, <a class="wikilink abifile-wikilink" data-content="Ge liquid. Test of Nose and Langevin dynamics. 2 atoms in a cell. Allows 10 time steps." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v2/Input/t87.in" target="_blank">t87.in</a>, <a class="wikilink abifile-wikilink" data-content="Basic test of geometry optimization for water First computing the ground state of a close configuration Uses 5 different methods to find the relaxed positions" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v2/Input/t88.in" target="_blank">t88.in</a></li><li>v3:  <a class="wikilink abifile-wikilink" data-content="8 atoms of Si in an elongated box. 4 of them are fixed,  while atoms 5 and 6 and 7 and 8 are grouped by pair,  such as the sum of their relative separations along y is constrained to 8.0 . This is a crazy constraint,  just to show that it can be done !" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v3/Input/t40.in" target="_blank">t40.in</a></li><li>v4:  <a class="wikilink abifile-wikilink" data-content="Test of molecular dynamics of B atoms,  in the isokinetic ensemble,  based on the Thomas-Fermi functional (very fast,  even with 32 atoms)." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v4/Input/t97.in" target="_blank">t97.in</a></li><li>v5:  <a class="wikilink abifile-wikilink" data-content="This is a test,  in the norm conserving case,  of the Molecular Dynamics flag ionmov==13. With the values of optcell=0,  1,  2 (isothermal (0),  isothermal/isenthalpic with homogeneous cell deformations (1) or full cell deformations (2).) This is the reversible integrator of Martyna,  Tuckerman,  Tobias and Klein. (Mol. Phys.,  1996,  p 1117). XG20090912 : Why is the result of dataset 1 and dataset 4 equivalent,  while the bmass varies by a factor of 2 ?" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v5/Input/t01.in" target="_blank">t01.in</a>, <a class="wikilink abifile-wikilink" data-content="uses ionmov 14 : a 14-step time integration by Blanes and Moan." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v5/Input/t03.in" target="_blank">t03.in</a>, <a class="wikilink abifile-wikilink" data-content="Simple test for double grid (fine/coarse) system,  the cutting radius and the initial fermi energy in recursion method Corresponding variables: recgratio,  recrcut, recefermi respectively Corresponding code: 68_recursion Other test: paral/tT, v5/t76" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v5/Input/t75.in" target="_blank">t75.in</a></li><li>v7:  <a class="wikilink abifile-wikilink" data-content="Hydrogen molecule: Test the Path-Integral Molecular Dynamics (PIMD) implementation in (N, V, T) ensemble. THe two thermostats (Nose-Hoover chains and Langevin) are tested together with the three coordinate systems (primitive,  normal mode,  staging). NOTE: MPI run with 2 procs is ok but fldiff comparison fails due to an extra section with --- Pseudopotential description --" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v7/Input/t08.in" target="_blank">t08.in</a>, <a class="wikilink abifile-wikilink" data-content="Boron,  32 atoms - Thomas-Fermi method (without orbitals) Test the temperature-dependent XC functional (ixc=50) (Ichimaru et al Phys. Rep. 149,  91-205 (1987) Test the von Weizsacker gradient correction (tfkinfunc=11 or 12)" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v7/Input/t11.in" target="_blank">t11.in</a></li><li>v8:  <a class="wikilink abifile-wikilink" data-content="H2 molecule: PIMD simulation with Langevin thermostat. Test of restart with HIST.nc file. Test of linear constraint." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v8/Input/t05.in" target="_blank">t05.in</a>, <a class="wikilink abifile-wikilink" data-content="Simple test of velocity verlet integrator" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v8/Input/t12.in" target="_blank">t12.in</a>, <a class="wikilink abifile-wikilink" data-content="LiH molecule test the use of spatial chemical potential. The molecule starts aligned along the x direction, while the Li chemical potential pulls the Li nucleus toward -z,  and the H chemical potential pushes the H nucleus toward +z. See whether the optimization runs properly." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v8/Input/t31.in" target="_blank">t31.in</a></li></ul></details><p>Used for controlling ion time steps. If <a class="wikilink codevar-wikilink" href="../rlx#ionmov">ionmov</a> is set to 1, 6 or 7, then
molecular dynamics is  used to update atomic positions in response to forces.
The parameter <a class="wikilink codevar-wikilink" href="../rlx#dtion">dtion</a> is a time step in atomic units of time. (One atomic
time unit is 2.418884e-17 seconds, which is the value of Planck&rsquo;s constant in
hartree*sec.) In this case the atomic masses, in amu (given in array &rdquo; <a class="wikilink codevar-wikilink" href="../rlx#amu">amu</a>
&ldquo;), are used in Newton&rsquo;s equation and the viscosity (for <a class="wikilink codevar-wikilink" href="../rlx#ionmov">ionmov</a> =1) and
number of time steps are provided to the code using input variables &ldquo;<a class="wikilink codevar-wikilink" href="../rlx#vis">vis</a>&rdquo;
and &ldquo;<a class="wikilink codevar-wikilink" href="../rlx#ntime">ntime</a>&rdquo;. The code actually converts from masses in amu to masses in
atomic units (in units of electron masses) but the user enters masses in
<a class="wikilink codevar-wikilink" href="../rlx#amu">amu</a> . (The conversion from amu to atomic units (electron masses) is
1822.88851 electron masses/amu.)<br />
A typical good value for <a class="wikilink codevar-wikilink" href="../rlx#dtion">dtion</a> is about 100. The user must try several
values for <a class="wikilink codevar-wikilink" href="../rlx#dtion">dtion</a> in order to establish the stable and efficient choice for
the accompanying amu, atom types and positions, and <a class="wikilink codevar-wikilink" href="../rlx#vis">vis</a> (viscosity).<br />
For quenched dynamics (<a class="wikilink codevar-wikilink" href="../rlx#ionmov">ionmov</a>=7), a larger time step might be taken, for
example 200.<br />
No meaning for RF calculations.</p>
<h2 id="dynimage"><strong>dynimage</strong><a class="headerlink" href="#dynimage" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> DYNamics of the IMAGE<br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-content="How to perform a PIMD calculation" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../topics/PIMD">topic_PIMD</a>, <a class="wikilink topic-wikilink" data-content="How to calculate transition paths" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../topics/TransPath">topic_TransPath</a><br />
<em>Variable type:</em> integer<br />
<em>Dimensions:</em> (<a class="wikilink codevar-wikilink" href="../rlx#nimage">nimage</a>)<br />
<em>Default value:</em> <em>1<br />
*Comment:</em> if <a class="wikilink codevar-wikilink" href="../rlx#imgmov">imgmov</a> in [2,5] (String Method, NEB), <b>dynimage(1)</b>=0 and <b>dynimage(<a class="wikilink codevar-wikilink" href="../rlx#nimage">nimage</a>)</b>=0.  </p>
<details class="note"><summary>Test list (click to open). Rarely used, [9/920] in all abinit tests, [0/116] in abinit tutorials</summary><ul><li>paral:  <a class="wikilink abifile-wikilink" data-content="Test the string method within parallelization over images Inspired by test v6#22. Hydrogen diatomic molecule in a cell,  close to BCC 7 images,  exploring the transition path. Three datasets,  testing each value of prtvolimg. Processors distribution automatically determined:" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/paral/Input/t08.in" target="_blank">t08.in</a>, <a class="wikilink abifile-wikilink" data-content="Test the string method within parallelization over images Inspired by test v6#22. Hydrogen diatomic molecule in a cell,  close to BCC 7 images,  exploring the transition path. Three datasets,  testing each value of prtvolimg. Processors distribution automatically determined:" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/paral/Input/t08.in" target="_blank">t08.in</a>, <a class="wikilink abifile-wikilink" data-content="Test the string method within parallelization over images Inspired by test v6#22. Hydrogen diatomic molecule in a cell,  close to BCC 7 images,  exploring the transition path. Three datasets,  testing each value of prtvolimg. Processors distribution automatically determined:" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/paral/Input/t08.in" target="_blank">t08.in</a>, <a class="wikilink abifile-wikilink" data-content="Test the string method within parallelization over images Inspired by test v6#22. Hydrogen diatomic molecule in a cell,  close to BCC 7 images,  exploring the transition path. Three datasets,  testing each value of prtvolimg. Processors distribution automatically determined:" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/paral/Input/t08.in" target="_blank">t08.in</a></li><li>v6:  <a class="wikilink abifile-wikilink" data-content="Hydrogen diatomic molecule in a cell,  close to BCC. One hydrogen atom is at the origin. The other lies initially along the 1 1 1 direction,  at a distance close to the one of the free dimer distance. The final position is the mirror with respect to the plane perpendicular to x,  passing at 1/2 0 0 . The cell parameter is less than twice the H2 interatomic distance,  so that the transition path come close to the x axis,  although it never reaches it (the closest being in the mirror plane). Test the images : 6 images,  exploring the transition path by a simple algorithm. MG: Does it work in parallel. I tried 2 CPUs and the run get stuck!" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v6/Input/t22.in" target="_blank">t22.in</a>, <a class="wikilink abifile-wikilink" data-content="Hydrogen diatomic molecule in a cell,  close to BCC Test the string method : 6 images,  exploring the transition path. Similar to test 22 (but with string)." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v6/Input/t24.in" target="_blank">t24.in</a>, <a class="wikilink abifile-wikilink" data-content="Hydrogen diatomic molecule in a cell,  close to BCC Test the string method : 6 images,  exploring the transition path but keeping X coordinated fixed. Similar to test 24" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v6/Input/t25.in" target="_blank">t25.in</a></li><li>v7:  <a class="wikilink abifile-wikilink" data-content="Hydrogen molecule: Test the Path-Integral Molecular Dynamics (PIMD) implementation in (N, V, T) ensemble. THe two thermostats (Nose-Hoover chains and Langevin) are tested together with the three coordinate systems (primitive,  normal mode,  staging). NOTE: MPI run with 2 procs is ok but fldiff comparison fails due to an extra section with --- Pseudopotential description --" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v7/Input/t08.in" target="_blank">t08.in</a></li><li>v8:  <a class="wikilink abifile-wikilink" data-content="H2 molecule: PIMD simulation with Langevin thermostat. Test of restart with HIST.nc file. Test of linear constraint." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v8/Input/t05.in" target="_blank">t05.in</a></li></ul></details><p>This input variable is relevant when sets of images are activated (see
<a class="wikilink codevar-wikilink" href="../rlx#imgmov">imgmov</a>). Not all images might be required to evolve from one time step to
the other. Indeed, in the String Method or the Nudged Elastic Band, one might
impose that the extremal configurations of the string are fixed. In case the
<a class="wikilink codevar-wikilink" href="../rlx#dynimage">dynimage</a>(iimage)=0, the image with index &ldquo;iimage&rdquo; will be consider as
fixed. Thus, there is no need to compute forces and stresses for this image at
each time step. The purpose of defining extremal images is to make the
input/output easier.  </p>
<p>In order to save CPU time, the computation of properties of static images
(<a class="wikilink codevar-wikilink" href="../rlx#dynimage">dynimage</a>(iimage)=0) can be avoided: see <a class="wikilink codevar-wikilink" href="../rlx#istatimg">istatimg</a> keyword.</p>
<h2 id="ecutsm"><strong>ecutsm</strong><a class="headerlink" href="#ecutsm" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> Energy CUToff SMearing<br />
<em>Characteristics:</em> <a class="wikilink" href="../../user-guide/abinit#32-more-about-abinit-input-variables">ENERGY</a><br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-content="How to perform numerically precise calculations with planewaves or projector-
augmented waves and pseudopotentials" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../topics/Planewaves">topic_Planewaves</a><br />
<em>Variable type:</em> real<br />
<em>Dimensions:</em> scalar<br />
<em>Default value:</em> 0.0  </p>
<details class="note"><summary>Test list (click to open). Moderately used, [115/920] in all abinit tests, [29/116] in abinit tutorials</summary><ul><li>gpu:  <a class="wikilink abifile-wikilink" data-content="Silicon - diamond structure - PAW - volume relaxation Test the use of GPU within Projector Augmented-Wave formalism. Test structure relaxation (multiple calls to gstate routine)." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/gpu/Input/t04.in" target="_blank">t04.in</a> &hellip;</li><li>mpiio:  <a class="wikilink abifile-wikilink" data-content="C-diamond,  Bulk,  2 atoms,  paral_kgb,  with PAW. Test of ground state with different occs(7 and 0), and also ionmov 2. Only with 0 and 4 procs." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/mpiio/Input/t22.in" target="_blank">t22.in</a>, <a class="wikilink abifile-wikilink" data-content="C-diamond,  Bulk,  2 atoms,  paral_kgb,  with PAW. Test the triple parallelisation. We cannot check the distribution npband*npfft*npkpt=2*2*2=8 processors, this number being not allowed in the test procedure. Here,  we only test the parallelisation over FFT and kpoints: npband*npfft*npkpt=1*2*2=4 processors. In test tY.in we check other distributions for guarantee." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/mpiio/Input/t24.in" target="_blank">t24.in</a> &hellip;</li><li>paral:  <a class="wikilink abifile-wikilink" data-content="Test the string method within parallelization over images Inspired by test v6#22. Hydrogen diatomic molecule in a cell,  close to BCC 7 images,  exploring the transition path. Three datasets,  testing each value of prtvolimg. Processors distribution automatically determined:" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/paral/Input/t08.in" target="_blank">t08.in</a>, <a class="wikilink abifile-wikilink" data-content="Test the string method within parallelization over images Inspired by test v6#22. Hydrogen diatomic molecule in a cell,  close to BCC 7 images,  exploring the transition path. Three datasets,  testing each value of prtvolimg. Processors distribution automatically determined:" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/paral/Input/t08.in" target="_blank">t08.in</a>, <a class="wikilink abifile-wikilink" data-content="Test the string method within parallelization over images Inspired by test v6#22. Hydrogen diatomic molecule in a cell,  close to BCC 7 images,  exploring the transition path. Three datasets,  testing each value of prtvolimg. Processors distribution automatically determined:" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/paral/Input/t08.in" target="_blank">t08.in</a> &hellip;</li><li>seq:  <a class="wikilink abifile-wikilink" data-content="PbTiO3 in the tetragonal geometry. Test berryopt=14 (finite reduced electric field calculation, relaxing cell parameters,  e.g. optcell=2) (XG120616 : tranferred to tests/seq)" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/seq/Input/tsv6_121.in" target="_blank">tsv6_121.in</a>, <a class="wikilink abifile-wikilink" data-content="PbTiO3 in the tetragonal geometry. Test berryopt=16 (finite reduced electric displacement field calculation, relaxing cell parameters,  e.g. optcell=2) (XG120616 : tranferred to tests/seq)" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/seq/Input/tsv6_122.in" target="_blank">tsv6_122.in</a>, <a class="wikilink abifile-wikilink" data-content="PbTiO3 in the tetragonal geometry. Test berryopt=4 (finite electric field calculation,  relaxing cell parameters,  e.g. optcell=2) (XG120616 : tranferred to tests/seq)" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/seq/Input/tsv6_123.in" target="_blank">tsv6_123.in</a> &hellip;</li><li>tutorespfn:  <a class="wikilink abifile-wikilink" data-content="AlAs in hypothetical wurzite (hexagonal) structure Structural optimization run" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/tutorespfn/Input/telast_1.in" target="_blank">telast_1.in</a>, <a class="wikilink abifile-wikilink" data-content="AlAs in hypothetical wurzite (hexagonal) structure Response function calculation for: * rigid-atom elastic tensor * rigid-atom piezoelectric tensor * interatomic force constants at gamma * Born effective charges" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/tutorespfn/Input/telast_2.in" target="_blank">telast_2.in</a>, <a class="wikilink abifile-wikilink" data-content="AlAs in hypothetical wurzite (hexagonal) structure Finite-difference calculation for c-axis strain increment +/- 0.0001" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/tutorespfn/Input/telast_4.in" target="_blank">telast_4.in</a> &hellip;</li><li>tutorial:  <a class="wikilink abifile-wikilink" data-content="Crystalline silicon : computation of the optimal lattice parameter Convergence with respect to the number of k points." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/tutorial/Input/tbase3_4.in" target="_blank">tbase3_4.in</a>, <a class="wikilink abifile-wikilink" data-content="Crystalline aluminum : optimization of the lattice parameter at fixed number of k points and broadening." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/tutorial/Input/tbase4_1.in" target="_blank">tbase4_1.in</a>, <a class="wikilink abifile-wikilink" data-content="Crystalline aluminum : optimization of the lattice parameter Convergence with respect to k points." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/tutorial/Input/tbase4_2.in" target="_blank">tbase4_2.in</a> &hellip;</li><li>v1:  <a class="wikilink abifile-wikilink" data-content="2 special k point Si in 2-atom diamond unit cell. Test ecutsm. Determine a smooth etot vs acel curve (one can try with ecutsm=0.0d0 to see the difference !) Stress is correct. For example, at acell=10.10 (volume=2.5757525E+02),  one obtains etotal=-8.7850566628 at acell=10.12 (volume=2.5910843E+02),  one obtains etotal=-8.7851433215, giving a finite-difference estimate of the stress ( d(etotal)/d(volume) ) at 10.11 to be -5.6522d-5,  while the code gives -5.6582015717E-05" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v1/Input/t76.in" target="_blank">t76.in</a>, <a class="wikilink abifile-wikilink" data-content="2 special k point Si in 2-atom diamond unit cell. Optimization of unit cell volume (optcell=1). Use 2 datasets,  and the variable getcell." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v1/Input/t78.in" target="_blank">t78.in</a>, <a class="wikilink abifile-wikilink" data-content="H2 molecule in a cell of variable size along the H2 chain =&amp;gt; H polymer. Optimization of cell size and atomic distances (optcell=4)." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v1/Input/t79.in" target="_blank">t79.in</a> &hellip;</li><li>v2:  <a class="wikilink abifile-wikilink" data-content="Arsenic in rhombohedral structure. Optimization of unit cell size and shape. With 4x4x4 k point grid,  ecut 3.0,  occopt 4 and tsmear 0.06d0, one get acell 3*7.325  angdeg 3*57.36 and xred 0.2277 These input parameters are too small,  but still give a realistic geometry,  since with the much better parameters 12x12x12 k point grid,  ecut 12.0,  occopt 4 and tsmear 0.04d0, one gets acell 3*7.633  angdeg 3*54.95 and xred 0.2296,  while the experimental values are acell 3*7.751   angdeg 3*54.554 and xred 0.2276" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v2/Input/t44.in" target="_blank">t44.in</a>, <a class="wikilink abifile-wikilink" data-content="Li,  simple cubic,  to test accurately RF in the metallic case. With non-linear XC core correction and non-zero ecutsm and dilatmx. Similar to test 92,  except ecutsm and dilatmx,  and no q(1/4 0 0). The total energy of the 1-atom cell is -.29337493523588 Ha The 2DTE with respect to a q(0 0 0) phonon is 5.5619d-8 . The 2DTE with respect to a transverse q(1/2 0 0) phonon is -0.12068188927 Ha." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v2/Input/t96.in" target="_blank">t96.in</a>, <a class="wikilink abifile-wikilink" data-content="Li,  simple cubic,  frozen-phonon calculations corresponding to test 96 (with non-zero ecutsm and dilatmx). The total energy of the double cell is -0.58674987047187 Ha, in excellent agreement with the result of test 96. The frozen-phonon (from energy) calculation of the 2DTE for the q(1/2 0 0) case gives -0.1206807 Ha . The frozen-phonon (from forces) calculation of the 2DTE for the q(1/2 0 0) case gives -0.1206794 Ha . Both are in rather good agreement with the result of test 96." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v2/Input/t97.in" target="_blank">t97.in</a> &hellip;</li><li>v3:  <a class="wikilink abifile-wikilink" data-content="Magnesium HCP. Optimize the cell volume and shape using different constraints. Checks whether the stresses are symmetric,  and whether the optimisation algorithm does not break the symmetry." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v3/Input/t42.in" target="_blank">t42.in</a>, <a class="wikilink abifile-wikilink" data-content="H2 molecule : examine the rotational freedom. The present test produces the following vibrational frequencies (with degeneracies indicated): 56.89 i cm-1 (2) 0.41   cm-1 (2) 1.05   cm-1 3800    cm-1 The large frequency corresponds to the stretching mode,  and has the right order of magnitude. The frequencies close to 1 cm-1 corresponds to translation modes,  and are small enough for usual applications. The  56.89 i cm-1 mode corresponds to rotation of the H2 molecule. The magnitude of this frequency might seem quite large. Here are the results of tests made to understand this phenomenon. First,  note that ecut 12  acell 3*14 Increasing the value of ecut to 25 decreases the magnitude of the frequency to 36.8 cm-1 . However,  in order to continue to make it smaller, the cell size must be increased,  and an oscillatory behaviour is observed : 3*16   45.6 i cm-1 3*18   22.7   cm-1 3*20   19.1 i cm-1 3*22   15.7 i cm-1 3*24   13.7   cm-1 Many other tests have been set up. In particular, it was observed that the frequency of the oscillatory behaviour changes with the ecut,  and also that using the Gamma point,  instead of the 1/4 1/4 1/4 k point (used in this test) degrades the convergence. The overall picture is as follows. There are different reasons for the translation and rotation modes to acquire a non-zero frequency when plane waves and supercells are used. Still,  as concerns translations,  only the existence of a discretization of the XC grid is important. For rotations,  supercell effects are also present : - alignement of dipole or quadrupoles - interaction between tails of wavefunctions,  accross cells Since the convergence in supercell size is oscillatory,  we infer that the breaking of the rotational invariance is mostly due to interaction between wavefunction tails. This will be checked by confining the system in a spherical well,  in a forthcoming test." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v3/Input/t80.in" target="_blank">t80.in</a> &hellip;</li><li>v4:  <a class="wikilink abifile-wikilink" data-content="2. Generate first-order responses for GaAs (zinc-blende) (from test v2 #98). Compute responses at Gamma: 1) Ground state calculation,  with a k-point sampling in the IBZ only; 2) Computation of the ddk response; 3) Computation of the dynamical matrix at Gamma,  as well as the Born effective charges,  and the dielectric tensor; Perform five datasets: * First dataset is the ground state calculation, * Second dataset compute ddk perturbation with the use of Legendre Polynomials for the nl operator, * Third dataset compute phonon and elec. field perturbation with the use of Legendre Polynomials for the nl operator, * Fourth dataset compute ddk perturbation with the use of spherical harmonics for the nl operator, * Fifth dataset compute phonon and elec. field perturbation with the use of spherical harmonics for the nl operator. Datasets 2 and 3 must give the same results as 4 and 5." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v4/Input/t02.in" target="_blank">t02.in</a>, <a class="wikilink abifile-wikilink" data-content="Test of the pspcod=8 input format (contributed by D.R. Hamann) This format is primarily intended to offer a flexible platform for those who wish to experiment with their own features in norm-conserving pseudopotentials.  A full description and discussion is given in Infos/Psp_infos/psp8.info.  This test case,  CaO,  is an example of self-interaction-corrected pseudopotentials in the style of D. Vogel,  P. Krueger,  and J. Pollmann,  Phys. Rev. B 52, 14 316 (1995).  The bandgap (direct) in the test example,  11.2eV, is in excellent agreement with experiment,  but is rather strongly dependent on the radius at which the self-interaction-correction of Ca is (smoothly) cut off.  The sic pseudopotentials used in this test should not be considered anything but early-stage experiments and should not be used for any other purpose." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v4/Input/t20.in" target="_blank">t20.in</a>, <a class="wikilink abifile-wikilink" data-content="Si/Ge heterojunction : 4 layer each. Output the density and potential. Prepare the macroaverage analysis,  see next tests." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v4/Input/t42.in" target="_blank">t42.in</a> &hellip;</li><li>v5:  <a class="wikilink abifile-wikilink" data-content="This is a test,  in the norm conserving case,  of the Molecular Dynamics flag ionmov==13. With the values of optcell=0,  1,  2 (isothermal (0),  isothermal/isenthalpic with homogeneous cell deformations (1) or full cell deformations (2).) This is the reversible integrator of Martyna,  Tuckerman,  Tobias and Klein. (Mol. Phys.,  1996,  p 1117). XG20090912 : Why is the result of dataset 1 and dataset 4 equivalent,  while the bmass varies by a factor of 2 ?" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v5/Input/t01.in" target="_blank">t01.in</a>, <a class="wikilink abifile-wikilink" data-content="uses ionmov 14 : a 14-step time integration by Blanes and Moan." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v5/Input/t03.in" target="_blank">t03.in</a>, <a class="wikilink abifile-wikilink" data-content="Crystalline Silicon Test the computation of second order eigenvalues for two qpoints,  (0 0 0) and (1/2 0 0),  contained in the EIGR2D files. Parameters are far from convergence,  nband=5 and nkpt=16." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v5/Input/t26.in" target="_blank">t26.in</a> &hellip;</li><li>v6:  <a class="wikilink abifile-wikilink" data-content="Hydrogen diatomic molecule in a cell,  close to BCC. One hydrogen atom is at the origin. The other lies initially along the 1 1 1 direction,  at a distance close to the one of the free dimer distance. The final position is the mirror with respect to the plane perpendicular to x,  passing at 1/2 0 0 . The cell parameter is less than twice the H2 interatomic distance,  so that the transition path come close to the x axis,  although it never reaches it (the closest being in the mirror plane). Test the images : 6 images,  exploring the transition path by a simple algorithm. MG: Does it work in parallel. I tried 2 CPUs and the run get stuck!" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v6/Input/t22.in" target="_blank">t22.in</a>, <a class="wikilink abifile-wikilink" data-content="Hydrogen diatomic molecule in a cell,  close to BCC Test the string method : 6 images,  exploring the transition path. Similar to test 22 (but with string)." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v6/Input/t24.in" target="_blank">t24.in</a>, <a class="wikilink abifile-wikilink" data-content="Hydrogen diatomic molecule in a cell,  close to BCC Test the string method : 6 images,  exploring the transition path but keeping X coordinated fixed. Similar to test 24" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v6/Input/t25.in" target="_blank">t25.in</a> &hellip;</li><li>v67mbpt:  <a class="wikilink abifile-wikilink" data-content="Test the cd_custom_imfrq,  cd_imfrqs and cd_full_grid options" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v67mbpt/Input/t21.in" target="_blank">t21.in</a>, <a class="wikilink abifile-wikilink" data-content="Test new integration method and grid options for the integral along the imaginary axis in contour deformation calculations." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v67mbpt/Input/t22.in" target="_blank">t22.in</a> &hellip;</li><li>v7:  <a class="wikilink abifile-wikilink" data-content="Nickel ferromagnetic fcc structure with an ONCVPSP pseudo containing valence density. We test several improvements that are available when the NC file contains the PS valence charge. More specifically,  we test the initialization of the density from atomic quantities (initro.F90), the correction to the forces due to the residuals and the extrapolation of the density for structural relaxations (well,  xred is fixed by symmetry). We also test the treatment of the core charge in G-space in the case of GS calculations By default,  abinit handles XCCC in real space,  here we use nc_xccc_gspace==1 to treat it in G-space using the same approach as the one used in PAW." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v7/Input/t10.in" target="_blank">t10.in</a>, <a class="wikilink abifile-wikilink" data-content="Test presence of nuclear magnetic dipole moments on atomic sites" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v7/Input/t32.in" target="_blank">t32.in</a>, <a class="wikilink abifile-wikilink" data-content="Diamond. Test temperature-dependent of the electronic structure,  with reduction of the number of q points to be computed,  thanks to thmflag=7 . The temperature-dependent shifts must be averaged over degenerate states. This temperature-dependent calculation make use of a k-point interpolation to speed up the convergence study with respect to the q-points integration. Two q-points are computed." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v7/Input/t50.in" target="_blank">t50.in</a> &hellip;</li><li>v8:  <a class="wikilink abifile-wikilink" data-content="Geometry optimization with LBFGS" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v8/Input/t02.in" target="_blank">t02.in</a>, <a class="wikilink abifile-wikilink" data-content="Computation of elastic tensor and internal strain in DFPT+PAW Ground state is computed. DFPT is computed" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v8/Input/t07.in" target="_blank">t07.in</a>, <a class="wikilink abifile-wikilink" data-content="WARNING: This code is still under development. Use at your own risk! Preparatory run for the calculation of the Fan-Migdal + Debye-Waller self-energy C in diamond structure. Very rough 2x2x2 q-point grid (3 qpoints);  low ecut." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v8/Input/t41.in" target="_blank">t41.in</a> &hellip;</li><li>vdwxc:  <a class="wikilink abifile-wikilink" data-content="Argon FCC. Test the vdW-DFT-D2 (Grimme) semi-empirical potential." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/vdwxc/Input/t10.in" target="_blank">t10.in</a> &hellip;</li><li>wannier90:  <a class="wikilink abifile-wikilink" data-content="Argon FCC. Test the approach of Silvestrelli to access van der Waals interactions." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/wannier90/Input/t11.in" target="_blank">t11.in</a>, <a class="wikilink abifile-wikilink" data-content="Argon FCC Test the approach of Ambrosetti and Silvestrelli to assess van der Waals interactions." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/wannier90/Input/t12.in" target="_blank">t12.in</a>, <a class="wikilink abifile-wikilink" data-content="Argon FCC Test the approach of Silvestrelli to assess van der Waals interactions,  combining the QHO model with Wannier functions." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/wannier90/Input/t13.in" target="_blank">t13.in</a> &hellip;</li></ul></details><p>This input variable is important when performing relaxation of unit cell size
and shape (non-zero <a class="wikilink codevar-wikilink" href="../rlx#optcell">optcell</a>). Using a non-zero <a class="wikilink codevar-wikilink" href="../rlx#ecutsm">ecutsm</a>, the total
energy curves as a function of <a class="wikilink codevar-wikilink" href="../basic#ecut">ecut</a>, or <a class="wikilink codevar-wikilink" href="../basic#acell">acell</a>, can be smoothed,
keeping consistency with the stress (and automatically including the Pulay
stress). The recommended value is 0.5 Ha. Actually, when <a class="wikilink codevar-wikilink" href="../rlx#optcell">optcell</a>/=0,
ABINIT requires <a class="wikilink codevar-wikilink" href="../rlx#ecutsm">ecutsm</a> to be larger than zero. If you want to optimize
cell shape and size without smoothing the total energy curve (a dangerous
thing to do), use a very small <a class="wikilink codevar-wikilink" href="../rlx#ecutsm">ecutsm</a>, on the order of one microHartree.</p>
<p>Technical information :<br />
See Bernasconi et al, J. Phys. Chem. Solids 56, 501 (1995) for a related
method.<br />
<a class="wikilink codevar-wikilink" href="../rlx#ecutsm">ecutsm</a> allows to define an effective kinetic energy for plane waves, close
to, but lower than the maximal kinetic energy <a class="wikilink codevar-wikilink" href="../basic#ecut">ecut</a>. For kinetic energies
less than <a class="wikilink codevar-wikilink" href="../basic#ecut">ecut</a>-<a class="wikilink codevar-wikilink" href="../rlx#ecutsm">ecutsm</a>, nothing is modified, while between
<a class="wikilink codevar-wikilink" href="../basic#ecut">ecut</a>-<a class="wikilink codevar-wikilink" href="../rlx#ecutsm">ecutsm</a> and <a class="wikilink codevar-wikilink" href="../basic#ecut">ecut</a> , the kinetic energy is multiplied by:<br />
1.0 / ( x  2  (3+x-6x  2  +3x  3  ))<br />
where x = (<a class="wikilink codevar-wikilink" href="../basic#ecut">ecut</a> - kinetic_energy)/<a class="wikilink codevar-wikilink" href="../rlx#ecutsm">ecutsm</a><br />
Note that x  2  ( 3+x-6x  2  +3x  3  ) is 0 at x=0, with vanishing derivative,
and that at x=1 , it is 1, with also vanishing derivative.<br />
If <a class="wikilink codevar-wikilink" href="../rlx#ecutsm">ecutsm</a> is zero, the unmodified kinetic energy is used.<br />
<a class="wikilink codevar-wikilink" href="../rlx#ecutsm">ecutsm</a> can be specified in Ha (the default), Ry, eV or Kelvin, since
<a class="wikilink codevar-wikilink" href="../rlx#ecutsm">ecutsm</a> has the &lsquo;<a class="wikilink" href="../../user-guide/abinit#32-more-about-abinit-input-variables">ENERGY</a>&lsquo; characteristics. (1 Ha=27.2113845 eV).<br />
A few test for Silicon (diamond structure, 2 k-points) have shown 0.5 Ha to be
largely enough for <a class="wikilink codevar-wikilink" href="../basic#ecut">ecut</a> between 2Ha and 6Ha, to get smooth curves. It is
likely that this value is OK as soon as <a class="wikilink codevar-wikilink" href="../basic#ecut">ecut</a> is larger than 4Ha.</p>
<h2 id="friction"><strong>friction</strong><a class="headerlink" href="#friction" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> internal FRICTION coefficient<br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-content="How to perform a molecular dynamics calculation" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../topics/MolecularDynamics">topic_MolecularDynamics</a><br />
<em>Variable type:</em> real<br />
<em>Dimensions:</em> scalar<br />
<em>Default value:</em> 0.001  </p>
<details class="note"><summary>Test list (click to open). Rarely used, [2/920] in all abinit tests, [0/116] in abinit tutorials</summary><ul><li>v2:  <a class="wikilink abifile-wikilink" data-content="Ge liquid. Test of Nose and Langevin dynamics. 2 atoms in a cell. Allows 10 time steps." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v2/Input/t87.in" target="_blank">t87.in</a>, <a class="wikilink abifile-wikilink" data-content="Basic test of geometry optimization for water First computing the ground state of a close configuration Uses 5 different methods to find the relaxed positions" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v2/Input/t88.in" target="_blank">t88.in</a></li></ul></details><p>Gives the internal friction coefficient (atomic units) for Langevin dynamics
(when <a class="wikilink codevar-wikilink" href="../rlx#ionmov">ionmov</a>=9): fixed temperature simulations with random forces.</p>
<p>The equation of motion is :<br />
M  I  d  2  R  I  /dt  2  = F  I  - <a class="wikilink codevar-wikilink" href="../rlx#friction">friction</a> M  I  dR  I  /dt - F_random
I<br />
where F_random  I  is a Gaussian random force with average zero, and variance
2 <a class="wikilink codevar-wikilink" href="../rlx#friction">friction</a> M  I  kT.<br />
The atomic unit of friction is hartrees*electronic mass*(atomic time
units)/Bohr  2  . See J. Chelikowsky, J. Phys. D : Appl Phys. 33(2000)R33.</p>
<h2 id="fxcartfactor"><strong>fxcartfactor</strong><a class="headerlink" href="#fxcartfactor" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> Forces to (X) CARTesian coordinates FACTOR<br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-content="How to calculate transition paths" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../topics/TransPath">topic_TransPath</a>, <a class="wikilink topic-wikilink" data-content="How to perform a geometry optimization" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../topics/GeoOpt">topic_GeoOpt</a><br />
<em>Variable type:</em> real<br />
<em>Dimensions:</em> scalar<br />
<em>Default value:</em> 1 (Bohr^2)/Hartree  </p>
<details class="note"><summary>Test list (click to open). Rarely used, [7/920] in all abinit tests, [0/116] in abinit tutorials</summary><ul><li>paral:  <a class="wikilink abifile-wikilink" data-content="Test the string method within parallelization over images Inspired by test v6#22. Hydrogen diatomic molecule in a cell,  close to BCC 7 images,  exploring the transition path. Three datasets,  testing each value of prtvolimg. Processors distribution automatically determined:" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/paral/Input/t08.in" target="_blank">t08.in</a>, <a class="wikilink abifile-wikilink" data-content="Test the string method within parallelization over images Inspired by test v6#22. Hydrogen diatomic molecule in a cell,  close to BCC 7 images,  exploring the transition path. Three datasets,  testing each value of prtvolimg. Processors distribution automatically determined:" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/paral/Input/t08.in" target="_blank">t08.in</a>, <a class="wikilink abifile-wikilink" data-content="Test the string method within parallelization over images Inspired by test v6#22. Hydrogen diatomic molecule in a cell,  close to BCC 7 images,  exploring the transition path. Three datasets,  testing each value of prtvolimg. Processors distribution automatically determined:" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/paral/Input/t08.in" target="_blank">t08.in</a>, <a class="wikilink abifile-wikilink" data-content="Test the string method within parallelization over images Inspired by test v6#22. Hydrogen diatomic molecule in a cell,  close to BCC 7 images,  exploring the transition path. Three datasets,  testing each value of prtvolimg. Processors distribution automatically determined:" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/paral/Input/t08.in" target="_blank">t08.in</a></li><li>v6:  <a class="wikilink abifile-wikilink" data-content="Bismuth,  rhombohedral,  two atoms per unit cell Scalar relativistic. Relax four different images,  using the steepest descent algorithm." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v6/Input/t21.in" target="_blank">t21.in</a>, <a class="wikilink abifile-wikilink" data-content="Hydrogen diatomic molecule in a cell,  close to BCC Test the string method : 6 images,  exploring the transition path. Similar to test 22 (but with string)." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v6/Input/t24.in" target="_blank">t24.in</a>, <a class="wikilink abifile-wikilink" data-content="Hydrogen diatomic molecule in a cell,  close to BCC Test the string method : 6 images,  exploring the transition path but keeping X coordinated fixed. Similar to test 24" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v6/Input/t25.in" target="_blank">t25.in</a></li></ul></details><p>The forces multiplied by <a class="wikilink codevar-wikilink" href="../rlx#fxcartfactor">fxcartfactor</a> will be treated like difference in
cartesian coordinates in the process of optimization. This is a simple
preconditioner.<br />
TO BE UPDATED See (<a class="wikilink codevar-wikilink" href="../rlx#ionmov">ionmov</a>=2, non-zero <a class="wikilink codevar-wikilink" href="../rlx#optcell">optcell</a>). For example, the
stopping criterion defined by <a class="wikilink codevar-wikilink" href="../rlx#tolmxf">tolmxf</a> relates to these scaled stresses.</p>
<h2 id="ga_algor"><strong>ga_algor</strong><a class="headerlink" href="#ga_algor" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> Genetic Algorithm - ALGOrithm selection<br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-content="How to perform a geometry optimization" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../topics/GeoOpt">topic_GeoOpt</a><br />
<em>Variable type:</em> integer<br />
<em>Dimensions:</em> scalar<br />
<em>Default value:</em> 1  </p>
<details class="note"><summary>Test list (click to open). Rarely used, [0/920] in all abinit tests, [0/116] in abinit tutorials</summary></details><p>Choosing method to make the structure selection. Only the enthalpy is used now
but we plan to include, energy, electronic band gap and alchemical potentials.
Right now only value of 1 (enthalpy) works.</p>
<h2 id="ga_fitness"><strong>ga_fitness</strong><a class="headerlink" href="#ga_fitness" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> Genetic Algorithm FITNESS function selection<br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-content="How to perform a geometry optimization" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../topics/GeoOpt">topic_GeoOpt</a><br />
<em>Variable type:</em> integer<br />
<em>Dimensions:</em> scalar<br />
<em>Default value:</em> 1  </p>
<details class="note"><summary>Test list (click to open). Rarely used, [1/920] in all abinit tests, [0/116] in abinit tutorials</summary><ul><li>v6:  <a class="wikilink abifile-wikilink" data-content="Genetic algorithm structure random search. Hydrogen has been used as an example. No physical meaning. Number of images is only 10 but experience shows that at least 20 are fine." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v6/Input/t27.in" target="_blank">t27.in</a></li></ul></details><p>Different methodologies to perform the roulette-wheel selection of parents.
Even though, the objective function is the crystalline enthalpy (H_i), the
weight of the population elements to be chosen from in a roulette-wheel
selection can be given through different functions. We consider the following
cases.<br />
1. F = H_i / Sum H_i<br />
2. F = exp(-(H_i-H_min)) / Sum exp(-(H_i-H_min))<br />
3. F = (1/n_i) / Sum (1/n_i). Where n_i is the position in the ordered list
of enthalpies</p>
<h2 id="ga_n_rules"><strong>ga_n_rules</strong><a class="headerlink" href="#ga_n_rules" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> Genetic Algorithm Number of RULES<br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-content="How to perform a geometry optimization" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../topics/GeoOpt">topic_GeoOpt</a><br />
<em>Variable type:</em> integer<br />
<em>Dimensions:</em> scalar<br />
<em>Default value:</em> 1  </p>
<details class="note"><summary>Test list (click to open). Rarely used, [1/920] in all abinit tests, [0/116] in abinit tutorials</summary><ul><li>v6:  <a class="wikilink abifile-wikilink" data-content="Genetic algorithm structure random search. Hydrogen has been used as an example. No physical meaning. Number of images is only 10 but experience shows that at least 20 are fine." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v6/Input/t27.in" target="_blank">t27.in</a></li></ul></details><p>Different genetic rules have been implemented and the user has the change to
choose between any of them. Right now we have 4 rules. See <a class="wikilink codevar-wikilink" href="../rlx#ga_rules">ga_rules</a></p>
<h2 id="ga_opt_percent"><strong>ga_opt_percent</strong><a class="headerlink" href="#ga_opt_percent" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> Genetic Algorithm OPTimal PERCENT<br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-content="How to perform a geometry optimization" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../topics/GeoOpt">topic_GeoOpt</a><br />
<em>Variable type:</em> real<br />
<em>Dimensions:</em> scalar<br />
<em>Default value:</em> 0.2  </p>
<details class="note"><summary>Test list (click to open). Rarely used, [1/920] in all abinit tests, [0/116] in abinit tutorials</summary><ul><li>v6:  <a class="wikilink abifile-wikilink" data-content="Genetic algorithm structure random search. Hydrogen has been used as an example. No physical meaning. Number of images is only 10 but experience shows that at least 20 are fine." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v6/Input/t27.in" target="_blank">t27.in</a></li></ul></details><p>Percentage of the population that according to the fitness function passes to
the following iteration.</p>
<h2 id="ga_rules"><strong>ga_rules</strong><a class="headerlink" href="#ga_rules" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> Genetic Algorithm RULES<br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-content="How to perform a geometry optimization" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../topics/GeoOpt">topic_GeoOpt</a><br />
<em>Variable type:</em> integer<br />
<em>Dimensions:</em> scalar<br />
<em>Default value:</em> 1  </p>
<details class="note"><summary>Test list (click to open). Rarely used, [1/920] in all abinit tests, [0/116] in abinit tutorials</summary><ul><li>v6:  <a class="wikilink abifile-wikilink" data-content="Genetic algorithm structure random search. Hydrogen has been used as an example. No physical meaning. Number of images is only 10 but experience shows that at least 20 are fine." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v6/Input/t27.in" target="_blank">t27.in</a></li></ul></details><p>Different genetic rules have been implemented and the user has the change to
choose between any of them. The chosen number of rules have been defined in
[[ga_n_rules]  </p>
<p>Implemented rules are<br />
1) crossover. Two parents are randomly chosen and two springs are mixed from
the two by (a) choosing randomly (through Fitness function) two parents and
then randomly rotating and shifting the coordinates withing that particular
cell. (b) Slice every one of the unit cell of the parents along a random
direction and creating the spring offs from the pieces of the two parents.<br />
2) Vector flip mutation. From the coordinates from a given parent, a piece of
it is inverted.<br />
3) random strain. A random anisotropic deformation is given to the unit cell.<br />
4) Coordinates mutation of &frac14; of the whole coordinates.  </p>
<h2 id="getcell"><strong>getcell</strong><a class="headerlink" href="#getcell" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> GET CELL parameters from &hellip;<br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-content="How to set parameters for a multi dataset calculation" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../topics/multidtset">topic_multidtset</a><br />
<em>Variable type:</em> integer<br />
<em>Dimensions:</em> scalar<br />
<em>Default value:</em> 0  </p>
<details class="note"><summary>Test list (click to open). Rarely used, [4/920] in all abinit tests, [1/116] in abinit tutorials</summary><ul><li>gpu:  <a class="wikilink abifile-wikilink" data-content="Silicon - diamond structure - PAW - volume relaxation Test the use of GPU within Projector Augmented-Wave formalism. Test structure relaxation (multiple calls to gstate routine)." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/gpu/Input/t04.in" target="_blank">t04.in</a></li><li>tutorespfn:  <a class="wikilink abifile-wikilink" data-content="Al fcc metal - elastic constant calculation" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/tutorespfn/Input/telast_6.in" target="_blank">telast_6.in</a></li><li>v1:  <a class="wikilink abifile-wikilink" data-content="2 special k point Si in 2-atom diamond unit cell. Optimization of unit cell volume (optcell=1). Use 2 datasets,  and the variable getcell." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v1/Input/t78.in" target="_blank">t78.in</a>, <a class="wikilink abifile-wikilink" data-content="Mg crystal,  1 atom per unit cell. Start with deformed FCC lattice, optimize the lattice towards FCC (optcell=2). Also test the output of the HIST.nc file (restartxf==-1)" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v1/Input/t80.in" target="_blank">t80.in</a></li></ul></details><p>This variable is typically used to chain the calculations, in the multi-
dataset mode (<a class="wikilink codevar-wikilink" href="../basic#ndtset">ndtset</a>&gt;0), since it describes from which dataset
<a class="wikilink codevar-wikilink" href="../basic#acell">acell</a> and <a class="wikilink codevar-wikilink" href="../basic#rprim">rprim</a> are to be taken, as input of the present dataset. The
cell parameters are <a class="wikilink" href="../../user-guide/abinit#32-more-about-abinit-input-variables">EVOLVING</a> variables, for which such a chain of
calculations is useful.<br />
If ==0, no use of previously computed values must occur.<br />
If it is positive, its value gives the index of the dataset from which the
data are to be used as input data. It must be the index of a dataset already
computed in the SAME run.<br />
If equal to -1, the output data of the previous dataset must be taken, which
is a frequently occurring case. However, if the first dataset is treated, -1
is equivalent to 0, since no dataset has yet been computed in the same run.<br />
If another negative number, it indicates the number of datasets to go backward
to find the needed data (once again, going back beyond the first dataset is
equivalent to using a null get variable).</p>
<h2 id="getvel"><strong>getvel</strong><a class="headerlink" href="#getvel" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> GET VEL from &hellip;<br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-content="How to set parameters for a multi dataset calculation" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../topics/multidtset">topic_multidtset</a><br />
<em>Variable type:</em> integer<br />
<em>Dimensions:</em> scalar<br />
<em>Default value:</em> 0  </p>
<details class="note"><summary>Test list (click to open). Rarely used, [0/920] in all abinit tests, [0/116] in abinit tutorials</summary></details><p>These variables are typically used to chain the calculations, in the multi-
dataset mode (<a class="wikilink codevar-wikilink" href="../basic#ndtset">ndtset</a>&gt;0) since they describe from which dataset the
corresponding output variables are to be taken, as input of the present
dataset. The atomic positions and velocities are <a class="wikilink" href="../../user-guide/abinit#32-more-about-abinit-input-variables">EVOLVING</a> variables, for
which such a chain of calculation is useful.<br />
Note that the use of ** getxcart ** and ** getxred ** differs when <a class="wikilink codevar-wikilink" href="../basic#acell">acell</a>
and <a class="wikilink codevar-wikilink" href="../basic#rprim">rprim</a> are different from one dataset to the other.<br />
If ==0, no use of previously computed values must occur.<br />
If it is positive, its value gives the index of the dataset from which the
data are to be used as input data. It must be the index of a dataset already
computed in the SAME run.<br />
If equal to -1, the output data of the previous dataset must be taken, which
is a frequently occurring case. However, if the first dataset is treated, -1
is equivalent to 0, since no dataset has yet been computed in the same run.<br />
If another negative number, it indicates the number of datasets to go backward
to find the needed data (once again, going back beyond the first dataset is
equivalent to using a null get variable).<br />
Note : ** getxred ** and ** getxcart ** cannot be simultaneously non-zero for
the same dataset. On the other hand the use of <a class="wikilink codevar-wikilink" href="../rlx#getvel">getvel</a> with ** getxred **
is allowed, despite the different coordinate system.</p>
<h2 id="getxcart"><strong>getxcart</strong><a class="headerlink" href="#getxcart" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> GET XCART from &hellip;<br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-content="How to set parameters for a multi dataset calculation" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../topics/multidtset">topic_multidtset</a><br />
<em>Variable type:</em> integer<br />
<em>Dimensions:</em> scalar<br />
<em>Default value:</em> 0  </p>
<details class="note"><summary>Test list (click to open). Rarely used, [8/920] in all abinit tests, [0/116] in abinit tutorials</summary><ul><li>v1:  <a class="wikilink abifile-wikilink" data-content="Test multi-dataset mode. H2 molecule in a box of increasing size. Use ecut=12 (see previous test). 5 different values of acell (convergence study). Use getxcart=-1." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v1/Input/t61.in" target="_blank">t61.in</a>, <a class="wikilink abifile-wikilink" data-content="Test multi-dataset mode. H2 molecule in a box of increasing size. Same as test 61,  except that it reads previous wavefunctions (use getxcart=-1 and getwfk=-1). mkmem/=0 ." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v1/Input/t64.in" target="_blank">t64.in</a>, <a class="wikilink abifile-wikilink" data-content="Test multi-dataset mode. H2 molecule with 1 and 4 k points. (use getxcart=-1 and getwfk=-1). Convergence study with k points." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v1/Input/t65.in" target="_blank">t65.in</a>, <a class="wikilink abifile-wikilink" data-content="Test multi-dataset mode. H2 molecule with 1 k points. Use getxcart=-1 and getwfk=-1 : first perform a geometry optimization, with minimal number of bands,  then compute more bands at fixed geometry, with an increased ecut." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v1/Input/t66.in" target="_blank">t66.in</a>, <a class="wikilink abifile-wikilink" data-content="Test multi-dataset mode. H2 molecule with 1 k points. Convergence study,  using a geometric series for ecut." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v1/Input/t74.in" target="_blank">t74.in</a>, <a class="wikilink abifile-wikilink" data-content="Same as test 66,  H2 molecule with 1 k points,  but uses the restartxf=-1 option to test the output of the HIST.nc file. Note that HIST.nc is not read since one should introduce a getrestartxf variable." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v1/Input/t75.in" target="_blank">t75.in</a></li><li>v3:  <a class="wikilink abifile-wikilink" data-content="H2 molecule : examine the rotational freedom. The present test produces the following vibrational frequencies (with degeneracies indicated): 56.89 i cm-1 (2) 0.41   cm-1 (2) 1.05   cm-1 3800    cm-1 The large frequency corresponds to the stretching mode,  and has the right order of magnitude. The frequencies close to 1 cm-1 corresponds to translation modes,  and are small enough for usual applications. The  56.89 i cm-1 mode corresponds to rotation of the H2 molecule. The magnitude of this frequency might seem quite large. Here are the results of tests made to understand this phenomenon. First,  note that ecut 12  acell 3*14 Increasing the value of ecut to 25 decreases the magnitude of the frequency to 36.8 cm-1 . However,  in order to continue to make it smaller, the cell size must be increased,  and an oscillatory behaviour is observed : 3*16   45.6 i cm-1 3*18   22.7   cm-1 3*20   19.1 i cm-1 3*22   15.7 i cm-1 3*24   13.7   cm-1 Many other tests have been set up. In particular, it was observed that the frequency of the oscillatory behaviour changes with the ecut,  and also that using the Gamma point,  instead of the 1/4 1/4 1/4 k point (used in this test) degrades the convergence. The overall picture is as follows. There are different reasons for the translation and rotation modes to acquire a non-zero frequency when plane waves and supercells are used. Still,  as concerns translations,  only the existence of a discretization of the XC grid is important. For rotations,  supercell effects are also present : - alignement of dipole or quadrupoles - interaction between tails of wavefunctions,  accross cells Since the convergence in supercell size is oscillatory,  we infer that the breaking of the rotational invariance is mostly due to interaction between wavefunction tails. This will be checked by confining the system in a spherical well,  in a forthcoming test." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v3/Input/t80.in" target="_blank">t80.in</a></li><li>v6:  <a class="wikilink abifile-wikilink" data-content="H2 molecule in a reasonably large box Compute the Fan and Diagonal Debye-Waller corrections to the eigenenergies." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v6/Input/t68.in" target="_blank">t68.in</a></li></ul></details><p>These variables are typically used to chain the calculations, in the multi-
dataset mode (<a class="wikilink codevar-wikilink" href="../basic#ndtset">ndtset</a>&gt;0) since they describe from which dataset the
corresponding output variables are to be taken, as input of the present
dataset. The atomic positions and velocities are <a class="wikilink" href="../../user-guide/abinit#32-more-about-abinit-input-variables">EVOLVING</a> variables, for
which such a chain of calculation is useful.<br />
Note that the use of <a class="wikilink codevar-wikilink" href="../rlx#getxcart">getxcart</a> and ** getxred ** differs when <a class="wikilink codevar-wikilink" href="../basic#acell">acell</a> and
<a class="wikilink codevar-wikilink" href="../basic#rprim">rprim</a> are different from one dataset to the other.<br />
If ==0, no use of previously computed values must occur.<br />
If it is positive, its value gives the index of the dataset from which the
data are to be used as input data. It must be the index of a dataset already
computed in the SAME run.<br />
If equal to -1, the output data of the previous dataset must be taken, which
is a frequently occurring case. However, if the first dataset is treated, -1
is equivalent to 0, since no dataset has yet been computed in the same run.<br />
If another negative number, it indicates the number of datasets to go backward
to find the needed data (once again, going back beyond the first dataset is
equivalent to using a null get variable).<br />
Note : ** getxred ** and <a class="wikilink codevar-wikilink" href="../rlx#getxcart">getxcart</a> cannot be simultaneously non-zero for
the same dataset. On the other hand the use of ** getvel ** with ** getxred **
is allowed, despite the different coordinate system.</p>
<h2 id="getxred"><strong>getxred</strong><a class="headerlink" href="#getxred" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> GET XRED from &hellip;<br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-content="How to set parameters for a multi dataset calculation" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../topics/multidtset">topic_multidtset</a><br />
<em>Variable type:</em> integer<br />
<em>Dimensions:</em> scalar<br />
<em>Default value:</em> 0  </p>
<details class="note"><summary>Test list (click to open). Rarely used, [7/920] in all abinit tests, [1/116] in abinit tutorials</summary><ul><li>tutorespfn:  <a class="wikilink abifile-wikilink" data-content="AlAs in hypothetical wurzite (hexagonal) structure Structural optimization run" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/tutorespfn/Input/telast_1.in" target="_blank">telast_1.in</a></li><li>v1:  <a class="wikilink abifile-wikilink" data-content="Test multi-dataset mode. H2 molecule in a box (psp 1h.pspnc). acell constant,  small (7 5 5). 10 different values of ecut (convergence study). Use getxred=-1." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v1/Input/t60.in" target="_blank">t60.in</a>, <a class="wikilink abifile-wikilink" data-content="Test multi-dataset mode. H2 molecule in a box of increasing size. Same as test 61,  except that it reads previous wavefunctions (use getxcart=-1 and getwfk=-1). mkmem/=0 ." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v1/Input/t64.in" target="_blank">t64.in</a>, <a class="wikilink abifile-wikilink" data-content="Test multi-dataset mode. H2 molecule with 1 k points. Convergence study,  using a geometric series for ecut." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v1/Input/t74.in" target="_blank">t74.in</a>, <a class="wikilink abifile-wikilink" data-content="Mg crystal,  1 atom per unit cell. Start with deformed FCC lattice, optimize the lattice towards FCC (optcell=2). Also test the output of the HIST.nc file (restartxf==-1)" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v1/Input/t80.in" target="_blank">t80.in</a></li><li>v2:  <a class="wikilink abifile-wikilink" data-content="Arsenic in rhombohedral structure. Optimization of unit cell size and shape. With 4x4x4 k point grid,  ecut 3.0,  occopt 4 and tsmear 0.06d0, one get acell 3*7.325  angdeg 3*57.36 and xred 0.2277 These input parameters are too small,  but still give a realistic geometry,  since with the much better parameters 12x12x12 k point grid,  ecut 12.0,  occopt 4 and tsmear 0.04d0, one gets acell 3*7.633  angdeg 3*54.95 and xred 0.2296,  while the experimental values are acell 3*7.751   angdeg 3*54.554 and xred 0.2276" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v2/Input/t44.in" target="_blank">t44.in</a></li><li>v6:  <a class="wikilink abifile-wikilink" data-content="Hydrogen diatomic molecule in a cell,  close to BCC. One hydrogen atom is at the origin. The other lies initially along the 1 1 1 direction,  at a distance close to the one of the free dimer distance. The final position is the mirror with respect to the plane perpendicular to x,  passing at 1/2 0 0 . The cell parameter is less than twice the H2 interatomic distance,  so that the transition path come close to the x axis,  although it never reaches it (the closest being in the mirror plane). Test the images : 6 images,  exploring the transition path by a simple algorithm. MG: Does it work in parallel. I tried 2 CPUs and the run get stuck!" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v6/Input/t22.in" target="_blank">t22.in</a></li></ul></details><p>These variables are typically used to chain the calculations, in the multi-
dataset mode (<a class="wikilink codevar-wikilink" href="../basic#ndtset">ndtset</a>&gt;0) since they describe from which dataset the
corresponding output variables are to be taken, as input of the present
dataset. The atomic positions and velocities are <a class="wikilink" href="../../user-guide/abinit#32-more-about-abinit-input-variables">EVOLVING</a> variables, for
which such a chain of calculation is useful.<br />
Note that the use of ** getxcart ** and <a class="wikilink codevar-wikilink" href="../rlx#getxred">getxred</a> differs when <a class="wikilink codevar-wikilink" href="../basic#acell">acell</a> and
<a class="wikilink codevar-wikilink" href="../basic#rprim">rprim</a> are different from one dataset to the other.<br />
If ==0, no use of previously computed values must occur.<br />
If it is positive, its value gives the index of the dataset from which the
data are to be used as input data. It must be the index of a dataset already
computed in the SAME run.<br />
If equal to -1, the output data of the previous dataset must be taken, which
is a frequently occurring case. However, if the first dataset is treated, -1
is equivalent to 0, since no dataset has yet been computed in the same run.<br />
If another negative number, it indicates the number of datasets to go backward
to find the needed data (once again, going back beyond the first dataset is
equivalent to using a null get variable).<br />
Note : <a class="wikilink codevar-wikilink" href="../rlx#getxred">getxred</a> and ** getxcart ** cannot be simultaneously non-zero for
the same dataset. On the other hand the use of ** getvel ** with <a class="wikilink codevar-wikilink" href="../rlx#getxred">getxred</a>
is allowed, despite the different coordinate system.</p>
<h2 id="goprecon"><strong>goprecon</strong><a class="headerlink" href="#goprecon" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> Geometry Optimization PRECONditioner equations<br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-content="How to perform a geometry optimization" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../topics/GeoOpt">topic_GeoOpt</a><br />
<em>Variable type:</em> integer<br />
<em>Dimensions:</em> scalar<br />
<em>Default value:</em> 0  </p>
<details class="note"><summary>Test list (click to open). Rarely used, [1/920] in all abinit tests, [0/116] in abinit tutorials</summary><ul><li>v5:  <a class="wikilink abifile-wikilink" data-content="Test of geometry optimization using using ionmov=1, 4, 5 and 7 Hydrogen molecule inside a box of 12 Bohr This test was created to test ionmov=5" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v5/Input/t02.in" target="_blank">t02.in</a></li></ul></details><p>Set the kind of preconditioner to be used for Geometry Optimization<br />
(Note : Under development now (2011.05.20))</p>
<ul>
<li><a class="wikilink codevar-wikilink" href="../rlx#goprecon">goprecon</a>=0 : No preconditioner </li>
<li><a class="wikilink codevar-wikilink" href="../rlx#goprecon">goprecon</a>=[1-9] : Linear preconditioner </li>
<li><a class="wikilink codevar-wikilink" href="../rlx#goprecon">goprecon</a>=[11-19] : Non-linear preconditioner </li>
</ul>
<h2 id="goprecprm"><strong>goprecprm</strong><a class="headerlink" href="#goprecprm" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> Geometry Optimization PREconditioner PaRaMeters equations<br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-content="How to perform a geometry optimization" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../topics/GeoOpt">topic_GeoOpt</a><br />
<em>Variable type:</em> real<br />
<em>Dimensions:</em> (3)<br />
<em>Default value:</em> 0  </p>
<details class="note"><summary>Test list (click to open). Rarely used, [1/920] in all abinit tests, [0/116] in abinit tutorials</summary><ul><li>v5:  <a class="wikilink abifile-wikilink" data-content="Test of geometry optimization using using ionmov=1, 4, 5 and 7 Hydrogen molecule inside a box of 12 Bohr This test was created to test ionmov=5" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v5/Input/t02.in" target="_blank">t02.in</a></li></ul></details><p>Set the paramenters use by the preconditioner to be used for Geometry
Optimization<br />
(Note : Under development now (2011.06.06))</p>
<h2 id="iatcon"><strong>iatcon</strong><a class="headerlink" href="#iatcon" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> Indices of AToms in CONstraint equations<br />
<em>Characteristics:</em> <a class="wikilink" href="../../user-guide/abinit#32-more-about-abinit-input-variables">NO_MULTI</a>, <a class="wikilink" href="../../user-guide/abinit#32-more-about-abinit-input-variables">INPUT_ONLY</a><br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-content="How to constaint the geometry of the system in geometry optimization, molecular
dynamics or searches" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../topics/GeoConstraints">topic_GeoConstraints</a><br />
<em>Variable type:</em> integer<br />
<em>Dimensions:</em> (<a class="wikilink codevar-wikilink" href="../rlx#natcon">natcon</a>,<a class="wikilink codevar-wikilink" href="../rlx#nconeq">nconeq</a>)<br />
<em>Default value:</em> 0  </p>
<details class="note"><summary>Test list (click to open). Rarely used, [3/920] in all abinit tests, [0/116] in abinit tutorials</summary><ul><li>v3:  <a class="wikilink abifile-wikilink" data-content="8 atoms of Si in an elongated box. 4 of them are fixed,  while atoms 5 and 6 and 7 and 8 are grouped by pair,  such as the sum of their relative separations along y is constrained to 8.0 . This is a crazy constraint,  just to show that it can be done !" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v3/Input/t40.in" target="_blank">t40.in</a></li><li>v6:  <a class="wikilink abifile-wikilink" data-content="Hydrogen diatomic molecule in a cell,  close to BCC. One hydrogen atom is at the origin. The other lies initially along the 1 1 1 direction,  at a distance close to the one of the free dimer distance. The final position is the mirror with respect to the plane perpendicular to x,  passing at 1/2 0 0 . The cell parameter is less than twice the H2 interatomic distance,  so that the transition path come close to the x axis,  although it never reaches it (the closest being in the mirror plane). Test the images : 6 images,  exploring the transition path by a simple algorithm. MG: Does it work in parallel. I tried 2 CPUs and the run get stuck!" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v6/Input/t22.in" target="_blank">t22.in</a></li><li>v8:  <a class="wikilink abifile-wikilink" data-content="H2 molecule: PIMD simulation with Langevin thermostat. Test of restart with HIST.nc file. Test of linear constraint." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v8/Input/t05.in" target="_blank">t05.in</a></li></ul></details><p>Gives the indices of the atoms appearing in each of the <a class="wikilink codevar-wikilink" href="../rlx#nconeq">nconeq</a> independent
equations constraining the motion of atoms during structural optimization or
molecular dynamics (see <a class="wikilink codevar-wikilink" href="../rlx#nconeq">nconeq</a> , <a class="wikilink codevar-wikilink" href="../rlx#natcon">natcon</a>, and <a class="wikilink codevar-wikilink" href="../rlx#wtatcon">wtatcon</a>).<br />
(Note : combined with wtatcon to give internal representation of the latter -
this should be described)</p>
<h2 id="iatfix"><strong>iatfix</strong><a class="headerlink" href="#iatfix" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> Indices of AToms that are FIXed <br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-content="How to constaint the geometry of the system in geometry optimization, molecular
dynamics or searches" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../topics/GeoConstraints">topic_GeoConstraints</a><br />
<em>Variable type:</em> integer<br />
<em>Dimensions:</em> (<a class="wikilink codevar-wikilink" href="../rlx#natfix">natfix</a>)<br />
<em>Default value:</em> None<br />
<em>Only relevant if:</em> <a class="wikilink codevar-wikilink" href="../rlx#natfix">natfix</a> &gt; 0  </p>
<details class="note"><summary>Test list (click to open). Moderately used, [16/920] in all abinit tests, [2/116] in abinit tutorials</summary><ul><li>fast:  <a class="wikilink abifile-wikilink" data-content="brdmin with natfix. Output also DEN,  POT and GEO files" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/fast/Input/t28.in" target="_blank">t28.in</a>, <a class="wikilink abifile-wikilink" data-content="move with natfix. Output also GEO files" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/fast/Input/t29.in" target="_blank">t29.in</a></li><li>paral:  <a class="wikilink abifile-wikilink" data-content="Test the string method within parallelization over images Inspired by test v6#22. Hydrogen diatomic molecule in a cell,  close to BCC 7 images,  exploring the transition path. Three datasets,  testing each value of prtvolimg. Processors distribution automatically determined:" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/paral/Input/t08.in" target="_blank">t08.in</a>, <a class="wikilink abifile-wikilink" data-content="Test the string method within parallelization over images Inspired by test v6#22. Hydrogen diatomic molecule in a cell,  close to BCC 7 images,  exploring the transition path. Three datasets,  testing each value of prtvolimg. Processors distribution automatically determined:" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/paral/Input/t08.in" target="_blank">t08.in</a>, <a class="wikilink abifile-wikilink" data-content="Test the string method within parallelization over images Inspired by test v6#22. Hydrogen diatomic molecule in a cell,  close to BCC 7 images,  exploring the transition path. Three datasets,  testing each value of prtvolimg. Processors distribution automatically determined:" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/paral/Input/t08.in" target="_blank">t08.in</a>, <a class="wikilink abifile-wikilink" data-content="Test the string method within parallelization over images Inspired by test v6#22. Hydrogen diatomic molecule in a cell,  close to BCC 7 images,  exploring the transition path. Three datasets,  testing each value of prtvolimg. Processors distribution automatically determined:" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/paral/Input/t08.in" target="_blank">t08.in</a></li><li>tutoparal:  <a class="wikilink abifile-wikilink" data-content="Hydronium ion + NH3 molecule Ground state calculation keeping O and H atoms fixed" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/tutoparal/Input/tstring_01.in" target="_blank">tstring_01.in</a></li><li>tutorespfn:  <a class="wikilink abifile-wikilink" data-content="AlAs in hypothetical wurzite (hexagonal) structure Structural optimization run" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/tutorespfn/Input/telast_1.in" target="_blank">telast_1.in</a></li><li>v1:  <a class="wikilink abifile-wikilink" data-content="Test the option ionmov=4. 8-atom Si cube,  non spin-polarized. Same test as number 22a of fast,  except for the change of ionmov,  no reading of wavefunctions,  iscf to 2,  and different tolerance criteria." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v1/Input/t44.in" target="_blank">t44.in</a>, <a class="wikilink abifile-wikilink" data-content="SiH4 : optimize the geometry with partially constrained atomic positions. 1 special k-point,  in a box 8x8x8 (too small),  and 8Ha cut-off. Test ionmov=2 Permutes the three axes x, y, z in three datasets. Since the initialization is different,  the three datasets deliver slightly different results,  though quite close to each others. Should be identical at convergence." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v1/Input/t85.in" target="_blank">t85.in</a>, <a class="wikilink abifile-wikilink" data-content="SiH4 : molecular dynamics with partially constrained atomic positions. 1 special k-point,  in a box 8x8x8 (too small),  and 8Ha cut-off. Test ionmov=1" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v1/Input/t86.in" target="_blank">t86.in</a></li><li>v2:  <a class="wikilink abifile-wikilink" data-content="Basic test of geometry optimization for water First computing the ground state of a close configuration Uses 5 different methods to find the relaxed positions" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v2/Input/t88.in" target="_blank">t88.in</a></li><li>v3:  <a class="wikilink abifile-wikilink" data-content="8 atoms of Si in an elongated box. 4 of them are fixed,  while atoms 5 and 6 and 7 and 8 are grouped by pair,  such as the sum of their relative separations along y is constrained to 8.0 . This is a crazy constraint,  just to show that it can be done !" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v3/Input/t40.in" target="_blank">t40.in</a></li><li>v6:  <a class="wikilink abifile-wikilink" data-content="Hydrogen diatomic molecule in a cell,  close to BCC. One hydrogen atom is at the origin. The other lies initially along the 1 1 1 direction,  at a distance close to the one of the free dimer distance. The final position is the mirror with respect to the plane perpendicular to x,  passing at 1/2 0 0 . The cell parameter is less than twice the H2 interatomic distance,  so that the transition path come close to the x axis,  although it never reaches it (the closest being in the mirror plane). Test the images : 6 images,  exploring the transition path by a simple algorithm. MG: Does it work in parallel. I tried 2 CPUs and the run get stuck!" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v6/Input/t22.in" target="_blank">t22.in</a>, <a class="wikilink abifile-wikilink" data-content="Hydrogen diatomic molecule in a cell,  close to BCC Test the string method : 6 images,  exploring the transition path. Similar to test 22 (but with string)." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v6/Input/t24.in" target="_blank">t24.in</a>, <a class="wikilink abifile-wikilink" data-content="Hydrogen diatomic molecule in a cell,  close to BCC Test the string method : 6 images,  exploring the transition path but keeping X coordinated fixed. Similar to test 24" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v6/Input/t25.in" target="_blank">t25.in</a></li></ul></details><p>Give the index (in the range 1 to <a class="wikilink codevar-wikilink" href="../basic#natom">natom</a> ) of each atom which is to be held
fixed for structural optimization or molecular dynamics. The variable
<a class="wikilink codevar-wikilink" href="../rlx#iatfix">iatfix</a> lists those fixed in the three directions, while the variables
<a class="wikilink codevar-wikilink" href="../rlx#iatfixx">iatfixx</a>, <a class="wikilink codevar-wikilink" href="../rlx#iatfixy">iatfixy</a>, and <a class="wikilink codevar-wikilink" href="../rlx#iatfixz">iatfixz</a>, allow to fix some atoms along x, y
or z directions, or a combination of these.</p>
<p>WARNING : The implementation is inconsistent !! For <a class="wikilink codevar-wikilink" href="../rlx#ionmov">ionmov</a> ==1, the fixing
of directions was done in cartesian coordinates, while for the other values of
<a class="wikilink codevar-wikilink" href="../rlx#ionmov">ionmov</a>, it was done in reduced coordinates. Sorry for this.</p>
<p>There is no harm in fixing one atom in the three directions using <a class="wikilink codevar-wikilink" href="../rlx#iatfix">iatfix</a>,
then fixing it again in other directions by mentioning it in ** iatfixx ** ,
** iatfixy ** or ** iatfixz ** .<br />
The internal representation of these input data is done by the mean of one
variable <a class="wikilink codevar-wikilink" href="../rlx#iatfix">iatfix</a>(3,<a class="wikilink codevar-wikilink" href="../basic#natom">natom</a>), defined for each direction and each atom,
being 0 if the atom is not fixed along the direction, and 1 if the atom is
fixed along the direction. When some atoms are fixed along 1 or 2 directions,
the use of symmetries is restricted to symmetry operations whose (3x3)
matrices <a class="wikilink codevar-wikilink" href="../basic#symrel">symrel</a> are diagonal.<br />
If the atom manipulator is used, <a class="wikilink codevar-wikilink" href="../rlx#iatfix">iatfix</a> will be related to the
preprocessed set of atoms, generated by the atom manipulator. The user must
thus foresee the effect of this atom manipulator (see <a class="wikilink codevar-wikilink" href="../geo#objarf">objarf</a>).</p>
<h2 id="iatfixx"><strong>iatfixx</strong><a class="headerlink" href="#iatfixx" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> Indices of AToms that are FIXed along the X direction<br />
<em>Characteristics:</em> <a class="wikilink" href="../../user-guide/abinit#32-more-about-abinit-input-variables">INPUT_ONLY</a><br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-content="How to constaint the geometry of the system in geometry optimization, molecular
dynamics or searches" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../topics/GeoConstraints">topic_GeoConstraints</a><br />
<em>Variable type:</em> integer<br />
<em>Dimensions:</em> (<a class="wikilink codevar-wikilink" href="../rlx#natfixx">natfixx</a>)<br />
<em>Default value:</em> None<br />
<em>Only relevant if:</em> <a class="wikilink codevar-wikilink" href="../rlx#natfixx">natfixx</a> &gt; 0  </p>
<details class="note"><summary>Test list (click to open). Rarely used, [2/920] in all abinit tests, [0/116] in abinit tutorials</summary><ul><li>v1:  <a class="wikilink abifile-wikilink" data-content="SiH4 : optimize the geometry with partially constrained atomic positions. 1 special k-point,  in a box 8x8x8 (too small),  and 8Ha cut-off. Test ionmov=2 Permutes the three axes x, y, z in three datasets. Since the initialization is different,  the three datasets deliver slightly different results,  though quite close to each others. Should be identical at convergence." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v1/Input/t85.in" target="_blank">t85.in</a></li><li>v6:  <a class="wikilink abifile-wikilink" data-content="Hydrogen diatomic molecule in a cell,  close to BCC Test the string method : 6 images,  exploring the transition path but keeping X coordinated fixed. Similar to test 24" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v6/Input/t25.in" target="_blank">t25.in</a></li></ul></details><p>Give the index (in the range 1 to <a class="wikilink codevar-wikilink" href="../basic#natom">natom</a> ) of each atom which is to be held
fixed ALONG THE X direction for structural optimization or molecular dynamics.
The variable <a class="wikilink codevar-wikilink" href="../rlx#iatfix">iatfix</a> lists those fixed in the three directions, while the
variables <a class="wikilink codevar-wikilink" href="../rlx#iatfixx">iatfixx</a>, <a class="wikilink codevar-wikilink" href="../rlx#iatfixy">iatfixy</a>, and <a class="wikilink codevar-wikilink" href="../rlx#iatfixz">iatfixz</a>, allow to fix some atoms
along x, y or z directions, or a combination of these. See the variable
<a class="wikilink codevar-wikilink" href="../rlx#iatfix">iatfix</a> for more information.</p>
<h2 id="iatfixy"><strong>iatfixy</strong><a class="headerlink" href="#iatfixy" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> Indices of AToms that are FIXed along the Y direction<br />
<em>Characteristics:</em> <a class="wikilink" href="../../user-guide/abinit#32-more-about-abinit-input-variables">INPUT_ONLY</a><br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-content="How to constaint the geometry of the system in geometry optimization, molecular
dynamics or searches" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../topics/GeoConstraints">topic_GeoConstraints</a><br />
<em>Variable type:</em> integer<br />
<em>Dimensions:</em> (<a class="wikilink codevar-wikilink" href="../rlx#natfixy">natfixy</a>)<br />
<em>Default value:</em> None<br />
<em>Only relevant if:</em> <a class="wikilink codevar-wikilink" href="../rlx#natfixy">natfixy</a> &gt; 0  </p>
<details class="note"><summary>Test list (click to open). Rarely used, [2/920] in all abinit tests, [0/116] in abinit tutorials</summary><ul><li>v1:  <a class="wikilink abifile-wikilink" data-content="SiH4 : optimize the geometry with partially constrained atomic positions. 1 special k-point,  in a box 8x8x8 (too small),  and 8Ha cut-off. Test ionmov=2 Permutes the three axes x, y, z in three datasets. Since the initialization is different,  the three datasets deliver slightly different results,  though quite close to each others. Should be identical at convergence." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v1/Input/t85.in" target="_blank">t85.in</a>, <a class="wikilink abifile-wikilink" data-content="SiH4 : molecular dynamics with partially constrained atomic positions. 1 special k-point,  in a box 8x8x8 (too small),  and 8Ha cut-off. Test ionmov=1" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v1/Input/t86.in" target="_blank">t86.in</a></li></ul></details><p>Give the index (in the range 1 to <a class="wikilink codevar-wikilink" href="../basic#natom">natom</a> ) of each atom which is to be held
fixed ALONG THE Y direction for structural optimization or molecular dynamics.
The variable <a class="wikilink codevar-wikilink" href="../rlx#iatfix">iatfix</a> lists those fixed in the three directions, while the
variables <a class="wikilink codevar-wikilink" href="../rlx#iatfixx">iatfixx</a>, <a class="wikilink codevar-wikilink" href="../rlx#iatfixy">iatfixy</a>, and <a class="wikilink codevar-wikilink" href="../rlx#iatfixz">iatfixz</a>, allow to fix some atoms
along x, y or z directions, or a combination of these. See the variable
<a class="wikilink codevar-wikilink" href="../rlx#iatfix">iatfix</a> for more information.</p>
<h2 id="iatfixz"><strong>iatfixz</strong><a class="headerlink" href="#iatfixz" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> Indices of AToms that are FIXed along the Z direction<br />
<em>Characteristics:</em> <a class="wikilink" href="../../user-guide/abinit#32-more-about-abinit-input-variables">INPUT_ONLY</a><br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-content="How to constaint the geometry of the system in geometry optimization, molecular
dynamics or searches" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../topics/GeoConstraints">topic_GeoConstraints</a><br />
<em>Variable type:</em> integer<br />
<em>Dimensions:</em> (<a class="wikilink codevar-wikilink" href="../rlx#natfixz">natfixz</a>)<br />
<em>Default value:</em> None<br />
<em>Only relevant if:</em> <a class="wikilink codevar-wikilink" href="../rlx#natfixz">natfixz</a> &gt; 0  </p>
<details class="note"><summary>Test list (click to open). Rarely used, [2/920] in all abinit tests, [0/116] in abinit tutorials</summary><ul><li>v1:  <a class="wikilink abifile-wikilink" data-content="SiH4 : optimize the geometry with partially constrained atomic positions. 1 special k-point,  in a box 8x8x8 (too small),  and 8Ha cut-off. Test ionmov=2 Permutes the three axes x, y, z in three datasets. Since the initialization is different,  the three datasets deliver slightly different results,  though quite close to each others. Should be identical at convergence." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v1/Input/t85.in" target="_blank">t85.in</a>, <a class="wikilink abifile-wikilink" data-content="SiH4 : molecular dynamics with partially constrained atomic positions. 1 special k-point,  in a box 8x8x8 (too small),  and 8Ha cut-off. Test ionmov=1" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v1/Input/t86.in" target="_blank">t86.in</a></li></ul></details><p>Give the index (in the range 1 to <a class="wikilink codevar-wikilink" href="../basic#natom">natom</a> ) of each atom which is to be held
fixed ALONG THE Z direction for structural optimization or molecular dynamics.
The variable <a class="wikilink codevar-wikilink" href="../rlx#iatfix">iatfix</a> lists those fixed in the three directions, while the
variables <a class="wikilink codevar-wikilink" href="../rlx#iatfixx">iatfixx</a>, <a class="wikilink codevar-wikilink" href="../rlx#iatfixy">iatfixy</a>, and <a class="wikilink codevar-wikilink" href="../rlx#iatfixz">iatfixz</a>, allow to fix some atoms
along x, y or z directions, or a combination of these. See the variable
<a class="wikilink codevar-wikilink" href="../rlx#iatfix">iatfix</a> for more information.</p>
<h2 id="imgmov"><strong>imgmov</strong><a class="headerlink" href="#imgmov" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> IMaGe MOVEs<br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-content="How to perform a PIMD calculation" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../topics/PIMD">topic_PIMD</a>, <a class="wikilink topic-wikilink" data-content="How to calculate transition paths" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../topics/TransPath">topic_TransPath</a><br />
<em>Variable type:</em> integer<br />
<em>Dimensions:</em> scalar<br />
<em>Default value:</em> 0  </p>
<details class="note"><summary>Test list (click to open). Moderately used, [11/920] in all abinit tests, [0/116] in abinit tutorials</summary><ul><li>paral:  <a class="wikilink abifile-wikilink" data-content="Test the string method within parallelization over images Inspired by test v6#22. Hydrogen diatomic molecule in a cell,  close to BCC 7 images,  exploring the transition path. Three datasets,  testing each value of prtvolimg. Processors distribution automatically determined:" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/paral/Input/t08.in" target="_blank">t08.in</a>, <a class="wikilink abifile-wikilink" data-content="Test the string method within parallelization over images Inspired by test v6#22. Hydrogen diatomic molecule in a cell,  close to BCC 7 images,  exploring the transition path. Three datasets,  testing each value of prtvolimg. Processors distribution automatically determined:" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/paral/Input/t08.in" target="_blank">t08.in</a>, <a class="wikilink abifile-wikilink" data-content="Test the string method within parallelization over images Inspired by test v6#22. Hydrogen diatomic molecule in a cell,  close to BCC 7 images,  exploring the transition path. Three datasets,  testing each value of prtvolimg. Processors distribution automatically determined:" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/paral/Input/t08.in" target="_blank">t08.in</a>, <a class="wikilink abifile-wikilink" data-content="Test the string method within parallelization over images Inspired by test v6#22. Hydrogen diatomic molecule in a cell,  close to BCC 7 images,  exploring the transition path. Three datasets,  testing each value of prtvolimg. Processors distribution automatically determined:" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/paral/Input/t08.in" target="_blank">t08.in</a></li><li>v6:  <a class="wikilink abifile-wikilink" data-content="Bismuth,  rhombohedral,  two atoms per unit cell Scalar relativistic. Relax four different images,  using the steepest descent algorithm." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v6/Input/t21.in" target="_blank">t21.in</a>, <a class="wikilink abifile-wikilink" data-content="Hydrogen diatomic molecule in a cell,  close to BCC Test the string method : 6 images,  exploring the transition path. Similar to test 22 (but with string)." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v6/Input/t24.in" target="_blank">t24.in</a>, <a class="wikilink abifile-wikilink" data-content="Hydrogen diatomic molecule in a cell,  close to BCC Test the string method : 6 images,  exploring the transition path but keeping X coordinated fixed. Similar to test 24" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v6/Input/t25.in" target="_blank">t25.in</a>, <a class="wikilink abifile-wikilink" data-content="(TO BE MODIFIED BY GREGORY) Hydrogen diatomic molecule in a cell,  close to BCC Test the string method : 6 images,  exploring the transition path but keeping X coordinated fixed. Similar to test 24" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v6/Input/t26.in" target="_blank">t26.in</a>, <a class="wikilink abifile-wikilink" data-content="Genetic algorithm structure random search. Hydrogen has been used as an example. No physical meaning. Number of images is only 10 but experience shows that at least 20 are fine." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v6/Input/t27.in" target="_blank">t27.in</a></li><li>v7:  <a class="wikilink abifile-wikilink" data-content="Hydrogen molecule: Test the Path-Integral Molecular Dynamics (PIMD) implementation in (N, V, T) ensemble. THe two thermostats (Nose-Hoover chains and Langevin) are tested together with the three coordinate systems (primitive,  normal mode,  staging). NOTE: MPI run with 2 procs is ok but fldiff comparison fails due to an extra section with --- Pseudopotential description --" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v7/Input/t08.in" target="_blank">t08.in</a></li><li>v8:  <a class="wikilink abifile-wikilink" data-content="H2 molecule: PIMD simulation with Langevin thermostat. Test of restart with HIST.nc file. Test of linear constraint." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v8/Input/t05.in" target="_blank">t05.in</a></li></ul></details><p>Control the collective changes of images (see <a class="wikilink codevar-wikilink" href="../rlx#nimage">nimage</a>,<a class="wikilink codevar-wikilink" href="../paral#npimage">npimage</a>,
<a class="wikilink codevar-wikilink" href="../rlx#dynimage">dynimage</a>, <a class="wikilink codevar-wikilink" href="../rlx#ntimimage">ntimimage</a>, <a class="wikilink codevar-wikilink" href="../rlx#tolimg">tolimg</a>, <a class="wikilink codevar-wikilink" href="../rlx#istatimg">istatimg</a>, <a class="wikilink codevar-wikilink" href="../files#prtvolimg">prtvolimg</a>).<br />
Similar to <a class="wikilink codevar-wikilink" href="../rlx#ionmov">ionmov</a> in spirit, although here, a population of self-
consistent calculations for different geometries is managed, while with
<a class="wikilink codevar-wikilink" href="../rlx#ionmov">ionmov</a>, only one geometry for self-consistent calculation is managed.<br />
In this respect the maximal number of time step for image propagation is
<a class="wikilink codevar-wikilink" href="../rlx#ntimimage">ntimimage</a>, corresponding to the input variable <a class="wikilink codevar-wikilink" href="../rlx#ntime">ntime</a> of the single
geometry case. Also, the stopping criterion is governed by <a class="wikilink codevar-wikilink" href="../rlx#tolimg">tolimg</a>,
corresponding to the input variable <a class="wikilink codevar-wikilink" href="../basic#toldfe">toldfe</a> of the single geometry case.
The stopping condition is crude: the image propagation is stopped when the
mean value (over dynamic images) of the absolute difference of total energy
(previous and current time step) is less than <a class="wikilink codevar-wikilink" href="../rlx#tolimg">tolimg</a>.</p>
<p>Actually, there might be combinations of <a class="wikilink codevar-wikilink" href="../rlx#ionmov">ionmov</a> and <a class="wikilink codevar-wikilink" href="../rlx#imgmov">imgmov</a> in which
the two mechanisms are at work. Usually, however, only one mechanism will be
activated (so, usually, either <a class="wikilink codevar-wikilink" href="../rlx#ntimimage">ntimimage</a> is bigger than one OR <a class="wikilink codevar-wikilink" href="../rlx#ntime">ntime</a>
is bigger than one). In order for the user to acquire a mental representation
of the interplay between <a class="wikilink codevar-wikilink" href="../rlx#ionmov">ionmov</a> and <a class="wikilink codevar-wikilink" href="../rlx#imgmov">imgmov</a>, here is a F90 pseudo-code
presenting the interplay between the different above-mentioned input
variables, as well as with the parallelism (see input variable <a class="wikilink codevar-wikilink" href="../paral#npimage">npimage</a>).</p>
<div class="codehilite"><pre><span></span>do itimimage=1,ntimimage
  do iimage=1,nimage
    (possibly, parallelisation over images)
    do itime=1,ntime
      Compute the forces and stresses for image(iimage)
      Examine whether the stopping criterion defined by tolmxf is fulfilled
      Predict the next geometry for image(iimage) using ionmov
    enddo
  enddo
  Examine whether the stopping criterion defined by tolimg is fulfilled
  Predict the next geometries for all images using imgmov
enddo
</pre></div>


<ul>
<li>= 0 =&gt; simply ** copy ** images from previous timimage step.</li>
<li>= 1 =&gt; move images according to ** Steepest Descent ** following the (scaled) forces, the scaling factor being <a class="wikilink codevar-wikilink" href="../rlx#fxcartfactor">fxcartfactor</a>. </li>
<li>= 2 =&gt; ** String Method ** for finding Minimal Energy Path (MEP) connecting to minima (see PRB 66, 052301 (2002)); the algorithm variant can be selected with the <a class="wikilink codevar-wikilink" href="../rlx#string_algo">string_algo</a> keyword (Simplified String Method by default). The solver for the Ordinary Differential Equation (ODE) can be selected with <a class="wikilink codevar-wikilink" href="../rlx#mep_solver">mep_solver</a> (steepest-descent by default). See also <a class="wikilink codevar-wikilink" href="../rlx#mep_mxstep">mep_mxstep</a> keyword.</li>
<li>= 3 =&gt; (tentatively, not yet coded) ** Metadynamics **.</li>
<li>= 4 =&gt; (tentatively, not yet coded) ** Genetic Algorithm**.</li>
<li>= 5 =&gt; ** Nudged Elastic Band (NEB) ** for finding Minimal Energy Path (MEP) connecting two minima; the algorithm variant can be selected with the <a class="wikilink codevar-wikilink" href="../rlx#neb_algo">neb_algo</a> keyword (NEB+improved tangent by default). The solver for the Ordinary Differential Equation (ODE) can be selected with <a class="wikilink codevar-wikilink" href="../rlx#mep_solver">mep_solver</a> (steepest-descent by default). The spring constant connecting images along the path is defined by <a class="wikilink codevar-wikilink" href="../rlx#neb_spring">neb_spring</a>. See also <a class="wikilink codevar-wikilink" href="../rlx#mep_mxstep">mep_mxstep</a> keyword.</li>
<li>= 9 or 13 =&gt; ** Path-Integral Molecular Dynamics ** (see e.g. [D. Marx and M. Parrinello, J. Chem. Phys. 104, 4077 (1996)]). Will use 9 for ** Langevin thermostat ** (associated friction coefficient given by <a class="wikilink codevar-wikilink" href="../rlx#vis">vis</a>) and 13 for ** Nose-Hoover thermostat chains ** (associated input variables are the number of thermostats in the chains, <a class="wikilink codevar-wikilink" href="../rlx#nnos">nnos</a>, and the masses of these thermostats <a class="wikilink codevar-wikilink" href="../rlx#qmass">qmass</a>). <a class="wikilink codevar-wikilink" href="../rlx#nimage">nimage</a> is the Trotter number (no use of <a class="wikilink codevar-wikilink" href="../rlx#dynimage">dynimage</a>); possible transformations of coordinates are defined by <a class="wikilink codevar-wikilink" href="../rlx#pitransform">pitransform</a>; Fictitious masses of the atoms (possibly different from the true masses given by <a class="wikilink codevar-wikilink" href="../rlx#amu">amu</a>) can be specified by <a class="wikilink codevar-wikilink" href="../rlx#pimass">pimass</a>. At present, it is only possible to perform calculations in the (N,V,T) ensemble (<a class="wikilink codevar-wikilink" href="../rlx#optcell">optcell</a>=0).</li>
</ul>
<p>No meaning for RF calculations.</p>
<h2 id="ionmov"><strong>ionmov</strong><a class="headerlink" href="#ionmov" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> IONic MOVEs<br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-content="How to perform a molecular dynamics calculation" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../topics/MolecularDynamics">topic_MolecularDynamics</a>, <a class="wikilink topic-wikilink" data-content="How to perform a geometry optimization" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../topics/GeoOpt">topic_GeoOpt</a><br />
<em>Variable type:</em> integer<br />
<em>Dimensions:</em> scalar<br />
<em>Default value:</em> 0  </p>
<details class="note"><summary>Test list (click to open). Moderately used, [111/920] in all abinit tests, [19/116] in abinit tutorials</summary><ul><li>bigdft:  <a class="wikilink abifile-wikilink" data-content="H2 molecule with geometry optimisation using BFGS (test the reformating part of the code)." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/bigdft/Input/t10.in" target="_blank">t10.in</a>, <a class="wikilink abifile-wikilink" data-content="H2O in isolated boundary conditions (plane waves,  NC),  try tu use Damiens Poisson cutoff. Water molecule with HGH pseudo-potentials. This checks that a negative value in rhor will not make the Poisson solver crashes on XC computation." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/bigdft/Input/t22.in" target="_blank">t22.in</a> &hellip;</li><li>built-in:  <a class="wikilink abifile-wikilink" data-content="H2 molecule,  determination of the H-H distance by the Broyden algorithm." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/built-in/Input/testin_fast.in" target="_blank">testin_fast.in</a>, <a class="wikilink abifile-wikilink" data-content="Ca atom with PAW,  writing a ETSF_IO wavefunction file." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/built-in/Input/testin_etsf_io.in" target="_blank">testin_etsf_io.in</a> &hellip;</li><li>etsf_io:  <a class="wikilink abifile-wikilink" data-content="Ca atom with PAW,  writing a ETSF_IO wavefunction file. Very quick built-in test,  to check that ETSF_IO is installed." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/etsf_io/Input/t00.in" target="_blank">t00.in</a>, <a class="wikilink abifile-wikilink" data-content="Ca atom in PAW. Test the output of density in PAW formalism with ETSF_IO. Also test the output of the wavefunctions in PAW formalism with ETSF_IO." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/etsf_io/Input/t09.in" target="_blank">t09.in</a>, <a class="wikilink abifile-wikilink" data-content="Si2 molecule,  static,  spin-polarized. Same system as test 49 of v2,  except that nctime is non-zero. Test ionmov=6 (Verlet) as well as ionmov=7." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/etsf_io/Input/t21.in" target="_blank">t21.in</a> &hellip;</li><li>fast:  <a class="wikilink abifile-wikilink" data-content="H2 molecule,  determination of the H-H distance by the Broyden algorithm. Very quick built-in test,  to check that ABINIT is working." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/fast/Input/t00.in" target="_blank">t00.in</a>, <a class="wikilink abifile-wikilink" data-content="Conduct relaxation of molecule starting from wf.19,  using ionmov=2. (Broyden method)." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/fast/Input/t20.in" target="_blank">t20.in</a>, <a class="wikilink abifile-wikilink" data-content="21. Conduct same relaxation but use ionmov=1 (molecular dynamics)." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/fast/Input/t21.in" target="_blank">t21.in</a> &hellip;</li><li>gpu:  <a class="wikilink abifile-wikilink" data-content="Silicon - diamond structure - PAW - volume relaxation Test the use of GPU within Projector Augmented-Wave formalism. Test structure relaxation (multiple calls to gstate routine)." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/gpu/Input/t04.in" target="_blank">t04.in</a> &hellip;</li><li>mpiio:  <a class="wikilink abifile-wikilink" data-content="Si,  Bulk,  2 atoms,  paral_kgb. Test of ground state with different occs(7 and 0), and also ionmov 2. Only with 0 and 4 procs." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/mpiio/Input/t21.in" target="_blank">t21.in</a>, <a class="wikilink abifile-wikilink" data-content="C-diamond,  Bulk,  2 atoms,  paral_kgb,  with PAW. Test of ground state with different occs(7 and 0), and also ionmov 2. Only with 0 and 4 procs." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/mpiio/Input/t22.in" target="_blank">t22.in</a>, <a class="wikilink abifile-wikilink" data-content="C-diamond,  Bulk,  2 atoms,  paral_kgb,  with PAW. Test the triple parallelisation. We cannot check the distribution npband*npfft*npkpt=2*2*2=8 processors, this number being not allowed in the test procedure. Here,  we only test the parallelisation over FFT and kpoints: npband*npfft*npkpt=1*2*2=4 processors. In test tY.in we check other distributions for guarantee." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/mpiio/Input/t24.in" target="_blank">t24.in</a> &hellip;</li><li>paral:  <a class="wikilink abifile-wikilink" data-content="Si,  Bulk,  2 atoms,  paral_kgb. Test of ground state with different occs(7 and 0), and also ionmov 2. Only with 0 and 4 procs." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/paral/Input/t21.in" target="_blank">t21.in</a>, <a class="wikilink abifile-wikilink" data-content="C-diamond,  Bulk,  2 atoms,  paral_kgb,  with PAW. Test of ground state with different occs(7 and 0), and also ionmov 2. Only with 0 and 4 procs." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/paral/Input/t22.in" target="_blank">t22.in</a>, <a class="wikilink abifile-wikilink" data-content="C-diamond,  Bulk,  2 atoms,  paral_kgb,  with PAW. Test the triple parallelisation. We cannot check the distribution npband*npfft*npkpt=2*2*2=8 processors, this number being not allowed in the test procedure. Here,  we only test the parallelisation over FFT and kpoints: npband*npfft*npkpt=1*2*2=4 processors. In test tY.in we check other distributions for guarantee." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/paral/Input/t24.in" target="_blank">t24.in</a> &hellip;</li><li>seq:  <a class="wikilink abifile-wikilink" data-content="PbTiO3 in the tetragonal geometry. Test berryopt=14 (finite reduced electric field calculation, relaxing cell parameters,  e.g. optcell=2) (XG120616 : tranferred to tests/seq)" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/seq/Input/tsv6_121.in" target="_blank">tsv6_121.in</a>, <a class="wikilink abifile-wikilink" data-content="PbTiO3 in the tetragonal geometry. Test berryopt=16 (finite reduced electric displacement field calculation, relaxing cell parameters,  e.g. optcell=2) (XG120616 : tranferred to tests/seq)" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/seq/Input/tsv6_122.in" target="_blank">tsv6_122.in</a>, <a class="wikilink abifile-wikilink" data-content="PbTiO3 in the tetragonal geometry. Test berryopt=4 (finite electric field calculation,  relaxing cell parameters,  e.g. optcell=2) (XG120616 : tranferred to tests/seq)" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/seq/Input/tsv6_123.in" target="_blank">tsv6_123.in</a> &hellip;</li><li>tutoparal:  <a class="wikilink abifile-wikilink" data-content="Hydronium ion + NH3 molecule Ground state calculation keeping O and H atoms fixed" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/tutoparal/Input/tstring_01.in" target="_blank">tstring_01.in</a> &hellip;</li><li>tutorespfn:  <a class="wikilink abifile-wikilink" data-content="AlAs in hypothetical wurzite (hexagonal) structure Structural optimization run" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/tutorespfn/Input/telast_1.in" target="_blank">telast_1.in</a>, <a class="wikilink abifile-wikilink" data-content="Al fcc metal - elastic constant calculation" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/tutorespfn/Input/telast_6.in" target="_blank">telast_6.in</a>, <a class="wikilink abifile-wikilink" data-content="Finite difference calculation of the clamped-ion piezoelectric constants of AlAs" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/tutorespfn/Input/tffield_5.in" target="_blank">tffield_5.in</a> &hellip;</li><li>tutorial:  <a class="wikilink abifile-wikilink" data-content="H2 molecule in a big box This file will optimize automatically the interatomic distance" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/tutorial/Input/tbase1_3.in" target="_blank">tbase1_3.in</a>, <a class="wikilink abifile-wikilink" data-content="H2 molecule in a big box This file to optimize the H2 bond length,  compute the associated total energy,  then to compute the total energy of the isolated H atom." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/tutorial/Input/tbase2_1.in" target="_blank">tbase2_1.in</a>, <a class="wikilink abifile-wikilink" data-content="H2 molecule in a big box This file to optimize the H2 bond length,  compute the associated total energy,  then to compute the total energy of the isolated H atom. Here,  a double loop has been used." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/tutorial/Input/tbase2_2.in" target="_blank">tbase2_2.in</a> &hellip;</li><li>v1:  <a class="wikilink abifile-wikilink" data-content="Mo surface,  with 10 atom per cell. Not very large,  but the number of bands make it a medium-size problem : 36 bands." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v1/Input/t41.in" target="_blank">t41.in</a>, <a class="wikilink abifile-wikilink" data-content="Test the option ionmov=4. 8-atom Si cube,  non spin-polarized. Same test as number 22a of fast,  except for the change of ionmov,  no reading of wavefunctions,  iscf to 2,  and different tolerance criteria." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v1/Input/t44.in" target="_blank">t44.in</a>, <a class="wikilink abifile-wikilink" data-content="Test the option ionmov=4. Si2 molecule,  spin-polarized. Same test as number 16 of fast,  except for the change of ionmov,  no reading of wavefunctions,  iscf to 2,  and different tolerance criteria. Cannot be executed in parallel: --- !ERROR message: | the number of bands in the spin up case must be equal to the number of bands in the spin down case. This is not the case for the k point number : 1 The number of bands spin up and down are : 5 3 Action : change nband,  or use the sequential version of ABINIT. src_file: chkinp.F90 src_line: 1181 ..." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v1/Input/t45.in" target="_blank">t45.in</a> &hellip;</li><li>v2:  <a class="wikilink abifile-wikilink" data-content="Arsenic in rhombohedral structure. Optimization of unit cell size and shape. With 4x4x4 k point grid,  ecut 3.0,  occopt 4 and tsmear 0.06d0, one get acell 3*7.325  angdeg 3*57.36 and xred 0.2277 These input parameters are too small,  but still give a realistic geometry,  since with the much better parameters 12x12x12 k point grid,  ecut 12.0,  occopt 4 and tsmear 0.04d0, one gets acell 3*7.633  angdeg 3*54.95 and xred 0.2296,  while the experimental values are acell 3*7.751   angdeg 3*54.554 and xred 0.2276" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v2/Input/t44.in" target="_blank">t44.in</a>, <a class="wikilink abifile-wikilink" data-content="H2 molecule in a big box. Comparison of the modified Broyden algorithm (ionmov=3) with the original one (ionmov=2). Start with different values of xcart,  from 0.6 to 1.1,  by step of 0.1 . The number of Broyden steps needed to reach acceptable residual forces with the ionmov=3 algorithm are : 3,  2,  2,  3,  3,  4,  while with the ionmov=2 algorithms,  one get : 4,  3,  2,  5,  and then, either the algorithm does not converge within 8 steps,  or it converges to a saddle point of the energy ! This test was hard to make portable. This is why the tolerance for fldiff is very large." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v2/Input/t48.in" target="_blank">t48.in</a>, <a class="wikilink abifile-wikilink" data-content="Si2 molecule,  static,  spin-polarized. Same system as test 17 of fast,  except lower ecut. Test ionmov=6 (Verlet) as well as ionmov=7. Cannote be executed in parallel: --- !ERROR message: | the number of bands in the spin up case must be equal to the number of bands in the spin down case. This is not the case for the k point number : 1 The number of bands spin up and down are : 5 3 Action : change nband,  or use the sequential version of ABINIT. src_file: chkinp.F90 src_line: 1181" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v2/Input/t49.in" target="_blank">t49.in</a> &hellip;</li><li>v3:  <a class="wikilink abifile-wikilink" data-content="8 atoms of Si in an elongated box. 4 of them are fixed,  while atoms 5 and 6 and 7 and 8 are grouped by pair,  such as the sum of their relative separations along y is constrained to 8.0 . This is a crazy constraint,  just to show that it can be done !" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v3/Input/t40.in" target="_blank">t40.in</a>, <a class="wikilink abifile-wikilink" data-content="Magnesium HCP. Optimize the cell volume and shape using different constraints. Checks whether the stresses are symmetric,  and whether the optimisation algorithm does not break the symmetry." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v3/Input/t42.in" target="_blank">t42.in</a>, <a class="wikilink abifile-wikilink" data-content="H2 molecule : examine the rotational freedom. The present test produces the following vibrational frequencies (with degeneracies indicated): 56.89 i cm-1 (2) 0.41   cm-1 (2) 1.05   cm-1 3800    cm-1 The large frequency corresponds to the stretching mode,  and has the right order of magnitude. The frequencies close to 1 cm-1 corresponds to translation modes,  and are small enough for usual applications. The  56.89 i cm-1 mode corresponds to rotation of the H2 molecule. The magnitude of this frequency might seem quite large. Here are the results of tests made to understand this phenomenon. First,  note that ecut 12  acell 3*14 Increasing the value of ecut to 25 decreases the magnitude of the frequency to 36.8 cm-1 . However,  in order to continue to make it smaller, the cell size must be increased,  and an oscillatory behaviour is observed : 3*16   45.6 i cm-1 3*18   22.7   cm-1 3*20   19.1 i cm-1 3*22   15.7 i cm-1 3*24   13.7   cm-1 Many other tests have been set up. In particular, it was observed that the frequency of the oscillatory behaviour changes with the ecut,  and also that using the Gamma point,  instead of the 1/4 1/4 1/4 k point (used in this test) degrades the convergence. The overall picture is as follows. There are different reasons for the translation and rotation modes to acquire a non-zero frequency when plane waves and supercells are used. Still,  as concerns translations,  only the existence of a discretization of the XC grid is important. For rotations,  supercell effects are also present : - alignement of dipole or quadrupoles - interaction between tails of wavefunctions,  accross cells Since the convergence in supercell size is oscillatory,  we infer that the breaking of the rotational invariance is mostly due to interaction between wavefunction tails. This will be checked by confining the system in a spherical well,  in a forthcoming test." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v3/Input/t80.in" target="_blank">t80.in</a> &hellip;</li><li>v4:  <a class="wikilink abifile-wikilink" data-content="1. Si diamond. Usual geometry,  k points,  psps. One of the two atoms is displaced in order to have non-zeroforces. Test whether energy and forces (and soon stresses) are correctly given,  even with use of spherical harmonics. Perform first computation with the use of Legendre polynomials; Then perform a computation with the use of spherical harmonics. Both must give the same results." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v4/Input/t01.in" target="_blank">t01.in</a>, <a class="wikilink abifile-wikilink" data-content="Ca in orthorhombic structure. 1 k point,  low ecut. Perform a test of GS total energy computation,  using an experimental PAW pseudopotential (pspcod=7)." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v4/Input/t04.in" target="_blank">t04.in</a>, <a class="wikilink abifile-wikilink" data-content="O in orthorhombic structure. 1 k point,  low ecut. Experimental PAW pseudopotential. The aim of this test is to verify that different values of input variable 'ortalg' give the same results for total energy. 'ortalg' is use in the conjugate gradient algorithm which solves,  in the case of PAW calculations, a generalized eigenproblem (H.|Psi&amp;gt;=Lambda.S|Psi&amp;gt;). Three datasets are perfomed;  they must give the same total energies values (when converged)." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v4/Input/t05.in" target="_blank">t05.in</a> &hellip;</li><li>v5:  <a class="wikilink abifile-wikilink" data-content="This is a test,  in the norm conserving case,  of the Molecular Dynamics flag ionmov==13. With the values of optcell=0,  1,  2 (isothermal (0),  isothermal/isenthalpic with homogeneous cell deformations (1) or full cell deformations (2).) This is the reversible integrator of Martyna,  Tuckerman,  Tobias and Klein. (Mol. Phys.,  1996,  p 1117). XG20090912 : Why is the result of dataset 1 and dataset 4 equivalent,  while the bmass varies by a factor of 2 ?" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v5/Input/t01.in" target="_blank">t01.in</a>, <a class="wikilink abifile-wikilink" data-content="Test of geometry optimization using using ionmov=1, 4, 5 and 7 Hydrogen molecule inside a box of 12 Bohr This test was created to test ionmov=5" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v5/Input/t02.in" target="_blank">t02.in</a>, <a class="wikilink abifile-wikilink" data-content="uses ionmov 14 : a 14-step time integration by Blanes and Moan." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v5/Input/t03.in" target="_blank">t03.in</a> &hellip;</li><li>v6:  <a class="wikilink abifile-wikilink" data-content="Hydrogen diatomic molecule in a cell,  close to BCC. One hydrogen atom is at the origin. The other lies initially along the 1 1 1 direction,  at a distance close to the one of the free dimer distance. The final position is the mirror with respect to the plane perpendicular to x,  passing at 1/2 0 0 . The cell parameter is less than twice the H2 interatomic distance,  so that the transition path come close to the x axis,  although it never reaches it (the closest being in the mirror plane). Test the images : 6 images,  exploring the transition path by a simple algorithm. MG: Does it work in parallel. I tried 2 CPUs and the run get stuck!" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v6/Input/t22.in" target="_blank">t22.in</a>, <a class="wikilink abifile-wikilink" data-content="Delocalized internals test. Si diamond bulk with displaced atom Unconverged in all variables" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v6/Input/t23.in" target="_blank">t23.in</a>, <a class="wikilink abifile-wikilink" data-content="Hydrure of Lithium-Boron,  in a face-centered monoclinic cell. Test the recognition of the space group (C2/c (# 15)). Was lacking a case before Dec 2011." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v6/Input/t30.in" target="_blank">t30.in</a> &hellip;</li><li>v7:  <a class="wikilink abifile-wikilink" data-content="Test the extrapolation of then wave functions from one structural relaxation (time) step to the other: extrapwf=1 WARNING: this feature is stil experimental and probably buggy. Si,  Bulk,  2 atoms" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v7/Input/t09.in" target="_blank">t09.in</a>, <a class="wikilink abifile-wikilink" data-content="Nickel ferromagnetic fcc structure with an ONCVPSP pseudo containing valence density. We test several improvements that are available when the NC file contains the PS valence charge. More specifically,  we test the initialization of the density from atomic quantities (initro.F90), the correction to the forces due to the residuals and the extrapolation of the density for structural relaxations (well,  xred is fixed by symmetry). We also test the treatment of the core charge in G-space in the case of GS calculations By default,  abinit handles XCCC in real space,  here we use nc_xccc_gspace==1 to treat it in G-space using the same approach as the one used in PAW." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v7/Input/t10.in" target="_blank">t10.in</a>, <a class="wikilink abifile-wikilink" data-content="Boron,  32 atoms - Thomas-Fermi method (without orbitals) Test the temperature-dependent XC functional (ixc=50) (Ichimaru et al Phys. Rep. 149,  91-205 (1987) Test the von Weizsacker gradient correction (tfkinfunc=11 or 12)" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v7/Input/t11.in" target="_blank">t11.in</a> &hellip;</li><li>v8:  <a class="wikilink abifile-wikilink" data-content="Geometry optimization with LBFGS" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v8/Input/t02.in" target="_blank">t02.in</a>, <a class="wikilink abifile-wikilink" data-content="Simple test of velocity verlet integrator" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v8/Input/t12.in" target="_blank">t12.in</a>, <a class="wikilink abifile-wikilink" data-content="LiH molecule test the use of spatial chemical potential. The molecule starts aligned along the x direction, while the Li chemical potential pulls the Li nucleus. The z coordinate of the molecule is changed from 0 to 5 Bohr. There is a linear part to the chemical potential. The energy and force should be the same in datasets 11 to 15 (without chemical potentiel). One sees that this is not completely the case. The force along z vanishes in these cases,  because the translational invariance is imposed by ABINIT. In datasets 21 to 23,  the energy should increase linearly,  and the force along z should be non-zero and constant. Small fluctuations are observed. Dataset 24 and 25 corresponds to the cubic part of the chemical potential. Dataset 26 places the molecule at a symmetric value of z. The total energy should be the same as with dataset 11,  16,  or 21." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v8/Input/t30.in" target="_blank">t30.in</a> &hellip;</li><li>vdwxc:  <a class="wikilink abifile-wikilink" data-content="Argon FCC. Test the vdW-DFT-D2 (Grimme) semi-empirical potential." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/vdwxc/Input/t10.in" target="_blank">t10.in</a> &hellip;</li></ul></details><p>Choice of algorithm to control the displacements of ions, and eventually (see
<a class="wikilink codevar-wikilink" href="../rlx#optcell">optcell</a>) changes of cell shape and size.</p>
<ul>
<li>
<p>0=&gt; Do not move ions;   </p>
</li>
<li>
<p>1=&gt; Move atoms using molecular dynamics with optional viscous damping (friction linearly proportional to velocity). The viscous damping is controlled by the parameter &ldquo;<a class="wikilink codevar-wikilink" href="../rlx#vis">vis</a>&rdquo;. If actual undamped molecular dynamics is desired, set <a class="wikilink codevar-wikilink" href="../rlx#vis">vis</a> to 0. The implemented algorithm is the generalisation of the Numerov technique (6<sup>th</sup> order), but is NOT invariant upon time-reversal, so that the energy is not conserved. The value <a class="wikilink codevar-wikilink" href="../rlx#ionmov">ionmov</a>=6 will usually be preferred, although the algorithm that is implemented is lower-order. The time step is governed by <a class="wikilink codevar-wikilink" href="../rlx#dtion">dtion</a>. <br />
** Purpose: ** Molecular dynamics (if <a class="wikilink codevar-wikilink" href="../rlx#vis">vis</a>=0), Structural optimization (if <a class="wikilink codevar-wikilink" href="../rlx#vis">vis</a>&gt;0) <br />
** Cell optimization: ** No (Use <a class="wikilink codevar-wikilink" href="../rlx#optcell">optcell</a>=0 only) <br />
** Related variables: ** Viscous parameter <a class="wikilink codevar-wikilink" href="../rlx#vis">vis</a>, time step <a class="wikilink codevar-wikilink" href="../rlx#dtion">dtion</a>, index of atoms fixed <a class="wikilink codevar-wikilink" href="../rlx#iatfix">iatfix</a>   </p>
</li>
<li>
<p>2=&gt; Conduct structural optimization using the Broyden-Fletcher-Goldfarb-Shanno minimization (BFGS). This is much more efficient for structural optimization than viscous damping, when there are less than about 10 degrees of freedom to optimize. <br />
** Purpose: ** Structural optimization <br />
** Cell optimization: ** Yes (if <a class="wikilink codevar-wikilink" href="../rlx#optcell">optcell</a>/=0) <br />
** Related variables: **   </p>
</li>
<li>
<p>3=&gt; Conduct structural optimization using the Broyden-Fletcher-Goldfarb-Shanno minimization (BFGS), modified to take into account the total energy as well as the gradients (as in usual BFGS). <br />
See the paper by [Schlegel, J. Comp. Chem. 3, 214 (1982)]. Might be better
than <a class="wikilink codevar-wikilink" href="../rlx#ionmov">ionmov</a>=2 for few degrees of freedom (less than 3 or 4). Can be very
unstable - use with caution!<br />
** Purpose: ** Structural optimization <br />
** Cell optimization: ** Yes (if <a class="wikilink codevar-wikilink" href="../rlx#optcell">optcell</a>/=0) <br />
** Related variables: **   </p>
</li>
<li>
<p>4=&gt; Conjugate gradient algorithm for simultaneous optimization of potential and ionic degrees of freedom. It can be used with <a class="wikilink codevar-wikilink" href="../basic#iscf">iscf</a>=2 and <a class="wikilink codevar-wikilink" href="../basic#iscf">iscf</a> =5 or 6 (WARNING : this is under development, and does not work very well in many cases). <br />
** Purpose: ** Structural optimization <br />
** Cell optimization: ** No (Use <a class="wikilink codevar-wikilink" href="../rlx#optcell">optcell</a>=0 only) <br />
** Related variables: **   </p>
</li>
<li>
<p>5=&gt; Simple relaxation of ionic positions according to (converged) forces. Equivalent to <a class="wikilink codevar-wikilink" href="../rlx#ionmov">ionmov</a>=1 with zero masses, albeit the relaxation coefficient is not <a class="wikilink codevar-wikilink" href="../rlx#vis">vis</a>, but <a class="wikilink codevar-wikilink" href="../dev#iprcfc">iprcfc</a>. <br />
** Purpose: ** Structural optimization <br />
** Cell optimization: ** No (Use <a class="wikilink codevar-wikilink" href="../rlx#optcell">optcell</a>=0 only) <br />
** Related variables: **   </p>
</li>
<li>
<p>6=&gt; Molecular dynamics using the Verlet algorithm, see [Allen &amp; Tildesley &ldquo;Computer simulation of liquids&rdquo; 1987, p 81]. The only related parameter is the time step (<a class="wikilink codevar-wikilink" href="../rlx#dtion">dtion</a>). <br />
** Purpose: ** Molecular dynamics <br />
** Cell optimization: ** No (Use <a class="wikilink codevar-wikilink" href="../rlx#optcell">optcell</a>=0 only) <br />
** Related variables: ** time step <a class="wikilink codevar-wikilink" href="../rlx#dtion">dtion</a>, index of atoms fixed <a class="wikilink codevar-wikilink" href="../rlx#iatfix">iatfix</a>   </p>
</li>
<li>
<p>7=&gt; Quenched Molecular dynamics using the Verlet algorithm, and stopping each atom for which the scalar product of velocity and force is negative. The only related parameter is the time step (<a class="wikilink codevar-wikilink" href="../rlx#dtion">dtion</a>). The goal is not to produce a realistic dynamics, but to go as fast as possible to the minimum. For this purpose, it is advised to set all the masses to the same value (for example, use the Carbon mass, i.e. set <a class="wikilink codevar-wikilink" href="../rlx#amu">amu</a> to 12 for all type of atoms). <br />
** Purpose: ** Structural optimization <br />
** Cell optimization: ** No (Use <a class="wikilink codevar-wikilink" href="../rlx#optcell">optcell</a>=0 only) <br />
** Related variables: ** time step <a class="wikilink codevar-wikilink" href="../rlx#dtion">dtion</a>, index of atoms fixed <a class="wikilink codevar-wikilink" href="../rlx#iatfix">iatfix</a>   </p>
</li>
<li>
<p>8=&gt; Molecular dynamics with Nose-Hoover thermostat, using the Verlet algorithm. <br />
** Purpose: ** Molecular dynamics <br />
** Cell optimization: ** No (Use <a class="wikilink codevar-wikilink" href="../rlx#optcell">optcell</a>=0 only) <br />
** Related variables: ** time step (<a class="wikilink codevar-wikilink" href="../rlx#dtion">dtion</a>), Temperatures (<a class="wikilink codevar-wikilink" href="../rlx#mdtemp">mdtemp</a>), and thermostat mass (<a class="wikilink codevar-wikilink" href="../rlx#noseinert">noseinert</a>).   </p>
</li>
<li>
<p>9=&gt; Langevin molecular dynamics. <br />
** Purpose: ** Molecular dynamics <br />
** Cell optimization: ** No (Use <a class="wikilink codevar-wikilink" href="../rlx#optcell">optcell</a>=0 only) <br />
** Related variables: ** time step (<a class="wikilink codevar-wikilink" href="../rlx#dtion">dtion</a>), temperatures (<a class="wikilink codevar-wikilink" href="../rlx#mdtemp">mdtemp</a>) and friction coefficient (<a class="wikilink codevar-wikilink" href="../rlx#friction">friction</a>).   </p>
</li>
<li>
<p>10=&gt; Delocalized internal coordinates. with BFGS simple <br />
** Purpose: ** Structural optimization <br />
** Cell optimization: ** No (Use <a class="wikilink codevar-wikilink" href="../rlx#optcell">optcell</a>=0 only) <br />
** Related variables: **   </p>
</li>
<li>
<p>11=&gt; Delocalized internal coordinates. with BFGS using total energy <br />
** Purpose: ** Structural optimization <br />
** Cell optimization: ** No (Use <a class="wikilink codevar-wikilink" href="../rlx#optcell">optcell</a>=0 only) <br />
** Related variables: **   </p>
</li>
<li>
<p>12=&gt; Isokinetic ensemble molecular dynamics. The equation of motion of the ions in contact with a thermostat are solved with the algorithm proposed by Zhang [J. Chem. Phys. 106, 6102 (1997)], as worked out by Minary et al [J. Chem. Phys. 188, 2510 (2003)]. The conservation of the kinetic energy is obtained within machine precision, at each step. <br />
~~ Related parameters : the time step (<a class="wikilink codevar-wikilink" href="../rlx#dtion">dtion</a>), the temperatures
(<a class="wikilink codevar-wikilink" href="../rlx#mdtemp">mdtemp</a>), and the friction coefficient (<a class="wikilink codevar-wikilink" href="../rlx#friction">friction</a>). ~~<br />
** Purpose: ** Molecular dynamics <br />
** Cell optimization: ** No (Use <a class="wikilink codevar-wikilink" href="../rlx#optcell">optcell</a>=0 only) <br />
** Related variables: **   </p>
</li>
<li>
<p>13=&gt; Isothermal/isenthalpic ensemble. The equation of motion of the ions in contact with a thermostat and a barostat are solved with the algorithm proposed by Martyna, Tuckermann Tobias and Klein [Mol. Phys., 1996, p. 1117]. <br />
If optcell=1 or 2, the mass of the barostat (<a class="wikilink codevar-wikilink" href="../rlx#bmass">bmass</a>) must be given in
addition.<br />
** Purpose: ** Molecular dynamics <br />
** Cell optimization: ** Yes (if <a class="wikilink codevar-wikilink" href="../rlx#optcell">optcell</a>/=0) <br />
** Related variables: ** The time step (<a class="wikilink codevar-wikilink" href="../rlx#dtion">dtion</a>), the temperatures (<a class="wikilink codevar-wikilink" href="../rlx#mdtemp">mdtemp</a>), the number of thermostats (<a class="wikilink codevar-wikilink" href="../rlx#nnos">nnos</a>), and the masses of thermostats (<a class="wikilink codevar-wikilink" href="../rlx#qmass">qmass</a>).   </p>
</li>
<li>
<p>14=&gt; Simple molecular dynamics with a symplectic algorithm proposed by S.Blanes and P.C.Moans [called SRKNa14 in Practical symplectic partitioned Runge&ndash;Kutta and Runge&ndash;Kutta&ndash;Nystrom methods, Journal of Computational and Applied Mathematics archive, volume 142, issue 2 (May 2002), pages 313 - 330] of the kind first published by H. Yoshida [Construction of higher order symplectic integrators, Physics Letters A, volume 150, number 5 to 7, pages 262 - 268]. This algorithm requires at least 14 evaluation of the forces (actually 15 are done within Abinit) per time step. At this cost it usually gives much better energy conservation than the verlet algorithm (<a class="wikilink codevar-wikilink" href="../rlx#ionmov">ionmov</a> 6) for a 30 times bigger value of <a class="wikilink codevar-wikilink" href="../rlx#dtion">dtion</a>. Notice that the potential energy of the initial atomic configuration is never evaluated using this algorithm. <br />
** Purpose: ** Molecular dynamics <br />
** Cell optimization: ** No (Use <a class="wikilink codevar-wikilink" href="../rlx#optcell">optcell</a>=0 only) <br />
** Related variables: **   </p>
</li>
<li>
<p>20=&gt; Direct inversion of the iterative subspace. Given a starting point <a class="wikilink codevar-wikilink" href="../basic#xred">xred</a> that is a vector of length 3*<a class="wikilink codevar-wikilink" href="../basic#natom">natom</a> (reduced nuclei coordinates), and unit cell parameters (<a class="wikilink codevar-wikilink" href="../basic#rprimd">%rprimd</a>) this routine uses the DIIS (direct inversion of the iterative subspace) to minimize the gradient (forces) on atoms. The preconditioning used to compute errors from gradients is using an inverse hessian matrix obtained by a BFGS algorithm. This method is known to converge to the nearest point where gradients vanish. This is efficient to refine positions around a saddle point for instance. <br />
** Purpose: ** Structural optimization <br />
** Cell optimization: ** No (Use <a class="wikilink codevar-wikilink" href="../rlx#optcell">optcell</a>=0 only) <br />
** Related variables: ** DIIS memory <a class="wikilink codevar-wikilink" href="../rlx#diismemory">diismemory</a>   </p>
</li>
<li>
<p>22=&gt; Conduct structural optimization using the Limited-memory Broyden-Fletcher-Goldfarb-Shanno minimization (L-BFGS). The working routines were based on the original implementation of J. Nocera available on netlib.org. This algorithm can be much better than the native implementation of BFGS in ABINIT (<a class="wikilink codevar-wikilink" href="../rlx#ionmov">ionmov</a>=2) when one approaches convergence, perhaps because of better treatment of numerical details.<br />
** Purpose: ** Structural optimization <br />
** Cell optimization: ** Yes (if <a class="wikilink codevar-wikilink" href="../rlx#optcell">optcell</a>/=0) <br />
** Related variables: **   </p>
</li>
<li>
<p>23=&gt; Use of Learn on The Fly method (LOTF) for Molecular Dynamics. In the framework of isokinetic MD, the atomic forces and positions are computed by using LOTF interpolation. A SCF computation is performed only any <a class="wikilink codevar-wikilink" href="../dev#lotf_nitex">lotf_nitex</a> steps. The results of the SCF are used to compute the parameters of a short range classical potential (for the moment only the glue potential for gold is implemented). Then these parameters are continuously tuned to compute atomic trajectories. LOTF has to be enabled at configure time. If LOTF is not enabled and <a class="wikilink codevar-wikilink" href="../rlx#ionmov">ionmov</a>=23, abinit will set automatically <a class="wikilink codevar-wikilink" href="../rlx#ionmov">ionmov</a>=12. <br />
The LOTF cycle is divided in the following steps:<br />
a) Initialization (SFC at t=0) and computation of potential parameters.<br />
b) Extrapolation of the atomic forces and positions for <a class="wikilink codevar-wikilink" href="../dev#lotf_nitex">lotf_nitex</a> time
step. To perform this extrapolation, the potential computed in a) is used
(Verlet algorithm).<br />
c) SFC at t=<a class="wikilink codevar-wikilink" href="../dev#lotf_nitex">lotf_nitex</a>. Computation of the potential parameters.<br />
d) LOTF interpolation, linear interpolation of the potential parameters and
computation of the atomic forces and positions between t=0 and t=lotf_nitex.  </p>
</li>
</ul>
<p>** Purpose: ** Molecular Dynamics <br />
** Cell optimization: ** No (Use <a class="wikilink codevar-wikilink" href="../rlx#optcell">optcell</a>=0 only) <br />
** Related variables: ** <a class="wikilink codevar-wikilink" href="../rlx#dtion">dtion</a>, <a class="wikilink codevar-wikilink" href="../dev#lotf_classic">lotf_classic</a>, <a class="wikilink codevar-wikilink" href="../dev#lotf_nitex">lotf_nitex</a>, <a class="wikilink codevar-wikilink" href="../dev#lotf_nneigx">lotf_nneigx</a>, <a class="wikilink codevar-wikilink" href="../dev#lotf_version">lotf_version</a>.   </p>
<ul>
<li>
<p>24=&gt; Simple constant energy molecular dynamics using the velocity Verlet symplectic algorithm (second order), see e.g. [E. Hairer et al. Acta Numerica. 12, 399 (2003)]. The only related parameter is the time step (<a class="wikilink codevar-wikilink" href="../rlx#dtion">dtion</a>). <br />
** Purpose: ** Molecular dynamics <br />
** Cell optimization: ** No (Use <a class="wikilink codevar-wikilink" href="../rlx#optcell">optcell</a>=0 only) <br />
** Related variables: ** time step <a class="wikilink codevar-wikilink" href="../rlx#dtion">dtion</a>   </p>
</li>
<li>
<p>25=&gt; Hybrid Monte Carlo sampling of the ionic positions at fixed temperature and unit cell geometry (NVT ensemle). The underlying molecular dynamics corresponds to ionmov=24. The related parameters are the time step (<a class="wikilink codevar-wikilink" href="../rlx#dtion">dtion</a>) and thermostat temperature (<a class="wikilink codevar-wikilink" href="../rlx#mdtemp">mdtemp</a>). <br />
** Purpose: ** Monte Carlo sampling <br />
** Cell optimization: ** No (Use <a class="wikilink codevar-wikilink" href="../rlx#optcell">optcell</a>=0 only) <br />
** Related variables: ** time step <a class="wikilink codevar-wikilink" href="../rlx#dtion">dtion</a>, thermostat temperature <a class="wikilink codevar-wikilink" href="../rlx#mdtemp">mdtemp</a>,   </p>
</li>
<li>
<p>30=&gt; Using a supercell, calculate a self consistent phonon structure as in PRL 100 095901 (2008). The initial phonon eigenvectors and eigenvalues are read in, and then atoms are displaced according to the normal modes populated at a given temperature until convergence of the vibrational free energy (or so I hope) <br />
** Purpose: ** Phonon structure <br />
** Cell optimization: ** No (Use <a class="wikilink codevar-wikilink" href="../rlx#optcell">optcell</a>=0 only) <br />
** Related variables: **   </p>
</li>
</ul>
<p>No meaning for RF calculations.</p>
<h2 id="istatimg"><strong>istatimg</strong><a class="headerlink" href="#istatimg" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> Integer governing the computation of STATic IMaGes<br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-content="How to perform a PIMD calculation" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../topics/PIMD">topic_PIMD</a><br />
<em>Variable type:</em> integer<br />
<em>Dimensions:</em> scalar<br />
<em>Default value:</em> 1  </p>
<details class="note"><summary>Test list (click to open). Rarely used, [2/920] in all abinit tests, [0/116] in abinit tutorials</summary><ul><li>v6:  <a class="wikilink abifile-wikilink" data-content="Hydrogen diatomic molecule in a cell,  close to BCC Test the string method : 6 images,  exploring the transition path. Similar to test 22 (but with string)." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v6/Input/t24.in" target="_blank">t24.in</a>, <a class="wikilink abifile-wikilink" data-content="Hydrogen diatomic molecule in a cell,  close to BCC Test the string method : 6 images,  exploring the transition path but keeping X coordinated fixed. Similar to test 24" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v6/Input/t25.in" target="_blank">t25.in</a></li></ul></details><p>This input variable is relevant when sets of images are activated (see
<a class="wikilink codevar-wikilink" href="../rlx#imgmov">imgmov</a>).<br />
Not all images might be required to evolve from one time step to the other
(see<a class="wikilink codevar-wikilink" href="../rlx#dynimage">dynimage</a>): these are static images.<br />
If <a class="wikilink codevar-wikilink" href="../rlx#istatimg">istatimg</a>=0, the total energy of static images is not computed (but
static images are used to make the dynamic images evolve). This can be useful
to save CPU time.<br />
If <a class="wikilink codevar-wikilink" href="../rlx#istatimg">istatimg</a>=1, the total energy of static images is computed.</p>
<h2 id="mdtemp"><strong>mdtemp</strong><a class="headerlink" href="#mdtemp" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> Molecular Dynamics TEMPeratures<br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-content="How to perform a PIMD calculation" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../topics/PIMD">topic_PIMD</a>, <a class="wikilink topic-wikilink" data-content="How to perform a molecular dynamics calculation" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../topics/MolecularDynamics">topic_MolecularDynamics</a><br />
<em>Variable type:</em> real<br />
<em>Dimensions:</em> (2)<br />
<em>Default value:</em> [300, 300]  </p>
<details class="note"><summary>Test list (click to open). Rarely used, [9/920] in all abinit tests, [0/116] in abinit tutorials</summary><ul><li>mpiio:  <a class="wikilink abifile-wikilink" data-content="He FCC solid in conventional cell (4 atoms). Test the recursion algorithm (for high-temperature calculations). Only with 0 and 4 procs." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/mpiio/Input/t42.in" target="_blank">t42.in</a></li><li>paral:  <a class="wikilink abifile-wikilink" data-content="Gold,  8 atom supercell,  with PAW. Test lotf_nitex,  lotf_classic,  lotf_version,  lotf_nneigx. These variables,  concerning LOTF method,  are taken into account only if the enable_lotf=&amp;quot;yes&amp;quot; is used in configuration." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/paral/Input/t41.in" target="_blank">t41.in</a></li><li>v2:  <a class="wikilink abifile-wikilink" data-content="Ge liquid. Test of Nose and Langevin dynamics. 2 atoms in a cell. Allows 10 time steps." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v2/Input/t87.in" target="_blank">t87.in</a>, <a class="wikilink abifile-wikilink" data-content="Basic test of geometry optimization for water First computing the ground state of a close configuration Uses 5 different methods to find the relaxed positions" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v2/Input/t88.in" target="_blank">t88.in</a></li><li>v5:  <a class="wikilink abifile-wikilink" data-content="This is a test,  in the norm conserving case,  of the Molecular Dynamics flag ionmov==13. With the values of optcell=0,  1,  2 (isothermal (0),  isothermal/isenthalpic with homogeneous cell deformations (1) or full cell deformations (2).) This is the reversible integrator of Martyna,  Tuckerman,  Tobias and Klein. (Mol. Phys.,  1996,  p 1117). XG20090912 : Why is the result of dataset 1 and dataset 4 equivalent,  while the bmass varies by a factor of 2 ?" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v5/Input/t01.in" target="_blank">t01.in</a>, <a class="wikilink abifile-wikilink" data-content="Simple test for double grid (fine/coarse) system,  the cutting radius and the initial fermi energy in recursion method Corresponding variables: recgratio,  recrcut, recefermi respectively Corresponding code: 68_recursion Other test: paral/tT, v5/t76" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v5/Input/t75.in" target="_blank">t75.in</a>, <a class="wikilink abifile-wikilink" data-content="Test for an electron gas by recursion method (Vtrial is set to zero) Corresponding variable: rectesteg Corresponding code: 68_recursion Other test: paral/tT, v5/t76" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v5/Input/t76.in" target="_blank">t76.in</a></li><li>v7:  <a class="wikilink abifile-wikilink" data-content="Hydrogen molecule: Test the Path-Integral Molecular Dynamics (PIMD) implementation in (N, V, T) ensemble. THe two thermostats (Nose-Hoover chains and Langevin) are tested together with the three coordinate systems (primitive,  normal mode,  staging). NOTE: MPI run with 2 procs is ok but fldiff comparison fails due to an extra section with --- Pseudopotential description --" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v7/Input/t08.in" target="_blank">t08.in</a></li><li>v8:  <a class="wikilink abifile-wikilink" data-content="H2 molecule: PIMD simulation with Langevin thermostat. Test of restart with HIST.nc file. Test of linear constraint." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v8/Input/t05.in" target="_blank">t05.in</a></li></ul></details><p>Give the initial and final temperature of the Nose-Hoover thermostat
(<a class="wikilink codevar-wikilink" href="../rlx#ionmov">ionmov</a>=8) and Langevin dynamics (<a class="wikilink codevar-wikilink" href="../rlx#ionmov">ionmov</a>=9), in Kelvin. This
temperature will change linearly from the initial temperature ** mdtemp(1) **
at itime=1 to the final temperature ** mdtemp(2) ** at the end of the
<a class="wikilink codevar-wikilink" href="../rlx#ntime">ntime</a> timesteps.</p>
<h2 id="mdwall"><strong>mdwall</strong><a class="headerlink" href="#mdwall" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> Molecular Dynamics WALL location<br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-content="How to perform a molecular dynamics calculation" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../topics/MolecularDynamics">topic_MolecularDynamics</a><br />
<em>Variable type:</em> real<br />
<em>Dimensions:</em> scalar<br />
<em>Default value:</em> 10000.0<br />
<em>Comment:</em> the walls are extremely far away  </p>
<details class="note"><summary>Test list (click to open). Rarely used, [1/920] in all abinit tests, [0/116] in abinit tutorials</summary><ul><li>v2:  <a class="wikilink abifile-wikilink" data-content="Ge liquid. Test of Nose and Langevin dynamics. 2 atoms in a cell. Allows 10 time steps." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v2/Input/t87.in" target="_blank">t87.in</a></li></ul></details><p>Gives the location (atomic units) of walls on which the atoms will bounce
back. when <a class="wikilink codevar-wikilink" href="../rlx#ionmov">ionmov</a>=6, 7, 8 or 9. For each cartesian direction idir=1, 2 or
3, there is a pair of walls with coordinates xcart(idir)=-wall and
xcart(idir)=rprimd(idir,idir)+wall . Supposing the particle will cross the
wall, its velocity normal to the wall is reversed, so that it bounces back.<br />
By default, given in Bohr atomic units (1 Bohr=0.5291772108 Angstroms),
although Angstrom can be specified, if preferred, since <a class="wikilink codevar-wikilink" href="../rlx#mdwall">mdwall</a> has the
&lsquo;<a class="wikilink" href="../../user-guide/abinit#32-more-about-abinit-input-variables">LENGTH</a>&lsquo; characteristics.</p>
<h2 id="mep_mxstep"><strong>mep_mxstep</strong><a class="headerlink" href="#mep_mxstep" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> Minimal Energy Path search: MaXimum allowed STEP size<br />
<em>Characteristics:</em> <a class="wikilink" href="../../user-guide/abinit#32-more-about-abinit-input-variables">LENGTH</a><br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-content="How to calculate transition paths" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../topics/TransPath">topic_TransPath</a><br />
<em>Variable type:</em> real<br />
<em>Dimensions:</em> scalar<br />
<em>Default value:</em> 0.4 if <a class="wikilink codevar-wikilink" href="../rlx#imgmov">imgmov</a>==5,
100.0 otherwise.</p>
<details class="note"><summary>Test list (click to open). Rarely used, [1/920] in all abinit tests, [0/116] in abinit tutorials</summary><ul><li>v6:  <a class="wikilink abifile-wikilink" data-content="Hydrogen diatomic molecule in a cell,  close to BCC Test the string method : 6 images,  exploring the transition path. Similar to test 22 (but with string)." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v6/Input/t24.in" target="_blank">t24.in</a></li></ul></details><p>Relevant only when <a class="wikilink codevar-wikilink" href="../rlx#imgmov">imgmov</a>=1 (Steepest-Descent), 2 (String Method) or 5
(Nudged Elastic Band).<br />
The optimizer used to solve the Ordinary Differential Equation (ODE) can be
constrained with a maximum allowed step size for each image. By default this
feature is only activated for Nudged Elastic Band (NEB) and the value is
inspired by _ J. Chem. Phys. 128, 134106 (2008) _ .<br />
Note that the step size is defined for each image as _ step = SQRT[SUM(R_i dot
R_i)] _ where the _ R_i _ are the positions of the atoms in the cell.</p>
<h2 id="mep_solver"><strong>mep_solver</strong><a class="headerlink" href="#mep_solver" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> Minimal Energy Path ordinary differential equation SOLVER<br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-content="How to calculate transition paths" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../topics/TransPath">topic_TransPath</a><br />
<em>Variable type:</em> integer<br />
<em>Dimensions:</em> scalar<br />
<em>Default value:</em> None  </p>
<details class="note"><summary>Test list (click to open). Rarely used, [1/920] in all abinit tests, [0/116] in abinit tutorials</summary><ul><li>v6:  <a class="wikilink abifile-wikilink" data-content="Hydrogen diatomic molecule in a cell,  close to BCC Test the string method : 6 images,  exploring the transition path. Similar to test 22 (but with string)." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v6/Input/t24.in" target="_blank">t24.in</a></li></ul></details><p>Relevant only when <a class="wikilink codevar-wikilink" href="../rlx#imgmov">imgmov</a>=2 (String Method) or 5 (Nudged Elastic Band).<br />
Gives the algorithm used to solve the Ordinary Differential Equation (ODE)
when searching for a Minimal Energy Path (MEP).<br />
Possible values can be:  </p>
<ul>
<li>
<p>0=&gt; ** Steepest-Descent algorithm ** following the (scaled) forces, the scaling factor being <a class="wikilink codevar-wikilink" href="../rlx#fxcartfactor">fxcartfactor</a> (forward Euler method). <br />
Compatible with all MEP search methods.</p>
</li>
<li>
<p>1=&gt; ** Quick-min optimizer ** following the (scaled) forces, the scaling factor being <a class="wikilink codevar-wikilink" href="../rlx#fxcartfactor">fxcartfactor</a>. The &ldquo;quick minimizer&rdquo; improves upon the steepest-descent method by accelerating the system in the direction of the forces. The velocity (of the image) is projected long the force and cancelled if antiparallel to it. <br />
Compatible only with Nudged Elastic Band (<a class="wikilink codevar-wikilink" href="../rlx#imgmov">imgmov</a>=5).<br />
_ See, for instance: J. Chem. Phys. 128, 134106 (2008). _  </p>
</li>
<li>
<p>2=&gt; ** Local Broyden-Fletcher-Goldfarb-Shanno (L-BFGS) algorithm ** ; each image along the band is minimized with a different instance of the BFGS optimizer. <br />
Compatible only with Nudged Elastic Band (<a class="wikilink codevar-wikilink" href="../rlx#imgmov">imgmov</a>=5).<br />
_ See, for instance: J. Chem. Phys. 128, 134106 (2008). _<br />
IN <a class="wikilink" href="../../user-guide/abinit#32-more-about-abinit-input-variables">DEVELOP</a>PMENT - NOT RELIABLE</p>
</li>
<li>
<p>3=&gt; ** Global Broyden-Fletcher-Goldfarb-Shanno (GL-BFGS) algorithm ** ; all images along the band are minimized with a single instance of the BFGS optimizer. <br />
Compatible only with Nudged Elastic Band (<a class="wikilink codevar-wikilink" href="../rlx#imgmov">imgmov</a>=5).<br />
_ See, for instance: J. Chem. Phys. 128, 134106 (2008). _<br />
IN <a class="wikilink" href="../../user-guide/abinit#32-more-about-abinit-input-variables">DEVELOP</a>PMENT - NOT RELIABLE</p>
</li>
<li>
<p>4=&gt; ** Fourth-order Runge-Kutta method ** ; the images along the band are moved every four steps (1&lt;=istep&lt;=<a class="wikilink codevar-wikilink" href="../rlx#ntimimage">ntimimage</a>) following the Runge-Kutta algorithm, the time step being <a class="wikilink codevar-wikilink" href="../rlx#fxcartfactor">fxcartfactor</a>. <br />
Compatible only with Simplified String Method (<a class="wikilink codevar-wikilink" href="../rlx#imgmov">imgmov</a>=2 and
<a class="wikilink codevar-wikilink" href="../rlx#string_algo">string_algo</a>=1 or 2).<br />
_ See: J. Chem. Phys. 126, 164103 (2007). _</p>
</li>
</ul>
<p>All of the optimizers can be constrained with a maximum allowed step size for
each image; see <a class="wikilink codevar-wikilink" href="../rlx#mep_mxstep">mep_mxstep</a>. This is by default the case of the Nudged
Elastic Band (<a class="wikilink codevar-wikilink" href="../rlx#imgmov">imgmov</a>=5).</p>
<h2 id="natcon"><strong>natcon</strong><a class="headerlink" href="#natcon" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> Number of AToms in CONstraint equations<br />
<em>Characteristics:</em> <a class="wikilink" href="../../user-guide/abinit#32-more-about-abinit-input-variables">NO_MULTI</a>, <a class="wikilink" href="../../user-guide/abinit#32-more-about-abinit-input-variables">INPUT_ONLY</a><br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-content="How to constaint the geometry of the system in geometry optimization, molecular
dynamics or searches" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../topics/GeoConstraints">topic_GeoConstraints</a><br />
<em>Variable type:</em> integer<br />
<em>Dimensions:</em> (<a class="wikilink codevar-wikilink" href="../rlx#nconeq">nconeq</a>)<br />
<em>Default value:</em> 0  </p>
<details class="note"><summary>Test list (click to open). Rarely used, [3/920] in all abinit tests, [0/116] in abinit tutorials</summary><ul><li>v3:  <a class="wikilink abifile-wikilink" data-content="8 atoms of Si in an elongated box. 4 of them are fixed,  while atoms 5 and 6 and 7 and 8 are grouped by pair,  such as the sum of their relative separations along y is constrained to 8.0 . This is a crazy constraint,  just to show that it can be done !" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v3/Input/t40.in" target="_blank">t40.in</a></li><li>v6:  <a class="wikilink abifile-wikilink" data-content="Hydrogen diatomic molecule in a cell,  close to BCC. One hydrogen atom is at the origin. The other lies initially along the 1 1 1 direction,  at a distance close to the one of the free dimer distance. The final position is the mirror with respect to the plane perpendicular to x,  passing at 1/2 0 0 . The cell parameter is less than twice the H2 interatomic distance,  so that the transition path come close to the x axis,  although it never reaches it (the closest being in the mirror plane). Test the images : 6 images,  exploring the transition path by a simple algorithm. MG: Does it work in parallel. I tried 2 CPUs and the run get stuck!" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v6/Input/t22.in" target="_blank">t22.in</a></li><li>v8:  <a class="wikilink abifile-wikilink" data-content="H2 molecule: PIMD simulation with Langevin thermostat. Test of restart with HIST.nc file. Test of linear constraint." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v8/Input/t05.in" target="_blank">t05.in</a></li></ul></details><p>Gives the number of atoms appearing in each of the <a class="wikilink codevar-wikilink" href="../rlx#nconeq">nconeq</a> independent
equations constraining the motion of atoms during structural optimization or
molecular dynamics (see <a class="wikilink codevar-wikilink" href="../rlx#nconeq">nconeq</a> , <a class="wikilink codevar-wikilink" href="../rlx#iatcon">iatcon</a>, and <a class="wikilink codevar-wikilink" href="../rlx#wtatcon">wtatcon</a>).</p>
<h2 id="natfix"><strong>natfix</strong><a class="headerlink" href="#natfix" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> Number of Atoms that are FIXed<br />
<em>Characteristics:</em> <a class="wikilink" href="../../user-guide/abinit#32-more-about-abinit-input-variables">INPUT_ONLY</a><br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-content="How to constaint the geometry of the system in geometry optimization, molecular
dynamics or searches" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../topics/GeoConstraints">topic_GeoConstraints</a><br />
<em>Variable type:</em> integer<br />
<em>Dimensions:</em> scalar<br />
<em>Default value:</em> 0<br />
<em>Comment:</em> (no atoms held fixed)  </p>
<details class="note"><summary>Test list (click to open). Moderately used, [17/920] in all abinit tests, [2/116] in abinit tutorials</summary><ul><li>fast:  <a class="wikilink abifile-wikilink" data-content="8-atom Si cube to set up brdmin and MD runs with natfix. Output also DEN,  POT and GEO files" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/fast/Input/t27.in" target="_blank">t27.in</a>, <a class="wikilink abifile-wikilink" data-content="brdmin with natfix. Output also DEN,  POT and GEO files" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/fast/Input/t28.in" target="_blank">t28.in</a>, <a class="wikilink abifile-wikilink" data-content="move with natfix. Output also GEO files" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/fast/Input/t29.in" target="_blank">t29.in</a></li><li>paral:  <a class="wikilink abifile-wikilink" data-content="Test the string method within parallelization over images Inspired by test v6#22. Hydrogen diatomic molecule in a cell,  close to BCC 7 images,  exploring the transition path. Three datasets,  testing each value of prtvolimg. Processors distribution automatically determined:" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/paral/Input/t08.in" target="_blank">t08.in</a>, <a class="wikilink abifile-wikilink" data-content="Test the string method within parallelization over images Inspired by test v6#22. Hydrogen diatomic molecule in a cell,  close to BCC 7 images,  exploring the transition path. Three datasets,  testing each value of prtvolimg. Processors distribution automatically determined:" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/paral/Input/t08.in" target="_blank">t08.in</a>, <a class="wikilink abifile-wikilink" data-content="Test the string method within parallelization over images Inspired by test v6#22. Hydrogen diatomic molecule in a cell,  close to BCC 7 images,  exploring the transition path. Three datasets,  testing each value of prtvolimg. Processors distribution automatically determined:" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/paral/Input/t08.in" target="_blank">t08.in</a>, <a class="wikilink abifile-wikilink" data-content="Test the string method within parallelization over images Inspired by test v6#22. Hydrogen diatomic molecule in a cell,  close to BCC 7 images,  exploring the transition path. Three datasets,  testing each value of prtvolimg. Processors distribution automatically determined:" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/paral/Input/t08.in" target="_blank">t08.in</a></li><li>tutoparal:  <a class="wikilink abifile-wikilink" data-content="Hydronium ion + NH3 molecule Ground state calculation keeping O and H atoms fixed" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/tutoparal/Input/tstring_01.in" target="_blank">tstring_01.in</a></li><li>tutorespfn:  <a class="wikilink abifile-wikilink" data-content="AlAs in hypothetical wurzite (hexagonal) structure Structural optimization run" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/tutorespfn/Input/telast_1.in" target="_blank">telast_1.in</a></li><li>v1:  <a class="wikilink abifile-wikilink" data-content="Test the option ionmov=4. 8-atom Si cube,  non spin-polarized. Same test as number 22a of fast,  except for the change of ionmov,  no reading of wavefunctions,  iscf to 2,  and different tolerance criteria." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v1/Input/t44.in" target="_blank">t44.in</a>, <a class="wikilink abifile-wikilink" data-content="SiH4 : optimize the geometry with partially constrained atomic positions. 1 special k-point,  in a box 8x8x8 (too small),  and 8Ha cut-off. Test ionmov=2 Permutes the three axes x, y, z in three datasets. Since the initialization is different,  the three datasets deliver slightly different results,  though quite close to each others. Should be identical at convergence." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v1/Input/t85.in" target="_blank">t85.in</a>, <a class="wikilink abifile-wikilink" data-content="SiH4 : molecular dynamics with partially constrained atomic positions. 1 special k-point,  in a box 8x8x8 (too small),  and 8Ha cut-off. Test ionmov=1" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v1/Input/t86.in" target="_blank">t86.in</a></li><li>v2:  <a class="wikilink abifile-wikilink" data-content="Basic test of geometry optimization for water First computing the ground state of a close configuration Uses 5 different methods to find the relaxed positions" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v2/Input/t88.in" target="_blank">t88.in</a></li><li>v3:  <a class="wikilink abifile-wikilink" data-content="8 atoms of Si in an elongated box. 4 of them are fixed,  while atoms 5 and 6 and 7 and 8 are grouped by pair,  such as the sum of their relative separations along y is constrained to 8.0 . This is a crazy constraint,  just to show that it can be done !" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v3/Input/t40.in" target="_blank">t40.in</a></li><li>v6:  <a class="wikilink abifile-wikilink" data-content="Hydrogen diatomic molecule in a cell,  close to BCC. One hydrogen atom is at the origin. The other lies initially along the 1 1 1 direction,  at a distance close to the one of the free dimer distance. The final position is the mirror with respect to the plane perpendicular to x,  passing at 1/2 0 0 . The cell parameter is less than twice the H2 interatomic distance,  so that the transition path come close to the x axis,  although it never reaches it (the closest being in the mirror plane). Test the images : 6 images,  exploring the transition path by a simple algorithm. MG: Does it work in parallel. I tried 2 CPUs and the run get stuck!" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v6/Input/t22.in" target="_blank">t22.in</a>, <a class="wikilink abifile-wikilink" data-content="Hydrogen diatomic molecule in a cell,  close to BCC Test the string method : 6 images,  exploring the transition path. Similar to test 22 (but with string)." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v6/Input/t24.in" target="_blank">t24.in</a>, <a class="wikilink abifile-wikilink" data-content="Hydrogen diatomic molecule in a cell,  close to BCC Test the string method : 6 images,  exploring the transition path but keeping X coordinated fixed. Similar to test 24" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v6/Input/t25.in" target="_blank">t25.in</a></li></ul></details><p>Gives the number of atoms (not to exceed <a class="wikilink codevar-wikilink" href="../basic#natom">natom</a>) which are to be held fixed
during a structural optimization or molecular dynamics.<br />
When <a class="wikilink codevar-wikilink" href="../rlx#natfix">natfix</a> &gt; 0, <a class="wikilink codevar-wikilink" href="../rlx#natfix">natfix</a> entries should be provided in array
<a class="wikilink codevar-wikilink" href="../rlx#iatfix">iatfix</a> .</p>
<h2 id="natfixx"><strong>natfixx</strong><a class="headerlink" href="#natfixx" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> Number of Atoms that are FIXed along the X direction<br />
<em>Characteristics:</em> <a class="wikilink" href="../../user-guide/abinit#32-more-about-abinit-input-variables">INPUT_ONLY</a><br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-content="How to constaint the geometry of the system in geometry optimization, molecular
dynamics or searches" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../topics/GeoConstraints">topic_GeoConstraints</a><br />
<em>Variable type:</em> integer<br />
<em>Dimensions:</em> scalar<br />
<em>Default value:</em> 0  </p>
<details class="note"><summary>Test list (click to open). Rarely used, [2/920] in all abinit tests, [0/116] in abinit tutorials</summary><ul><li>v1:  <a class="wikilink abifile-wikilink" data-content="SiH4 : optimize the geometry with partially constrained atomic positions. 1 special k-point,  in a box 8x8x8 (too small),  and 8Ha cut-off. Test ionmov=2 Permutes the three axes x, y, z in three datasets. Since the initialization is different,  the three datasets deliver slightly different results,  though quite close to each others. Should be identical at convergence." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v1/Input/t85.in" target="_blank">t85.in</a></li><li>v6:  <a class="wikilink abifile-wikilink" data-content="Hydrogen diatomic molecule in a cell,  close to BCC Test the string method : 6 images,  exploring the transition path but keeping X coordinated fixed. Similar to test 24" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v6/Input/t25.in" target="_blank">t25.in</a></li></ul></details><p>Gives the number of atoms (not to exceed <a class="wikilink codevar-wikilink" href="../basic#natom">natom</a>) which are to be held fixed
along the X direction during a structural optimization or molecular dynamics.<br />
When <a class="wikilink codevar-wikilink" href="../rlx#natfixx">natfixx</a> &gt; 0, <a class="wikilink codevar-wikilink" href="../rlx#natfixx">natfixx</a> entries should be provided in array
<a class="wikilink codevar-wikilink" href="../rlx#iatfixx">iatfixx</a>.</p>
<h2 id="natfixy"><strong>natfixy</strong><a class="headerlink" href="#natfixy" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> Number of Atoms that are FIXed along the Y direction<br />
<em>Characteristics:</em> <a class="wikilink" href="../../user-guide/abinit#32-more-about-abinit-input-variables">INPUT_ONLY</a><br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-content="How to constaint the geometry of the system in geometry optimization, molecular
dynamics or searches" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../topics/GeoConstraints">topic_GeoConstraints</a><br />
<em>Variable type:</em> integer<br />
<em>Dimensions:</em> scalar<br />
<em>Default value:</em> 0  </p>
<details class="note"><summary>Test list (click to open). Rarely used, [2/920] in all abinit tests, [0/116] in abinit tutorials</summary><ul><li>v1:  <a class="wikilink abifile-wikilink" data-content="SiH4 : optimize the geometry with partially constrained atomic positions. 1 special k-point,  in a box 8x8x8 (too small),  and 8Ha cut-off. Test ionmov=2 Permutes the three axes x, y, z in three datasets. Since the initialization is different,  the three datasets deliver slightly different results,  though quite close to each others. Should be identical at convergence." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v1/Input/t85.in" target="_blank">t85.in</a>, <a class="wikilink abifile-wikilink" data-content="SiH4 : molecular dynamics with partially constrained atomic positions. 1 special k-point,  in a box 8x8x8 (too small),  and 8Ha cut-off. Test ionmov=1" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v1/Input/t86.in" target="_blank">t86.in</a></li></ul></details><p>Gives the number of atoms (not to exceed <a class="wikilink codevar-wikilink" href="../basic#natom">natom</a>) which are to be held fixed
along the Y direction during a structural optimization or molecular dynamics.<br />
When <a class="wikilink codevar-wikilink" href="../rlx#natfixy">natfixy</a> &gt; 0, <a class="wikilink codevar-wikilink" href="../rlx#natfixy">natfixy</a> entries should be provided in array
<a class="wikilink codevar-wikilink" href="../rlx#iatfixy">iatfixy</a></p>
<h2 id="natfixz"><strong>natfixz</strong><a class="headerlink" href="#natfixz" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> Number of Atoms that are FIXed along the Z direction<br />
<em>Characteristics:</em> <a class="wikilink" href="../../user-guide/abinit#32-more-about-abinit-input-variables">INPUT_ONLY</a><br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-content="How to constaint the geometry of the system in geometry optimization, molecular
dynamics or searches" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../topics/GeoConstraints">topic_GeoConstraints</a><br />
<em>Variable type:</em> integer<br />
<em>Dimensions:</em> scalar<br />
<em>Default value:</em> 0  </p>
<details class="note"><summary>Test list (click to open). Rarely used, [2/920] in all abinit tests, [0/116] in abinit tutorials</summary><ul><li>v1:  <a class="wikilink abifile-wikilink" data-content="SiH4 : optimize the geometry with partially constrained atomic positions. 1 special k-point,  in a box 8x8x8 (too small),  and 8Ha cut-off. Test ionmov=2 Permutes the three axes x, y, z in three datasets. Since the initialization is different,  the three datasets deliver slightly different results,  though quite close to each others. Should be identical at convergence." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v1/Input/t85.in" target="_blank">t85.in</a>, <a class="wikilink abifile-wikilink" data-content="SiH4 : molecular dynamics with partially constrained atomic positions. 1 special k-point,  in a box 8x8x8 (too small),  and 8Ha cut-off. Test ionmov=1" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v1/Input/t86.in" target="_blank">t86.in</a></li></ul></details><p>Gives the number of atoms (not to exceed <a class="wikilink codevar-wikilink" href="../basic#natom">natom</a>) which are to be held fixed
along the Z direction during a structural optimization or molecular dynamics.<br />
When <a class="wikilink codevar-wikilink" href="../rlx#natfixz">natfixz</a> &gt; 0, <a class="wikilink codevar-wikilink" href="../rlx#natfixz">natfixz</a> entries should be provided in array
<a class="wikilink codevar-wikilink" href="../rlx#iatfixz">iatfixz</a>.</p>
<h2 id="nconeq"><strong>nconeq</strong><a class="headerlink" href="#nconeq" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> Number of CONstraint EQuations<br />
<em>Characteristics:</em> <a class="wikilink" href="../../user-guide/abinit#32-more-about-abinit-input-variables">NO_MULTI</a><br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-content="How to constaint the geometry of the system in geometry optimization, molecular
dynamics or searches" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../topics/GeoConstraints">topic_GeoConstraints</a><br />
<em>Variable type:</em> integer<br />
<em>Dimensions:</em> scalar<br />
<em>Default value:</em> 0  </p>
<details class="note"><summary>Test list (click to open). Rarely used, [3/920] in all abinit tests, [0/116] in abinit tutorials</summary><ul><li>v3:  <a class="wikilink abifile-wikilink" data-content="8 atoms of Si in an elongated box. 4 of them are fixed,  while atoms 5 and 6 and 7 and 8 are grouped by pair,  such as the sum of their relative separations along y is constrained to 8.0 . This is a crazy constraint,  just to show that it can be done !" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v3/Input/t40.in" target="_blank">t40.in</a></li><li>v6:  <a class="wikilink abifile-wikilink" data-content="Hydrogen diatomic molecule in a cell,  close to BCC. One hydrogen atom is at the origin. The other lies initially along the 1 1 1 direction,  at a distance close to the one of the free dimer distance. The final position is the mirror with respect to the plane perpendicular to x,  passing at 1/2 0 0 . The cell parameter is less than twice the H2 interatomic distance,  so that the transition path come close to the x axis,  although it never reaches it (the closest being in the mirror plane). Test the images : 6 images,  exploring the transition path by a simple algorithm. MG: Does it work in parallel. I tried 2 CPUs and the run get stuck!" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v6/Input/t22.in" target="_blank">t22.in</a></li><li>v8:  <a class="wikilink abifile-wikilink" data-content="H2 molecule: PIMD simulation with Langevin thermostat. Test of restart with HIST.nc file. Test of linear constraint." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v8/Input/t05.in" target="_blank">t05.in</a></li></ul></details><p>Gives the number of independent equations constraining the motion of atoms
during structural optimization or molecular dynamics (see <a class="wikilink codevar-wikilink" href="../rlx#natcon">natcon</a> ,
<a class="wikilink codevar-wikilink" href="../rlx#iatcon">iatcon</a>, and <a class="wikilink codevar-wikilink" href="../rlx#wtatcon">wtatcon</a>).</p>
<h2 id="neb_algo"><strong>neb_algo</strong><a class="headerlink" href="#neb_algo" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> Nudged Elastic Band ALGOrithm<br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-content="How to calculate transition paths" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../topics/TransPath">topic_TransPath</a><br />
<em>Variable type:</em> integer<br />
<em>Dimensions:</em> scalar<br />
<em>Default value:</em> 1<br />
<em>Only relevant if:</em> <a class="wikilink codevar-wikilink" href="../rlx#imgmov">imgmov</a>==5  </p>
<details class="note"><summary>Test list (click to open). Rarely used, [1/920] in all abinit tests, [0/116] in abinit tutorials</summary><ul><li>v6:  <a class="wikilink abifile-wikilink" data-content="Hydrogen diatomic molecule in a cell,  close to BCC Test the string method : 6 images,  exploring the transition path. Similar to test 22 (but with string)." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v6/Input/t24.in" target="_blank">t24.in</a></li></ul></details><p>Gives the variant of the NEB method used.<br />
Possible values can be:  </p>
<ul>
<li>
<p>0=&gt; ** Original NEB method ** . <br />
_ See: Classical and Quantum Dynamics in Condensed Phase Simulations, edited
by Berne, Ciccotti, Coker (World Scientific, Singapore, 1998), pp. 385-404 _</p>
</li>
<li>
<p>1=&gt; ** NEB + improved tangent ** . <br />
The Improved Tangent Method builds on the NEB with an improved estimate of the
tangent direction and a resulting change of the component of the spring force
acting on the images.<br />
_ See: J. Chem. Phys. 113, 9978 (2000). _</p>
</li>
<li>
<p>2=&gt; ** Climbing-Image NEB (CI-NEB) ** . <br />
The CI-NEB method constitutes a small modification to the NEB method.
Information about the shape of the MEP is retained, but a rigorous convergence
to a saddle point is also obtained. By default the spring constants are
variable (see <a class="wikilink codevar-wikilink" href="../rlx#neb_spring">neb_spring</a>). As the image with the highest energy has to be
identified, the calculation begins with several iterations of the standard NEB
algorithm. The effective CI-NEB begins at the <a class="wikilink codevar-wikilink" href="../rlx#cineb_start">cineb_start</a> iteration.<br />
_ See: J. Chem. Phys. 113, 9901 (2000). _</p>
</li>
</ul>
<p>Note that, in all cases, it is possible to define the value of the spring
constant connecting images with <a class="wikilink codevar-wikilink" href="../rlx#neb_spring">neb_spring</a>, keeping it constant or
allowing it to vary between 2 values (to have higher resolution close to the
saddle point).</p>
<h2 id="neb_spring"><strong>neb_spring</strong><a class="headerlink" href="#neb_spring" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> Nudged Elastic Band: SPRING constant<br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-content="How to calculate transition paths" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../topics/TransPath">topic_TransPath</a><br />
<em>Variable type:</em> real<br />
<em>Dimensions:</em> (2)<br />
<em>Default value:</em> [0.02, 0.15] if <a class="wikilink codevar-wikilink" href="../rlx#neb_algo">neb_algo</a>==2,
[0.05, 0.05] otherwise.</p>
<p><em>Only relevant if:</em> <a class="wikilink codevar-wikilink" href="../rlx#imgmov">imgmov</a>==5  </p>
<details class="note"><summary>Test list (click to open). Rarely used, [1/920] in all abinit tests, [0/116] in abinit tutorials</summary><ul><li>v6:  <a class="wikilink abifile-wikilink" data-content="Hydrogen diatomic molecule in a cell,  close to BCC Test the string method : 6 images,  exploring the transition path. Similar to test 22 (but with string)." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v6/Input/t24.in" target="_blank">t24.in</a></li></ul></details><p>Gives the minimal and maximal values of the spring constant connecting images
for the NEB method.<br />
In the standard &ldquo;Nudged Elastic Band&rdquo; method, the spring constant is constant
along the path, but, in order to have higher resolution close to the saddle
point, it can be better to have stronger springs close to it.<br />
_ See: J. Chem. Phys. 113, 9901 (2000). _</p>
<h2 id="nimage"><strong>nimage</strong><a class="headerlink" href="#nimage" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> Number of IMAGEs<br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-content="How to perform a PIMD calculation" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../topics/PIMD">topic_PIMD</a>, <a class="wikilink topic-wikilink" data-content="How to calculate transition paths" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../topics/TransPath">topic_TransPath</a><br />
<em>Variable type:</em> integer<br />
<em>Dimensions:</em> scalar<br />
<em>Default value:</em> 1  </p>
<details class="note"><summary>Test list (click to open). Moderately used, [12/920] in all abinit tests, [0/116] in abinit tutorials</summary><ul><li>paral:  <a class="wikilink abifile-wikilink" data-content="Test the string method within parallelization over images Inspired by test v6#22. Hydrogen diatomic molecule in a cell,  close to BCC 7 images,  exploring the transition path. Three datasets,  testing each value of prtvolimg. Processors distribution automatically determined:" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/paral/Input/t08.in" target="_blank">t08.in</a>, <a class="wikilink abifile-wikilink" data-content="Test the string method within parallelization over images Inspired by test v6#22. Hydrogen diatomic molecule in a cell,  close to BCC 7 images,  exploring the transition path. Three datasets,  testing each value of prtvolimg. Processors distribution automatically determined:" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/paral/Input/t08.in" target="_blank">t08.in</a>, <a class="wikilink abifile-wikilink" data-content="Test the string method within parallelization over images Inspired by test v6#22. Hydrogen diatomic molecule in a cell,  close to BCC 7 images,  exploring the transition path. Three datasets,  testing each value of prtvolimg. Processors distribution automatically determined:" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/paral/Input/t08.in" target="_blank">t08.in</a>, <a class="wikilink abifile-wikilink" data-content="Test the string method within parallelization over images Inspired by test v6#22. Hydrogen diatomic molecule in a cell,  close to BCC 7 images,  exploring the transition path. Three datasets,  testing each value of prtvolimg. Processors distribution automatically determined:" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/paral/Input/t08.in" target="_blank">t08.in</a></li><li>v6:  <a class="wikilink abifile-wikilink" data-content="Bismuth,  rhombohedral,  two atoms per unit cell Scalar relativistic. Relax four different images,  using the steepest descent algorithm." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v6/Input/t21.in" target="_blank">t21.in</a>, <a class="wikilink abifile-wikilink" data-content="Hydrogen diatomic molecule in a cell,  close to BCC. One hydrogen atom is at the origin. The other lies initially along the 1 1 1 direction,  at a distance close to the one of the free dimer distance. The final position is the mirror with respect to the plane perpendicular to x,  passing at 1/2 0 0 . The cell parameter is less than twice the H2 interatomic distance,  so that the transition path come close to the x axis,  although it never reaches it (the closest being in the mirror plane). Test the images : 6 images,  exploring the transition path by a simple algorithm. MG: Does it work in parallel. I tried 2 CPUs and the run get stuck!" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v6/Input/t22.in" target="_blank">t22.in</a>, <a class="wikilink abifile-wikilink" data-content="Hydrogen diatomic molecule in a cell,  close to BCC Test the string method : 6 images,  exploring the transition path. Similar to test 22 (but with string)." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v6/Input/t24.in" target="_blank">t24.in</a>, <a class="wikilink abifile-wikilink" data-content="Hydrogen diatomic molecule in a cell,  close to BCC Test the string method : 6 images,  exploring the transition path but keeping X coordinated fixed. Similar to test 24" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v6/Input/t25.in" target="_blank">t25.in</a>, <a class="wikilink abifile-wikilink" data-content="Genetic algorithm structure random search. Hydrogen has been used as an example. No physical meaning. Number of images is only 10 but experience shows that at least 20 are fine." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v6/Input/t27.in" target="_blank">t27.in</a>, <a class="wikilink abifile-wikilink" data-content="Ga Al(1-x) Asx Test of alchemical calculations with images. Mostly checking the treatment of input variables and their echo. (Tests 30-36 are more slots for testing the ground state)" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v6/Input/t28.in" target="_blank">t28.in</a></li><li>v7:  <a class="wikilink abifile-wikilink" data-content="Hydrogen molecule: Test the Path-Integral Molecular Dynamics (PIMD) implementation in (N, V, T) ensemble. THe two thermostats (Nose-Hoover chains and Langevin) are tested together with the three coordinate systems (primitive,  normal mode,  staging). NOTE: MPI run with 2 procs is ok but fldiff comparison fails due to an extra section with --- Pseudopotential description --" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v7/Input/t08.in" target="_blank">t08.in</a></li><li>v8:  <a class="wikilink abifile-wikilink" data-content="H2 molecule: PIMD simulation with Langevin thermostat. Test of restart with HIST.nc file. Test of linear constraint." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v8/Input/t05.in" target="_blank">t05.in</a></li></ul></details><p>Give the number of images (or replicas) of the system, for which the forces
and stresses might be computed independently, in the context of the string
method, the genetic algorithm, hyperdynamics or Path-Integral Molecular
Dynamics depending on the value of <a class="wikilink codevar-wikilink" href="../rlx#imgmov">imgmov</a>). Related input variables :
<a class="wikilink codevar-wikilink" href="../rlx#dynimage">dynimage</a>, <a class="wikilink codevar-wikilink" href="../paral#npimage">npimage</a>, <a class="wikilink codevar-wikilink" href="../rlx#ntimimage">ntimimage</a> and <a class="wikilink codevar-wikilink" href="../files#prtvolimg">prtvolimg</a>.<br />
Images might differ by the position of atoms in the unit cell, their velocity,
as well as by their cell geometry. The following input variables might be used
to define the images :</p>
<ul>
<li><a class="wikilink codevar-wikilink" href="../basic#acell">acell</a> </li>
<li><a class="wikilink codevar-wikilink" href="../rlx#amu">amu</a> </li>
<li><a class="wikilink codevar-wikilink" href="../basic#angdeg">angdeg</a> </li>
<li><a class="wikilink codevar-wikilink" href="../paw#dmatpawu">dmatpawu</a> </li>
<li><a class="wikilink codevar-wikilink" href="../paw#jpawu">jpawu</a> </li>
<li><a class="wikilink codevar-wikilink" href="../gstate#mixalch">mixalch</a> </li>
<li><a class="wikilink codevar-wikilink" href="../basic#rprim">rprim</a> </li>
<li><a class="wikilink codevar-wikilink" href="../paw#upawu">upawu</a> </li>
<li><a class="wikilink codevar-wikilink" href="../rlx#vel">vel</a> </li>
<li><a class="wikilink codevar-wikilink" href="../rlx#vel_cell">vel_cell</a> </li>
<li><a class="wikilink codevar-wikilink" href="../basic#xangst">xangst</a> </li>
<li><a class="wikilink codevar-wikilink" href="../basic#xcart">xcart</a> </li>
<li><a class="wikilink codevar-wikilink" href="../basic#xred">xred</a> </li>
</ul>
<p>These input variables, non-modified, will be used to define the image with
index 1. For the image with the last index, the input file might specify the
values of such input variables, appended with &ldquo;_lastimg&rdquo;, e.g. :</p>
<ul>
<li>acell_lastimg </li>
<li>rprim_lastimg </li>
<li>xcart_lastimg </li>
<li>&hellip; </li>
</ul>
<p>By default, these values will be interpolated linearly to define values for
the other images, unless there exist specific values for some images, for
which the string &ldquo;last&rdquo; has to be replaced by the index of the image, e.g. for
the image number 4 :</p>
<ul>
<li>acell_4img </li>
<li>rprim_4img </li>
<li>xcart_4img </li>
<li>&hellip; </li>
</ul>
<p>It is notably possible to specify the starting point and the end point of the
path (of images), while specifying intermediate points.  </p>
<p>It usually happen that the images do not have the same symmetries and space
group. ABINIT has not been designed to use different set of symmetries for
different images. ABINIT will use the symmetry and space group of the image
number 2, that is expected to have a low number of symmetries. This might lead
to erroneous calculations, in case some image has even less symmetry. By
contrast, there is no problem if some other image has more symmetries than
those of the second image.</p>
<h2 id="nnos"><strong>nnos</strong><a class="headerlink" href="#nnos" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> Number of NOSe masses<br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-content="How to perform a PIMD calculation" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../topics/PIMD">topic_PIMD</a>, <a class="wikilink topic-wikilink" data-content="How to perform a molecular dynamics calculation" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../topics/MolecularDynamics">topic_MolecularDynamics</a><br />
<em>Variable type:</em> integer<br />
<em>Dimensions:</em> scalar<br />
<em>Default value:</em> 0  </p>
<details class="note"><summary>Test list (click to open). Rarely used, [2/920] in all abinit tests, [0/116] in abinit tutorials</summary><ul><li>v5:  <a class="wikilink abifile-wikilink" data-content="This is a test,  in the norm conserving case,  of the Molecular Dynamics flag ionmov==13. With the values of optcell=0,  1,  2 (isothermal (0),  isothermal/isenthalpic with homogeneous cell deformations (1) or full cell deformations (2).) This is the reversible integrator of Martyna,  Tuckerman,  Tobias and Klein. (Mol. Phys.,  1996,  p 1117). XG20090912 : Why is the result of dataset 1 and dataset 4 equivalent,  while the bmass varies by a factor of 2 ?" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v5/Input/t01.in" target="_blank">t01.in</a></li><li>v7:  <a class="wikilink abifile-wikilink" data-content="Hydrogen molecule: Test the Path-Integral Molecular Dynamics (PIMD) implementation in (N, V, T) ensemble. THe two thermostats (Nose-Hoover chains and Langevin) are tested together with the three coordinate systems (primitive,  normal mode,  staging). NOTE: MPI run with 2 procs is ok but fldiff comparison fails due to an extra section with --- Pseudopotential description --" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v7/Input/t08.in" target="_blank">t08.in</a></li></ul></details><p>Gives the number of thermostats in the Martyna et al. chain of oscillators
thermostats. The thermostat chains can be used either to perform Molecular
Dynamics (MD) (<a class="wikilink codevar-wikilink" href="../rlx#ionmov">ionmov</a>=13) or to perform Path Integral Molecular Dynamics
(PIMD) (<a class="wikilink codevar-wikilink" href="../rlx#imgmov">imgmov</a>=13).<br />
The mass of these thermostats is given by <a class="wikilink codevar-wikilink" href="../rlx#qmass">qmass</a>.  </p>
<h2 id="noseinert"><strong>noseinert</strong><a class="headerlink" href="#noseinert" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> NOSE thermostat INERTia factor<br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-content="How to perform a molecular dynamics calculation" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../topics/MolecularDynamics">topic_MolecularDynamics</a><br />
<em>Variable type:</em> real<br />
<em>Dimensions:</em> scalar<br />
<em>Default value:</em> 100000<br />
<em>Only relevant if:</em> <a class="wikilink codevar-wikilink" href="../rlx#ionmov">ionmov</a>==8  </p>
<details class="note"><summary>Test list (click to open). Rarely used, [1/920] in all abinit tests, [0/116] in abinit tutorials</summary><ul><li>v2:  <a class="wikilink abifile-wikilink" data-content="Ge liquid. Test of Nose and Langevin dynamics. 2 atoms in a cell. Allows 10 time steps." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v2/Input/t87.in" target="_blank">t87.in</a></li></ul></details><p>Give the inertia factor WT of the Nose-Hoover thermostat (when <a class="wikilink codevar-wikilink" href="../rlx#ionmov">ionmov</a>=8),
in atomic units of weight*length2, that is (electron mass)*(Bohr)2. The
equations of motion are : MI d2RI/dt2= FI - dX/dt MI dRI/dt and WT d2X/dt2=
Sum(I) MI (dRI/dt)2 - 3NkBT where I represent each nucleus, MI is the mass of
each nucleus (see <a class="wikilink codevar-wikilink" href="../rlx#amu">amu</a>), RI is the coordinate of each nucleus (see
<a class="wikilink codevar-wikilink" href="../basic#xcart">xcart</a>), dX/dt is a dynamical friction coefficient, and T is the
temperature of the thermostat (see <a class="wikilink codevar-wikilink" href="../rlx#mdtemp">mdtemp</a>).</p>
<h2 id="ntime"><strong>ntime</strong><a class="headerlink" href="#ntime" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> Number of TIME steps<br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-content="How to perform a molecular dynamics calculation" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../topics/MolecularDynamics">topic_MolecularDynamics</a>, <a class="wikilink topic-wikilink" data-content="How to perform a geometry optimization" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../topics/GeoOpt">topic_GeoOpt</a><br />
<em>Variable type:</em> integer<br />
<em>Dimensions:</em> scalar<br />
<em>Default value:</em> 0  </p>
<details class="note"><summary>Test list (click to open). Moderately used, [120/920] in all abinit tests, [19/116] in abinit tutorials</summary><ul><li>bigdft:  <a class="wikilink abifile-wikilink" data-content="H2 molecule with geometry optimisation using BFGS (test the reformating part of the code)." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/bigdft/Input/t10.in" target="_blank">t10.in</a>, <a class="wikilink abifile-wikilink" data-content="H2O in isolated boundary conditions (plane waves,  NC),  try tu use Damiens Poisson cutoff. Water molecule with HGH pseudo-potentials. This checks that a negative value in rhor will not make the Poisson solver crashes on XC computation." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/bigdft/Input/t22.in" target="_blank">t22.in</a> &hellip;</li><li>built-in:  <a class="wikilink abifile-wikilink" data-content="H2 molecule,  determination of the H-H distance by the Broyden algorithm." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/built-in/Input/testin_fast.in" target="_blank">testin_fast.in</a>, <a class="wikilink abifile-wikilink" data-content="Ca atom with PAW,  writing a ETSF_IO wavefunction file." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/built-in/Input/testin_etsf_io.in" target="_blank">testin_etsf_io.in</a> &hellip;</li><li>etsf_io:  <a class="wikilink abifile-wikilink" data-content="Ca atom with PAW,  writing a ETSF_IO wavefunction file. Very quick built-in test,  to check that ETSF_IO is installed." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/etsf_io/Input/t00.in" target="_blank">t00.in</a>, <a class="wikilink abifile-wikilink" data-content="Ca atom in PAW. Test the output of density in PAW formalism with ETSF_IO. Also test the output of the wavefunctions in PAW formalism with ETSF_IO." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/etsf_io/Input/t09.in" target="_blank">t09.in</a>, <a class="wikilink abifile-wikilink" data-content="Si2 molecule,  static,  spin-polarized. Same system as test 49 of v2,  except that nctime is non-zero. Test ionmov=6 (Verlet) as well as ionmov=7." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/etsf_io/Input/t21.in" target="_blank">t21.in</a> &hellip;</li><li>fast:  <a class="wikilink abifile-wikilink" data-content="H2 molecule,  determination of the H-H distance by the Broyden algorithm. Very quick built-in test,  to check that ABINIT is working." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/fast/Input/t00.in" target="_blank">t00.in</a>, <a class="wikilink abifile-wikilink" data-content="H2 diatomic molecule,  static,  to check accurate forces." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/fast/Input/t02.in" target="_blank">t02.in</a>, <a class="wikilink abifile-wikilink" data-content="Conduct relaxation of molecule starting from wf.19,  using ionmov=2. (Broyden method)." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/fast/Input/t20.in" target="_blank">t20.in</a> &hellip;</li><li>gpu:  <a class="wikilink abifile-wikilink" data-content="Silicon - diamond structure - PAW - volume relaxation Test the use of GPU within Projector Augmented-Wave formalism. Test structure relaxation (multiple calls to gstate routine)." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/gpu/Input/t04.in" target="_blank">t04.in</a> &hellip;</li><li>mpiio:  <a class="wikilink abifile-wikilink" data-content="Si,  Bulk,  2 atoms,  paral_kgb. Test of ground state with different occs(7 and 0), and also ionmov 2. Only with 0 and 4 procs." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/mpiio/Input/t21.in" target="_blank">t21.in</a>, <a class="wikilink abifile-wikilink" data-content="C-diamond,  Bulk,  2 atoms,  paral_kgb,  with PAW. Test of ground state with different occs(7 and 0), and also ionmov 2. Only with 0 and 4 procs." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/mpiio/Input/t22.in" target="_blank">t22.in</a>, <a class="wikilink abifile-wikilink" data-content="C-diamond,  Bulk,  2 atoms,  paral_kgb,  with PAW. Test the triple parallelisation. We cannot check the distribution npband*npfft*npkpt=2*2*2=8 processors, this number being not allowed in the test procedure. Here,  we only test the parallelisation over FFT and kpoints: npband*npfft*npkpt=1*2*2=4 processors. In test tY.in we check other distributions for guarantee." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/mpiio/Input/t24.in" target="_blank">t24.in</a> &hellip;</li><li>paral:  <a class="wikilink abifile-wikilink" data-content="Si,  Bulk,  2 atoms,  paral_kgb. Test of ground state with different occs(7 and 0), and also ionmov 2. Only with 0 and 4 procs." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/paral/Input/t21.in" target="_blank">t21.in</a>, <a class="wikilink abifile-wikilink" data-content="C-diamond,  Bulk,  2 atoms,  paral_kgb,  with PAW. Test of ground state with different occs(7 and 0), and also ionmov 2. Only with 0 and 4 procs." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/paral/Input/t22.in" target="_blank">t22.in</a>, <a class="wikilink abifile-wikilink" data-content="C-diamond,  Bulk,  2 atoms,  paral_kgb,  with PAW. Test the triple parallelisation. We cannot check the distribution npband*npfft*npkpt=2*2*2=8 processors, this number being not allowed in the test procedure. Here,  we only test the parallelisation over FFT and kpoints: npband*npfft*npkpt=1*2*2=4 processors. In test tY.in we check other distributions for guarantee." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/paral/Input/t24.in" target="_blank">t24.in</a> &hellip;</li><li>seq:  <a class="wikilink abifile-wikilink" data-content="PbTiO3 in the tetragonal geometry. Test berryopt=14 (finite reduced electric field calculation, relaxing cell parameters,  e.g. optcell=2) (XG120616 : tranferred to tests/seq)" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/seq/Input/tsv6_121.in" target="_blank">tsv6_121.in</a>, <a class="wikilink abifile-wikilink" data-content="PbTiO3 in the tetragonal geometry. Test berryopt=16 (finite reduced electric displacement field calculation, relaxing cell parameters,  e.g. optcell=2) (XG120616 : tranferred to tests/seq)" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/seq/Input/tsv6_122.in" target="_blank">tsv6_122.in</a>, <a class="wikilink abifile-wikilink" data-content="PbTiO3 in the tetragonal geometry. Test berryopt=4 (finite electric field calculation,  relaxing cell parameters,  e.g. optcell=2) (XG120616 : tranferred to tests/seq)" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/seq/Input/tsv6_123.in" target="_blank">tsv6_123.in</a> &hellip;</li><li>tutoparal:  <a class="wikilink abifile-wikilink" data-content="Hydronium ion + NH3 molecule Ground state calculation keeping O and H atoms fixed" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/tutoparal/Input/tstring_01.in" target="_blank">tstring_01.in</a> &hellip;</li><li>tutorespfn:  <a class="wikilink abifile-wikilink" data-content="AlAs in hypothetical wurzite (hexagonal) structure Structural optimization run" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/tutorespfn/Input/telast_1.in" target="_blank">telast_1.in</a>, <a class="wikilink abifile-wikilink" data-content="Al fcc metal - elastic constant calculation" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/tutorespfn/Input/telast_6.in" target="_blank">telast_6.in</a>, <a class="wikilink abifile-wikilink" data-content="Finite difference calculation of the clamped-ion piezoelectric constants of AlAs" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/tutorespfn/Input/tffield_5.in" target="_blank">tffield_5.in</a> &hellip;</li><li>tutorial:  <a class="wikilink abifile-wikilink" data-content="H2 molecule in a big box This file will optimize automatically the interatomic distance" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/tutorial/Input/tbase1_3.in" target="_blank">tbase1_3.in</a>, <a class="wikilink abifile-wikilink" data-content="H2 molecule in a big box This file to optimize the H2 bond length,  compute the associated total energy,  then to compute the total energy of the isolated H atom." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/tutorial/Input/tbase2_1.in" target="_blank">tbase2_1.in</a>, <a class="wikilink abifile-wikilink" data-content="H2 molecule in a big box This file to optimize the H2 bond length,  compute the associated total energy,  then to compute the total energy of the isolated H atom. Here,  a double loop has been used." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/tutorial/Input/tbase2_2.in" target="_blank">tbase2_2.in</a> &hellip;</li><li>v1:  <a class="wikilink abifile-wikilink" data-content="Test the option ionmov=4. 8-atom Si cube,  non spin-polarized. Same test as number 22a of fast,  except for the change of ionmov,  no reading of wavefunctions,  iscf to 2,  and different tolerance criteria." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v1/Input/t44.in" target="_blank">t44.in</a>, <a class="wikilink abifile-wikilink" data-content="Test the option ionmov=4. Si2 molecule,  spin-polarized. Same test as number 16 of fast,  except for the change of ionmov,  no reading of wavefunctions,  iscf to 2,  and different tolerance criteria. Cannot be executed in parallel: --- !ERROR message: | the number of bands in the spin up case must be equal to the number of bands in the spin down case. This is not the case for the k point number : 1 The number of bands spin up and down are : 5 3 Action : change nband,  or use the sequential version of ABINIT. src_file: chkinp.F90 src_line: 1181 ..." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v1/Input/t45.in" target="_blank">t45.in</a>, <a class="wikilink abifile-wikilink" data-content="Si2 molecule,  spin-polarized. Take the wavefunctions from test 45, and converge them further for initialisation of test 47. Cannot be executed in parallel: --- !ERROR message: | the number of bands in the spin up case must be equal to the number of bands in the spin down case. This is not the case for the k point number : 1 The number of bands spin up and down are : 5 3 Action : change nband,  or use the sequential version of ABINIT. src_file: chkinp.F90 src_line: 1181 ..." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v1/Input/t46.in" target="_blank">t46.in</a> &hellip;</li><li>v2:  <a class="wikilink abifile-wikilink" data-content="Arsenic in rhombohedral structure. Optimization of unit cell size and shape. With 4x4x4 k point grid,  ecut 3.0,  occopt 4 and tsmear 0.06d0, one get acell 3*7.325  angdeg 3*57.36 and xred 0.2277 These input parameters are too small,  but still give a realistic geometry,  since with the much better parameters 12x12x12 k point grid,  ecut 12.0,  occopt 4 and tsmear 0.04d0, one gets acell 3*7.633  angdeg 3*54.95 and xred 0.2296,  while the experimental values are acell 3*7.751   angdeg 3*54.554 and xred 0.2276" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v2/Input/t44.in" target="_blank">t44.in</a>, <a class="wikilink abifile-wikilink" data-content="H2 molecule in a big box. Comparison of the modified Broyden algorithm (ionmov=3) with the original one (ionmov=2). Start with different values of xcart,  from 0.6 to 1.1,  by step of 0.1 . The number of Broyden steps needed to reach acceptable residual forces with the ionmov=3 algorithm are : 3,  2,  2,  3,  3,  4,  while with the ionmov=2 algorithms,  one get : 4,  3,  2,  5,  and then, either the algorithm does not converge within 8 steps,  or it converges to a saddle point of the energy ! This test was hard to make portable. This is why the tolerance for fldiff is very large." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v2/Input/t48.in" target="_blank">t48.in</a>, <a class="wikilink abifile-wikilink" data-content="Si2 molecule,  static,  spin-polarized. Same system as test 17 of fast,  except lower ecut. Test ionmov=6 (Verlet) as well as ionmov=7. Cannote be executed in parallel: --- !ERROR message: | the number of bands in the spin up case must be equal to the number of bands in the spin down case. This is not the case for the k point number : 1 The number of bands spin up and down are : 5 3 Action : change nband,  or use the sequential version of ABINIT. src_file: chkinp.F90 src_line: 1181" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v2/Input/t49.in" target="_blank">t49.in</a> &hellip;</li><li>v3:  <a class="wikilink abifile-wikilink" data-content="BCC Tantalum (1 atoms per unit cell), using the HGH pseudopotential,  within LDA. Fixed cell dimensions: 2 slightly different cell sizes. Use only one k point. Test whether the stress is correctly given,  even with spin-orbit coupling. Difference of total energies : 0.0000588977 Ha Difference of unit cell volume : 0.05861 Bohr^3 Stress from finite difference : 1.0049 Ha/Bohr^3 Average stress : 1.0062 Ha/Bohr^3" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v3/Input/t01.in" target="_blank">t01.in</a>, <a class="wikilink abifile-wikilink" data-content="8 atoms of Si in an elongated box. 4 of them are fixed,  while atoms 5 and 6 and 7 and 8 are grouped by pair,  such as the sum of their relative separations along y is constrained to 8.0 . This is a crazy constraint,  just to show that it can be done !" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v3/Input/t40.in" target="_blank">t40.in</a>, <a class="wikilink abifile-wikilink" data-content="Magnesium HCP. Optimize the cell volume and shape using different constraints. Checks whether the stresses are symmetric,  and whether the optimisation algorithm does not break the symmetry." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v3/Input/t42.in" target="_blank">t42.in</a> &hellip;</li><li>v4:  <a class="wikilink abifile-wikilink" data-content="1. Si diamond. Usual geometry,  k points,  psps. One of the two atoms is displaced in order to have non-zeroforces. Test whether energy and forces (and soon stresses) are correctly given,  even with use of spherical harmonics. Perform first computation with the use of Legendre polynomials; Then perform a computation with the use of spherical harmonics. Both must give the same results." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v4/Input/t01.in" target="_blank">t01.in</a>, <a class="wikilink abifile-wikilink" data-content="Ca in orthorhombic structure. 1 k point,  low ecut. Perform a test of GS total energy computation,  using an experimental PAW pseudopotential (pspcod=7)." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v4/Input/t04.in" target="_blank">t04.in</a>, <a class="wikilink abifile-wikilink" data-content="O in orthorhombic structure. 1 k point,  low ecut. Experimental PAW pseudopotential. The aim of this test is to verify that different values of input variable 'ortalg' give the same results for total energy. 'ortalg' is use in the conjugate gradient algorithm which solves,  in the case of PAW calculations, a generalized eigenproblem (H.|Psi&amp;gt;=Lambda.S|Psi&amp;gt;). Three datasets are perfomed;  they must give the same total energies values (when converged)." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v4/Input/t05.in" target="_blank">t05.in</a> &hellip;</li><li>v5:  <a class="wikilink abifile-wikilink" data-content="This is a test,  in the norm conserving case,  of the Molecular Dynamics flag ionmov==13. With the values of optcell=0,  1,  2 (isothermal (0),  isothermal/isenthalpic with homogeneous cell deformations (1) or full cell deformations (2).) This is the reversible integrator of Martyna,  Tuckerman,  Tobias and Klein. (Mol. Phys.,  1996,  p 1117). XG20090912 : Why is the result of dataset 1 and dataset 4 equivalent,  while the bmass varies by a factor of 2 ?" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v5/Input/t01.in" target="_blank">t01.in</a>, <a class="wikilink abifile-wikilink" data-content="Test of geometry optimization using using ionmov=1, 4, 5 and 7 Hydrogen molecule inside a box of 12 Bohr This test was created to test ionmov=5" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v5/Input/t02.in" target="_blank">t02.in</a>, <a class="wikilink abifile-wikilink" data-content="uses ionmov 14 : a 14-step time integration by Blanes and Moan." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v5/Input/t03.in" target="_blank">t03.in</a> &hellip;</li><li>v6:  <a class="wikilink abifile-wikilink" data-content="Hydrogen diatomic molecule in a cell,  close to BCC. One hydrogen atom is at the origin. The other lies initially along the 1 1 1 direction,  at a distance close to the one of the free dimer distance. The final position is the mirror with respect to the plane perpendicular to x,  passing at 1/2 0 0 . The cell parameter is less than twice the H2 interatomic distance,  so that the transition path come close to the x axis,  although it never reaches it (the closest being in the mirror plane). Test the images : 6 images,  exploring the transition path by a simple algorithm. MG: Does it work in parallel. I tried 2 CPUs and the run get stuck!" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v6/Input/t22.in" target="_blank">t22.in</a>, <a class="wikilink abifile-wikilink" data-content="Delocalized internals test. Si diamond bulk with displaced atom Unconverged in all variables" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v6/Input/t23.in" target="_blank">t23.in</a>, <a class="wikilink abifile-wikilink" data-content="Genetic algorithm structure random search. Hydrogen has been used as an example. No physical meaning. Number of images is only 10 but experience shows that at least 20 are fine." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v6/Input/t27.in" target="_blank">t27.in</a> &hellip;</li><li>v7:  <a class="wikilink abifile-wikilink" data-content="Test the extrapolation of then wave functions from one structural relaxation (time) step to the other: extrapwf=1 WARNING: this feature is stil experimental and probably buggy. Si,  Bulk,  2 atoms" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v7/Input/t09.in" target="_blank">t09.in</a>, <a class="wikilink abifile-wikilink" data-content="Nickel ferromagnetic fcc structure with an ONCVPSP pseudo containing valence density. We test several improvements that are available when the NC file contains the PS valence charge. More specifically,  we test the initialization of the density from atomic quantities (initro.F90), the correction to the forces due to the residuals and the extrapolation of the density for structural relaxations (well,  xred is fixed by symmetry). We also test the treatment of the core charge in G-space in the case of GS calculations By default,  abinit handles XCCC in real space,  here we use nc_xccc_gspace==1 to treat it in G-space using the same approach as the one used in PAW." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v7/Input/t10.in" target="_blank">t10.in</a>, <a class="wikilink abifile-wikilink" data-content="Boron,  32 atoms - Thomas-Fermi method (without orbitals) Test the temperature-dependent XC functional (ixc=50) (Ichimaru et al Phys. Rep. 149,  91-205 (1987) Test the von Weizsacker gradient correction (tfkinfunc=11 or 12)" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v7/Input/t11.in" target="_blank">t11.in</a> &hellip;</li><li>v8:  <a class="wikilink abifile-wikilink" data-content="Geometry optimization with LBFGS" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v8/Input/t02.in" target="_blank">t02.in</a>, <a class="wikilink abifile-wikilink" data-content="Simple test of velocity verlet integrator" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v8/Input/t12.in" target="_blank">t12.in</a>, <a class="wikilink abifile-wikilink" data-content="LiH molecule test the use of spatial chemical potential. The molecule starts aligned along the x direction, while the Li chemical potential pulls the Li nucleus. The z coordinate of the molecule is changed from 0 to 5 Bohr. There is a linear part to the chemical potential. The energy and force should be the same in datasets 11 to 15 (without chemical potentiel). One sees that this is not completely the case. The force along z vanishes in these cases,  because the translational invariance is imposed by ABINIT. In datasets 21 to 23,  the energy should increase linearly,  and the force along z should be non-zero and constant. Small fluctuations are observed. Dataset 24 and 25 corresponds to the cubic part of the chemical potential. Dataset 26 places the molecule at a symmetric value of z. The total energy should be the same as with dataset 11,  16,  or 21." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v8/Input/t30.in" target="_blank">t30.in</a> &hellip;</li><li>vdwxc:  <a class="wikilink abifile-wikilink" data-content="Argon FCC. Test the vdW-DFT-D2 (Grimme) semi-empirical potential." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/vdwxc/Input/t10.in" target="_blank">t10.in</a> &hellip;</li></ul></details><p>Gives the number of molecular dynamics time steps or Broyden structural
optimization steps to be done if <a class="wikilink codevar-wikilink" href="../rlx#ionmov">ionmov</a> is non-zero.<br />
Note that at the present the option <a class="wikilink codevar-wikilink" href="../rlx#ionmov">ionmov</a>=1 is initialized with four
Runge-Kutta steps which costs some overhead in the startup. By contrast, the
initialisation of other <a class="wikilink codevar-wikilink" href="../rlx#ionmov">ionmov</a> values is only one SCF call.<br />
<a class="wikilink codevar-wikilink" href="../rlx#ntime">ntime</a> is ignored if <a class="wikilink codevar-wikilink" href="../rlx#ionmov">ionmov</a>=0.</p>
<h2 id="ntimimage"><strong>ntimimage</strong><a class="headerlink" href="#ntimimage" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> Number of TIME steps for IMAGE propagation<br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-content="How to perform a PIMD calculation" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../topics/PIMD">topic_PIMD</a>, <a class="wikilink topic-wikilink" data-content="How to calculate transition paths" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../topics/TransPath">topic_TransPath</a><br />
<em>Variable type:</em> integer<br />
<em>Dimensions:</em> scalar<br />
<em>Default value:</em> 1  </p>
<details class="note"><summary>Test list (click to open). Moderately used, [11/920] in all abinit tests, [0/116] in abinit tutorials</summary><ul><li>paral:  <a class="wikilink abifile-wikilink" data-content="Test the string method within parallelization over images Inspired by test v6#22. Hydrogen diatomic molecule in a cell,  close to BCC 7 images,  exploring the transition path. Three datasets,  testing each value of prtvolimg. Processors distribution automatically determined:" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/paral/Input/t08.in" target="_blank">t08.in</a>, <a class="wikilink abifile-wikilink" data-content="Test the string method within parallelization over images Inspired by test v6#22. Hydrogen diatomic molecule in a cell,  close to BCC 7 images,  exploring the transition path. Three datasets,  testing each value of prtvolimg. Processors distribution automatically determined:" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/paral/Input/t08.in" target="_blank">t08.in</a>, <a class="wikilink abifile-wikilink" data-content="Test the string method within parallelization over images Inspired by test v6#22. Hydrogen diatomic molecule in a cell,  close to BCC 7 images,  exploring the transition path. Three datasets,  testing each value of prtvolimg. Processors distribution automatically determined:" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/paral/Input/t08.in" target="_blank">t08.in</a>, <a class="wikilink abifile-wikilink" data-content="Test the string method within parallelization over images Inspired by test v6#22. Hydrogen diatomic molecule in a cell,  close to BCC 7 images,  exploring the transition path. Three datasets,  testing each value of prtvolimg. Processors distribution automatically determined:" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/paral/Input/t08.in" target="_blank">t08.in</a></li><li>v6:  <a class="wikilink abifile-wikilink" data-content="Bismuth,  rhombohedral,  two atoms per unit cell Scalar relativistic. Relax four different images,  using the steepest descent algorithm." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v6/Input/t21.in" target="_blank">t21.in</a>, <a class="wikilink abifile-wikilink" data-content="Hydrogen diatomic molecule in a cell,  close to BCC Test the string method : 6 images,  exploring the transition path. Similar to test 22 (but with string)." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v6/Input/t24.in" target="_blank">t24.in</a>, <a class="wikilink abifile-wikilink" data-content="Hydrogen diatomic molecule in a cell,  close to BCC Test the string method : 6 images,  exploring the transition path but keeping X coordinated fixed. Similar to test 24" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v6/Input/t25.in" target="_blank">t25.in</a>, <a class="wikilink abifile-wikilink" data-content="Genetic algorithm structure random search. Hydrogen has been used as an example. No physical meaning. Number of images is only 10 but experience shows that at least 20 are fine." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v6/Input/t27.in" target="_blank">t27.in</a>, <a class="wikilink abifile-wikilink" data-content="Ga Al(1-x) Asx Test of alchemical calculations with images. Mostly checking the treatment of input variables and their echo. (Tests 30-36 are more slots for testing the ground state)" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v6/Input/t28.in" target="_blank">t28.in</a></li><li>v7:  <a class="wikilink abifile-wikilink" data-content="Hydrogen molecule: Test the Path-Integral Molecular Dynamics (PIMD) implementation in (N, V, T) ensemble. THe two thermostats (Nose-Hoover chains and Langevin) are tested together with the three coordinate systems (primitive,  normal mode,  staging). NOTE: MPI run with 2 procs is ok but fldiff comparison fails due to an extra section with --- Pseudopotential description --" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v7/Input/t08.in" target="_blank">t08.in</a></li><li>v8:  <a class="wikilink abifile-wikilink" data-content="H2 molecule: PIMD simulation with Langevin thermostat. Test of restart with HIST.nc file. Test of linear constraint." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v8/Input/t05.in" target="_blank">t05.in</a></li></ul></details><p>Gives the maximal number of molecular dynamics time steps or structural
optimization steps to be done for the set of images, referred to as &lsquo;image-
timesteps&rsquo;. At each image-timestep, all the images are propagated
simultaneously, each according to the algorithm determined by <a class="wikilink codevar-wikilink" href="../rlx#imgmov">imgmov</a> and
the usual accompanying input variables, and then the next positions and
velocities for each image are determined from the set of results obtained for
all images.</p>
<h2 id="optcell"><strong>optcell</strong><a class="headerlink" href="#optcell" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> OPTimize the CELL shape and dimensions<br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-content="How to perform a PIMD calculation" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../topics/PIMD">topic_PIMD</a>, <a class="wikilink topic-wikilink" data-content="How to perform a geometry optimization" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../topics/GeoOpt">topic_GeoOpt</a><br />
<em>Variable type:</em> integer<br />
<em>Dimensions:</em> scalar<br />
<em>Default value:</em> 0  </p>
<details class="note"><summary>Test list (click to open). Moderately used, [54/920] in all abinit tests, [9/116] in abinit tutorials</summary><ul><li>bigdft:  <a class="wikilink abifile-wikilink" data-content="H2 molecule with geometry optimisation using BFGS (test the reformating part of the code)." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/bigdft/Input/t10.in" target="_blank">t10.in</a>, <a class="wikilink abifile-wikilink" data-content="H2O in isolated boundary conditions (plane waves,  NC),  try tu use Damiens Poisson cutoff. Water molecule with HGH pseudo-potentials. This checks that a negative value in rhor will not make the Poisson solver crashes on XC computation." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/bigdft/Input/t22.in" target="_blank">t22.in</a> &hellip;</li><li>built-in:  <a class="wikilink abifile-wikilink" data-content="Ca atom with PAW,  writing a ETSF_IO wavefunction file." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/built-in/Input/testin_etsf_io.in" target="_blank">testin_etsf_io.in</a> &hellip;</li><li>etsf_io:  <a class="wikilink abifile-wikilink" data-content="Ca atom with PAW,  writing a ETSF_IO wavefunction file. Very quick built-in test,  to check that ETSF_IO is installed." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/etsf_io/Input/t00.in" target="_blank">t00.in</a>, <a class="wikilink abifile-wikilink" data-content="Ca atom in PAW. Test the output of density in PAW formalism with ETSF_IO. Also test the output of the wavefunctions in PAW formalism with ETSF_IO." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/etsf_io/Input/t09.in" target="_blank">t09.in</a> &hellip;</li><li>gpu:  <a class="wikilink abifile-wikilink" data-content="Silicon - diamond structure - PAW - volume relaxation Test the use of GPU within Projector Augmented-Wave formalism. Test structure relaxation (multiple calls to gstate routine)." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/gpu/Input/t04.in" target="_blank">t04.in</a> &hellip;</li><li>mpiio:  <a class="wikilink abifile-wikilink" data-content="C-diamond,  Bulk,  2 atoms,  paral_kgb,  with PAW. Test of ground state with different occs(7 and 0), and also ionmov 2. Only with 0 and 4 procs." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/mpiio/Input/t22.in" target="_blank">t22.in</a>, <a class="wikilink abifile-wikilink" data-content="C-diamond,  Bulk,  2 atoms,  paral_kgb,  with PAW. Test the triple parallelisation. We cannot check the distribution npband*npfft*npkpt=2*2*2=8 processors, this number being not allowed in the test procedure. Here,  we only test the parallelisation over FFT and kpoints: npband*npfft*npkpt=1*2*2=4 processors. In test tY.in we check other distributions for guarantee." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/mpiio/Input/t24.in" target="_blank">t24.in</a> &hellip;</li><li>paral:  <a class="wikilink abifile-wikilink" data-content="C-diamond,  Bulk,  2 atoms,  paral_kgb,  with PAW. Test of ground state with different occs(7 and 0), and also ionmov 2. Only with 0 and 4 procs." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/paral/Input/t22.in" target="_blank">t22.in</a>, <a class="wikilink abifile-wikilink" data-content="C-diamond,  Bulk,  2 atoms,  paral_kgb,  with PAW. Test the triple parallelisation. We cannot check the distribution npband*npfft*npkpt=2*2*2=8 processors, this number being not allowed in the test procedure. Here,  we only test the parallelisation over FFT and kpoints: npband*npfft*npkpt=1*2*2=4 processors. In test tY.in we check other distributions for guarantee." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/paral/Input/t24.in" target="_blank">t24.in</a>, <a class="wikilink abifile-wikilink" data-content="C-diamond,  Bulk,  2 atoms,  paral_kgb,  with PAW. Test of ground state with different occs(7 and 0), and also ionmov 2. Only with 4 procs. Same of test R:test the automatic parallelisation" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/paral/Input/t26.in" target="_blank">t26.in</a> &hellip;</li><li>seq:  <a class="wikilink abifile-wikilink" data-content="PbTiO3 in the tetragonal geometry. Test berryopt=14 (finite reduced electric field calculation, relaxing cell parameters,  e.g. optcell=2) (XG120616 : tranferred to tests/seq)" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/seq/Input/tsv6_121.in" target="_blank">tsv6_121.in</a>, <a class="wikilink abifile-wikilink" data-content="PbTiO3 in the tetragonal geometry. Test berryopt=16 (finite reduced electric displacement field calculation, relaxing cell parameters,  e.g. optcell=2) (XG120616 : tranferred to tests/seq)" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/seq/Input/tsv6_122.in" target="_blank">tsv6_122.in</a>, <a class="wikilink abifile-wikilink" data-content="PbTiO3 in the tetragonal geometry. Test berryopt=4 (finite electric field calculation,  relaxing cell parameters,  e.g. optcell=2) (XG120616 : tranferred to tests/seq)" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/seq/Input/tsv6_123.in" target="_blank">tsv6_123.in</a> &hellip;</li><li>tutoparal:  <a class="wikilink abifile-wikilink" data-content="Hydronium ion + NH3 molecule Ground state calculation keeping O and H atoms fixed" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/tutoparal/Input/tstring_01.in" target="_blank">tstring_01.in</a> &hellip;</li><li>tutorespfn:  <a class="wikilink abifile-wikilink" data-content="AlAs in hypothetical wurzite (hexagonal) structure Structural optimization run" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/tutorespfn/Input/telast_1.in" target="_blank">telast_1.in</a>, <a class="wikilink abifile-wikilink" data-content="Al fcc metal - elastic constant calculation" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/tutorespfn/Input/telast_6.in" target="_blank">telast_6.in</a>, <a class="wikilink abifile-wikilink" data-content="Finite difference calculation of the clamped-ion piezoelectric constants of AlAs" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/tutorespfn/Input/tffield_5.in" target="_blank">tffield_5.in</a> &hellip;</li><li>tutorial:  <a class="wikilink abifile-wikilink" data-content="Crystalline silicon : computation of the optimal lattice parameter Convergence with respect to the number of k points." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/tutorial/Input/tbase3_4.in" target="_blank">tbase3_4.in</a>, <a class="wikilink abifile-wikilink" data-content="Crystalline aluminum : optimization of the lattice parameter at fixed number of k points and broadening." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/tutorial/Input/tbase4_1.in" target="_blank">tbase4_1.in</a>, <a class="wikilink abifile-wikilink" data-content="Crystalline aluminum : optimization of the lattice parameter Convergence with respect to k points." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/tutorial/Input/tbase4_2.in" target="_blank">tbase4_2.in</a> &hellip;</li><li>v1:  <a class="wikilink abifile-wikilink" data-content="2 special k point Si in 2-atom diamond unit cell. Optimization of unit cell volume (optcell=1). Use 2 datasets,  and the variable getcell." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v1/Input/t78.in" target="_blank">t78.in</a>, <a class="wikilink abifile-wikilink" data-content="H2 molecule in a cell of variable size along the H2 chain =&amp;gt; H polymer. Optimization of cell size and atomic distances (optcell=4)." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v1/Input/t79.in" target="_blank">t79.in</a>, <a class="wikilink abifile-wikilink" data-content="Mg crystal,  1 atom per unit cell. Start with deformed FCC lattice, optimize the lattice towards FCC (optcell=2). Also test the output of the HIST.nc file (restartxf==-1)" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v1/Input/t80.in" target="_blank">t80.in</a> &hellip;</li><li>v2:  <a class="wikilink abifile-wikilink" data-content="Arsenic in rhombohedral structure. Optimization of unit cell size and shape. With 4x4x4 k point grid,  ecut 3.0,  occopt 4 and tsmear 0.06d0, one get acell 3*7.325  angdeg 3*57.36 and xred 0.2277 These input parameters are too small,  but still give a realistic geometry,  since with the much better parameters 12x12x12 k point grid,  ecut 12.0,  occopt 4 and tsmear 0.04d0, one gets acell 3*7.633  angdeg 3*54.95 and xred 0.2296,  while the experimental values are acell 3*7.751   angdeg 3*54.554 and xred 0.2276" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v2/Input/t44.in" target="_blank">t44.in</a> &hellip;</li><li>v3:  <a class="wikilink abifile-wikilink" data-content="Magnesium HCP. Optimize the cell volume and shape using different constraints. Checks whether the stresses are symmetric,  and whether the optimisation algorithm does not break the symmetry." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v3/Input/t42.in" target="_blank">t42.in</a> &hellip;</li><li>v4:  <a class="wikilink abifile-wikilink" data-content="1. Si diamond. Usual geometry,  k points,  psps. One of the two atoms is displaced in order to have non-zeroforces. Test whether energy and forces (and soon stresses) are correctly given,  even with use of spherical harmonics. Perform first computation with the use of Legendre polynomials; Then perform a computation with the use of spherical harmonics. Both must give the same results." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v4/Input/t01.in" target="_blank">t01.in</a>, <a class="wikilink abifile-wikilink" data-content="Ca in orthorhombic structure. 1 k point,  low ecut. Perform a test of GS total energy computation,  using an experimental PAW pseudopotential (pspcod=7)." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v4/Input/t04.in" target="_blank">t04.in</a>, <a class="wikilink abifile-wikilink" data-content="O in orthorhombic structure. 1 k point,  low ecut. Experimental PAW pseudopotential. The aim of this test is to verify that different values of input variable 'ortalg' give the same results for total energy. 'ortalg' is use in the conjugate gradient algorithm which solves,  in the case of PAW calculations, a generalized eigenproblem (H.|Psi&amp;gt;=Lambda.S|Psi&amp;gt;). Three datasets are perfomed;  they must give the same total energies values (when converged)." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v4/Input/t05.in" target="_blank">t05.in</a> &hellip;</li><li>v5:  <a class="wikilink abifile-wikilink" data-content="This is a test,  in the norm conserving case,  of the Molecular Dynamics flag ionmov==13. With the values of optcell=0,  1,  2 (isothermal (0),  isothermal/isenthalpic with homogeneous cell deformations (1) or full cell deformations (2).) This is the reversible integrator of Martyna,  Tuckerman,  Tobias and Klein. (Mol. Phys.,  1996,  p 1117). XG20090912 : Why is the result of dataset 1 and dataset 4 equivalent,  while the bmass varies by a factor of 2 ?" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v5/Input/t01.in" target="_blank">t01.in</a>, <a class="wikilink abifile-wikilink" data-content="Test to prepare a conductivity calculation (test v5#t10) Ground state calculation for 4 Al atoms The number of bands is unrealistic. This test produces files that are inputs for test 10. It tests the following parts of the code: optic_paw,  optic_paw_core Note that,  in addition to the psp file,  the core wave functions file is needed." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v5/Input/t09.in" target="_blank">t09.in</a>, <a class="wikilink abifile-wikilink" data-content="White tin,  body-centered tetragonal. This test checks that in an optcell 2 run,  the symmetry of the body-centered case is not broken. This test is run for two values of strprecon also, to test the strprecon variable. Coded and contributed by J. Zwanziger." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v5/Input/t77.in" target="_blank">t77.in</a> &hellip;</li><li>v6:  <a class="wikilink abifile-wikilink" data-content="Genetic algorithm structure random search. Hydrogen has been used as an example. No physical meaning. Number of images is only 10 but experience shows that at least 20 are fine." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v6/Input/t27.in" target="_blank">t27.in</a>, <a class="wikilink abifile-wikilink" data-content="Hydrure of Lithium-Boron,  in a face-centered monoclinic cell. Test the recognition of the space group (C2/c (# 15)). Was lacking a case before Dec 2011." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v6/Input/t30.in" target="_blank">t30.in</a> &hellip;</li><li>v7:  <a class="wikilink abifile-wikilink" data-content="Hydrogen molecule: Test the Path-Integral Molecular Dynamics (PIMD) implementation in (N, V, T) ensemble. THe two thermostats (Nose-Hoover chains and Langevin) are tested together with the three coordinate systems (primitive,  normal mode,  staging). NOTE: MPI run with 2 procs is ok but fldiff comparison fails due to an extra section with --- Pseudopotential description --" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v7/Input/t08.in" target="_blank">t08.in</a>, <a class="wikilink abifile-wikilink" data-content="Test the extrapolation of then wave functions from one structural relaxation (time) step to the other: extrapwf=1 WARNING: this feature is stil experimental and probably buggy. Si,  Bulk,  2 atoms" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v7/Input/t09.in" target="_blank">t09.in</a>, <a class="wikilink abifile-wikilink" data-content="Nickel ferromagnetic fcc structure with an ONCVPSP pseudo containing valence density. We test several improvements that are available when the NC file contains the PS valence charge. More specifically,  we test the initialization of the density from atomic quantities (initro.F90), the correction to the forces due to the residuals and the extrapolation of the density for structural relaxations (well,  xred is fixed by symmetry). We also test the treatment of the core charge in G-space in the case of GS calculations By default,  abinit handles XCCC in real space,  here we use nc_xccc_gspace==1 to treat it in G-space using the same approach as the one used in PAW." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v7/Input/t10.in" target="_blank">t10.in</a> &hellip;</li><li>vdwxc:  <a class="wikilink abifile-wikilink" data-content="Argon FCC. Test the vdW-DFT-D2 (Grimme) semi-empirical potential." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/vdwxc/Input/t10.in" target="_blank">t10.in</a> &hellip;</li></ul></details><p>Allows to optimize the unit cell shape and dimensions, when <a class="wikilink codevar-wikilink" href="../rlx#ionmov">ionmov</a>&gt;=2 or
3. The configuration for which the stress almost vanishes is iteratively
determined, by using the same algorithms as for the nuclei positions. Will
eventually modify <a class="wikilink codevar-wikilink" href="../basic#acell">acell</a> and/or <a class="wikilink codevar-wikilink" href="../basic#rprim">rprim</a>. The ionic positions are ALWAYS
updated, according to the forces. A target stress tensor might be defined, see
<a class="wikilink codevar-wikilink" href="../rlx#strtarget">strtarget</a>.</p>
<ul>
<li><a class="wikilink codevar-wikilink" href="../rlx#optcell">optcell</a>=0 : modify nuclear positions, since <a class="wikilink codevar-wikilink" href="../rlx#ionmov">ionmov</a>=2 or 3, but no cell shape and dimension optimisation.</li>
<li><a class="wikilink codevar-wikilink" href="../rlx#optcell">optcell</a>=1 : optimisation of volume only (do not modify <a class="wikilink codevar-wikilink" href="../basic#rprim">rprim</a>, and allow an homogeneous dilatation of the three components of <a class="wikilink codevar-wikilink" href="../basic#acell">acell</a>)</li>
<li><a class="wikilink codevar-wikilink" href="../rlx#optcell">optcell</a>=2 : full optimization of cell geometry (modify <a class="wikilink codevar-wikilink" href="../basic#acell">acell</a> and <a class="wikilink codevar-wikilink" href="../basic#rprim">rprim</a> - normalize the vectors of <a class="wikilink codevar-wikilink" href="../basic#rprim">rprim</a> to generate the <a class="wikilink codevar-wikilink" href="../basic#acell">acell</a>). This is the usual mode for cell shape and volume optimization. It takes into account the symmetry of the system, so that only the effectively relevant degrees of freedom are optimized.</li>
<li><a class="wikilink codevar-wikilink" href="../rlx#optcell">optcell</a>=3 : constant-volume optimization of cell geometry (modify <a class="wikilink codevar-wikilink" href="../basic#acell">acell</a> and <a class="wikilink codevar-wikilink" href="../basic#rprim">rprim</a> under constraint - normalize the vectors of <a class="wikilink codevar-wikilink" href="../basic#rprim">rprim</a> to generate the <a class="wikilink codevar-wikilink" href="../basic#acell">acell</a>)</li>
<li><a class="wikilink codevar-wikilink" href="../rlx#optcell">optcell</a>=4, 5 or 6 : optimize <a class="wikilink codevar-wikilink" href="../basic#acell">acell</a>(1), <a class="wikilink codevar-wikilink" href="../basic#acell">acell</a>(2), or <a class="wikilink codevar-wikilink" href="../basic#acell">acell</a>(3), respectively (only works if the two other vectors are orthogonal to the optimized one, the latter being along its cartesian axis).</li>
<li><a class="wikilink codevar-wikilink" href="../rlx#optcell">optcell</a>=7, 8 or 9 : optimize the cell geometry while keeping the first, second or third vector unchanged (only works if the two other vectors are orthogonal to the one left unchanged, the latter being along its cartesian axis).</li>
</ul>
<p>A few details require attention when performing unit cell optimisation :</p>
<ul>
<li>one has to get rid of the discontinuities due to discrete changes of plane wave number with cell size, by using a suitable value of <a class="wikilink codevar-wikilink" href="../rlx#ecutsm">ecutsm</a>;</li>
<li>one has to allow for the possibility of a larger sphere of plane waves, by using <a class="wikilink codevar-wikilink" href="../rlx#dilatmx">dilatmx</a>;</li>
<li>one might have to adjust the scale of stresses to the scale of forces, by using <a class="wikilink codevar-wikilink" href="../rlx#strfact">strfact</a>.</li>
<li>if all the reduced coordinates of atoms are fixed by symmetry, one cannot use <a class="wikilink codevar-wikilink" href="../basic#toldff">toldff</a> to stop the SCF cycle. (Suggestion : use <a class="wikilink codevar-wikilink" href="../basic#toldfe">toldfe</a> with a small value, like 1.0d-10)</li>
</ul>
<p>It is STRONGLY suggested first to optimize the ionic positions without cell
shape and size optimization (<a class="wikilink codevar-wikilink" href="../rlx#optcell">optcell</a>=0), then start the cell shape and
size optimization from the cell with relaxed ionic positions. Presently
(v3.1), one cannot restart (<a class="wikilink codevar-wikilink" href="../rlx#restartxf">restartxf</a>) a calculation with a non-zero
<a class="wikilink codevar-wikilink" href="../rlx#optcell">optcell</a> value from the (x,f) history of another run with a different non-
zero <a class="wikilink codevar-wikilink" href="../rlx#optcell">optcell</a> value. There are still a few problems at that level.</p>
<h2 id="pimass"><strong>pimass</strong><a class="headerlink" href="#pimass" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> Path Integral fictitious MASSes<br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-content="How to perform a PIMD calculation" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../topics/PIMD">topic_PIMD</a><br />
<em>Variable type:</em> real<br />
<em>Dimensions:</em> (<a class="wikilink codevar-wikilink" href="../basic#ntypat">ntypat</a>)<br />
<em>Default value:</em> <a class="wikilink codevar-wikilink" href="../basic#ntypat">ntypat</a><br />
<em>Only relevant if:</em> <a class="wikilink codevar-wikilink" href="../rlx#imgmov">imgmov</a>=9 or 13  </p>
<details class="note"><summary>Test list (click to open). Rarely used, [2/920] in all abinit tests, [0/116] in abinit tutorials</summary><ul><li>v6:  <a class="wikilink abifile-wikilink" data-content="(TO BE MODIFIED BY GREGORY) Hydrogen diatomic molecule in a cell,  close to BCC Test the string method : 6 images,  exploring the transition path but keeping X coordinated fixed. Similar to test 24" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v6/Input/t26.in" target="_blank">t26.in</a></li><li>v7:  <a class="wikilink abifile-wikilink" data-content="Hydrogen molecule: Test the Path-Integral Molecular Dynamics (PIMD) implementation in (N, V, T) ensemble. THe two thermostats (Nose-Hoover chains and Langevin) are tested together with the three coordinate systems (primitive,  normal mode,  staging). NOTE: MPI run with 2 procs is ok but fldiff comparison fails due to an extra section with --- Pseudopotential description --" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v7/Input/t08.in" target="_blank">t08.in</a></li></ul></details><p>Only relevant if <a class="wikilink codevar-wikilink" href="../rlx#imgmov">imgmov</a>=9 or 13 (Path-Integral Molecular Dynamics).<br />
Gives the fictitious masses ( _ D. Marx and M. Parrinello, J. Chem. Phys. 104,
4077 (1996) _ ) in atomic mass units for each kind of atom in cell. These
masses are the inertial masses used in performing Path Integral Molecular
Dynamics (PIMD), they are different from the true masses (<a class="wikilink codevar-wikilink" href="../rlx#amu">amu</a>) used to
define the quantum spring that relates the different beads in PIMD. They can
be chosen arbitrarily, but an appropriate choice will lead the different
variables to move on the same time scale in order to optimize the sampling
efficiency of the PIMD trajectory.<br />
If <a class="wikilink codevar-wikilink" href="../rlx#pitransform">pitransform</a>=1 (normal mode transformation), or <a class="wikilink codevar-wikilink" href="../rlx#pitransform">pitransform</a>=2
(staging transformation), <a class="wikilink codevar-wikilink" href="../rlx#pimass">pimass</a> is automatically set to its optimal
value.  </p>
<h2 id="pimd_constraint"><strong>pimd_constraint</strong><a class="headerlink" href="#pimd_constraint" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> Path-Integral Molecular Dynamics: CONSTRAINT to be applied on a reaction coordinate<br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-content="How to perform a PIMD calculation" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../topics/PIMD">topic_PIMD</a><br />
<em>Variable type:</em> integer<br />
<em>Dimensions:</em> scalar<br />
<em>Default value:</em> 0<br />
<em>Only relevant if:</em> <a class="wikilink codevar-wikilink" href="../rlx#imgmov">imgmov</a>=9 or 13  </p>
<details class="note"><summary>Test list (click to open). Rarely used, [1/920] in all abinit tests, [0/116] in abinit tutorials</summary><ul><li>v8:  <a class="wikilink abifile-wikilink" data-content="H2 molecule: PIMD simulation with Langevin thermostat. Test of restart with HIST.nc file. Test of linear constraint." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v8/Input/t05.in" target="_blank">t05.in</a></li></ul></details><p>Only relevant for Path-Integral Molecular Dynamics.<br />
Selects a constraint to be applied during the PIMD trajectory. The constraint
is holonomic (it is a relation between the position variables).In practice,
the total forces applied to the atomic positions are modified so as to respect
the constraint.<br />
To date, the available constraints are:  </p>
<ul>
<li><strong>0</strong>: no constraint</li>
<li><strong>1</strong>: <em>&ldquo;Blue Moon Ensemble&rdquo; method</em>.<br />
The constraint is a linear combination of the positions of atomic centroids
(this linear combination is kept constant during the simulation).<br />
Sum[W_i * X_i]=constant<br />
The X_i are the coordinates of the atomic centroids. The weights W_i have to
be specified with the <a class="wikilink codevar-wikilink" href="../rlx#wtatcon">wtatcon</a>(3,<a class="wikilink codevar-wikilink" href="../rlx#natcon">natcon</a>,<a class="wikilink codevar-wikilink" href="../rlx#nconeq">nconeq</a>),
<a class="wikilink codevar-wikilink" href="../rlx#iatcon">iatcon</a>(<a class="wikilink codevar-wikilink" href="../rlx#natcon">natcon</a>) and <a class="wikilink codevar-wikilink" href="../rlx#natcon">natcon</a> input parameters (where <a class="wikilink codevar-wikilink" href="../rlx#nconeq">nconeq</a> is
fixed to 1).<br />
More details on the implementation in: <a href="http://doi.org/10.1273/cbij.7.12">Y. Komeiji,Chem-Bio Informatics
Journal 7, 12-23 (2007)</a>.</li>
</ul>
<h2 id="pitransform"><strong>pitransform</strong><a class="headerlink" href="#pitransform" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> Path Integral coordinate TRANSFORMation<br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-content="How to perform a PIMD calculation" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../topics/PIMD">topic_PIMD</a><br />
<em>Variable type:</em> integer<br />
<em>Dimensions:</em> scalar<br />
<em>Default value:</em> 0  </p>
<details class="note"><summary>Test list (click to open). Rarely used, [3/920] in all abinit tests, [0/116] in abinit tutorials</summary><ul><li>v6:  <a class="wikilink abifile-wikilink" data-content="(TO BE MODIFIED BY GREGORY) Hydrogen diatomic molecule in a cell,  close to BCC Test the string method : 6 images,  exploring the transition path but keeping X coordinated fixed. Similar to test 24" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v6/Input/t26.in" target="_blank">t26.in</a></li><li>v7:  <a class="wikilink abifile-wikilink" data-content="Hydrogen molecule: Test the Path-Integral Molecular Dynamics (PIMD) implementation in (N, V, T) ensemble. THe two thermostats (Nose-Hoover chains and Langevin) are tested together with the three coordinate systems (primitive,  normal mode,  staging). NOTE: MPI run with 2 procs is ok but fldiff comparison fails due to an extra section with --- Pseudopotential description --" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v7/Input/t08.in" target="_blank">t08.in</a></li><li>v8:  <a class="wikilink abifile-wikilink" data-content="H2 molecule: PIMD simulation with Langevin thermostat. Test of restart with HIST.nc file. Test of linear constraint." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v8/Input/t05.in" target="_blank">t05.in</a></li></ul></details><p>Only relevant if <a class="wikilink codevar-wikilink" href="../rlx#imgmov">imgmov</a>=9 or 13 (Path-Integral Molecular Dynamics).
Coordinate transformation used in the integration of the Path Integral
Molecular Dynamics equations of motion. The transformation, with an
appropriate choice of fictitious masses (<a class="wikilink codevar-wikilink" href="../rlx#pimass">pimass</a>), is used to force the
different modes to move on the same time scale, and thus optimize the
efficiency of the statistical sampling in the corresponding statistical
ensemble. Available with a Langevin thermostat (<a class="wikilink codevar-wikilink" href="../rlx#imgmov">imgmov</a>=9) or with Nose-
Hoover chains (<a class="wikilink codevar-wikilink" href="../rlx#imgmov">imgmov</a>=13). See M. Tuckerman et al, J. Chem. Phys. 104,
5579 (1996).</p>
<p>If equal to 0, no transformation is applied (primitive coordinates).<br />
If equal to 1, normal mode transformation (in that case, <a class="wikilink codevar-wikilink" href="../rlx#nimage">nimage</a> must be
absolutely EVEN).<br />
If equal to 2, staging transformation.</p>
<h2 id="prtatlist"><strong>prtatlist</strong><a class="headerlink" href="#prtatlist" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> PRinT by ATom LIST of ATom<br />
<em>Characteristics:</em> <a class="wikilink" href="../../user-guide/abinit#32-more-about-abinit-input-variables">NO_MULTI</a><br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-content="How to print some useful quantities" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../topics/printing">topic_printing</a>, <a class="wikilink topic-wikilink" data-content="How to tune the output of computed quantities" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../topics/Output">topic_Output</a><br />
<em>Variable type:</em> integer<br />
<em>Dimensions:</em> (<a class="wikilink codevar-wikilink" href="../basic#natom">natom</a>)<br />
<em>Default value:</em> 0  </p>
<details class="note"><summary>Test list (click to open). Rarely used, [1/920] in all abinit tests, [0/116] in abinit tutorials</summary><ul><li>v4:  <a class="wikilink abifile-wikilink" data-content="Test of molecular dynamics of B atoms,  in the isokinetic ensemble,  based on the Thomas-Fermi functional (very fast,  even with 32 atoms)." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v4/Input/t97.in" target="_blank">t97.in</a></li></ul></details><p>This is an array of the numbers associated to the index atoms that the user
want to print in the output or log files, this is useful when you have a large
number of atoms and you are only interested to follow specific atoms, the
numbers associated should be consistent with the list in <a class="wikilink codevar-wikilink" href="../basic#xcart">xcart</a> or
<a class="wikilink codevar-wikilink" href="../basic#xred">xred</a>. This input varible does not affect the contents of the &ldquo;OUT.nc&rdquo; or
&ldquo;HIST.nc&rdquo;, those are NetCDF files containing the information about all the
atoms.</p>
<h2 id="qmass"><strong>qmass</strong><a class="headerlink" href="#qmass" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> Q thermostat MASS<br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-content="How to perform a PIMD calculation" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../topics/PIMD">topic_PIMD</a>, <a class="wikilink topic-wikilink" data-content="How to perform a molecular dynamics calculation" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../topics/MolecularDynamics">topic_MolecularDynamics</a><br />
<em>Variable type:</em> real<br />
<em>Dimensions:</em> (<a class="wikilink codevar-wikilink" href="../rlx#nnos">nnos</a>)<br />
<em>Default value:</em> *10.0  </p>
<details class="note"><summary>Test list (click to open). Rarely used, [2/920] in all abinit tests, [0/116] in abinit tutorials</summary><ul><li>v5:  <a class="wikilink abifile-wikilink" data-content="This is a test,  in the norm conserving case,  of the Molecular Dynamics flag ionmov==13. With the values of optcell=0,  1,  2 (isothermal (0),  isothermal/isenthalpic with homogeneous cell deformations (1) or full cell deformations (2).) This is the reversible integrator of Martyna,  Tuckerman,  Tobias and Klein. (Mol. Phys.,  1996,  p 1117). XG20090912 : Why is the result of dataset 1 and dataset 4 equivalent,  while the bmass varies by a factor of 2 ?" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v5/Input/t01.in" target="_blank">t01.in</a></li><li>v7:  <a class="wikilink abifile-wikilink" data-content="Hydrogen molecule: Test the Path-Integral Molecular Dynamics (PIMD) implementation in (N, V, T) ensemble. THe two thermostats (Nose-Hoover chains and Langevin) are tested together with the three coordinate systems (primitive,  normal mode,  staging). NOTE: MPI run with 2 procs is ok but fldiff comparison fails due to an extra section with --- Pseudopotential description --" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v7/Input/t08.in" target="_blank">t08.in</a></li></ul></details><p>This are the masses of the chains of <a class="wikilink codevar-wikilink" href="../rlx#nnos">nnos</a> thermostats to be used when
<a class="wikilink codevar-wikilink" href="../rlx#ionmov">ionmov</a>=13 (Molecular Dynamics) or <a class="wikilink codevar-wikilink" href="../rlx#imgmov">imgmov</a>=13 (Path Integral Molecular
Dynamics).</p>
<p>If <a class="wikilink codevar-wikilink" href="../rlx#ionmov">ionmov</a>=13 (Molecular Dynamics), this temperature control can be used
with  <a class="wikilink codevar-wikilink" href="../rlx#optcell">optcell</a> =0, 1 (homogeneous cell deformation) or 2 (full cell
deformation).<br />
If <a class="wikilink codevar-wikilink" href="../rlx#imgmov">imgmov</a>=13 (Path Integral Molecular Dynamics), this temperature control
can be used with  <a class="wikilink codevar-wikilink" href="../rlx#optcell">optcell</a> =0 (NVT ensemble) or 2 (fully flexible NPT
ensemble). In that case, <a class="wikilink codevar-wikilink" href="../rlx#optcell">optcell</a>=2 iS NOT USABLE yet.</p>
<h2 id="random_atpos"><strong>random_atpos</strong><a class="headerlink" href="#random_atpos" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> RANDOM ATomic POSitions<br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-content="How to specify a crystal, with atomic positions and symmetries" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../topics/crystal">topic_crystal</a>, <a class="wikilink topic-wikilink" data-content="How to perform a geometry optimization" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../topics/GeoOpt">topic_GeoOpt</a><br />
<em>Variable type:</em> integer<br />
<em>Dimensions:</em> scalar<br />
<em>Default value:</em> 0  </p>
<details class="note"><summary>Test list (click to open). Rarely used, [1/920] in all abinit tests, [0/116] in abinit tutorials</summary><ul><li>v6:  <a class="wikilink abifile-wikilink" data-content="Genetic algorithm structure random search. Hydrogen has been used as an example. No physical meaning. Number of images is only 10 but experience shows that at least 20 are fine." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v6/Input/t27.in" target="_blank">t27.in</a></li></ul></details><p>Control the inner coordinates, which can be generated randomly by using 4
different methods depending ont its value<br />
(0) if zero, no random generation and xred are taken as they have been
introduced by the user<br />
(1) if one, particles are generated completely random within the unit cell.<br />
(2) if two, particles are generated randomly but the inner particle distance
is always larger than a factor of the sum of the covalent bonds between the
atoms (note : this is incompatible with the definition of alchemical mixing,
in which <a class="wikilink codevar-wikilink" href="../basic#ntypat">ntypat</a> differs from <a class="wikilink codevar-wikilink" href="../gstate#npsp">npsp</a>)</p>
<h2 id="restartxf"><strong>restartxf</strong><a class="headerlink" href="#restartxf" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> RESTART from (X,F) history<br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-content="How to perform a PIMD calculation" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../topics/PIMD">topic_PIMD</a>, <a class="wikilink topic-wikilink" data-content="How to perform a molecular dynamics calculation" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../topics/MolecularDynamics">topic_MolecularDynamics</a>, <a class="wikilink topic-wikilink" data-content="How to perform a geometry optimization" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../topics/GeoOpt">topic_GeoOpt</a><br />
<em>Variable type:</em> integer<br />
<em>Dimensions:</em> scalar<br />
<em>Default value:</em> 0  </p>
<details class="note"><summary>Test list (click to open). Rarely used, [3/920] in all abinit tests, [0/116] in abinit tutorials</summary><ul><li>v1:  <a class="wikilink abifile-wikilink" data-content="Same as test 66,  H2 molecule with 1 k points,  but uses the restartxf=-1 option to test the output of the HIST.nc file. Note that HIST.nc is not read since one should introduce a getrestartxf variable." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v1/Input/t75.in" target="_blank">t75.in</a>, <a class="wikilink abifile-wikilink" data-content="Mg crystal,  1 atom per unit cell. Start with deformed FCC lattice, optimize the lattice towards FCC (optcell=2). Also test the output of the HIST.nc file (restartxf==-1)" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v1/Input/t80.in" target="_blank">t80.in</a></li><li>v8:  <a class="wikilink abifile-wikilink" data-content="H2 molecule: PIMD simulation with Langevin thermostat. Test of restart with HIST.nc file. Test of linear constraint." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v8/Input/t05.in" target="_blank">t05.in</a></li></ul></details><p>Control the restart of a molecular dynamics or structural optimization job.  </p>
<p>** restartxf&gt;0 (Deprecated) ** :The code reads from the input wf file, the previous history of atomic coordinates and corresponding forces, in order to continue the work done by the job that produced this wf file. If <a class="wikilink codevar-wikilink" href="../rlx#optcell">optcell</a>/=0, the history of <a class="wikilink codevar-wikilink" href="../basic#acell">acell</a> and <a class="wikilink codevar-wikilink" href="../basic#rprim">rprim</a> variables is also taken into account. The code will take into consideration the whole history (if <a class="wikilink codevar-wikilink" href="../rlx#restartxf">restartxf</a>=1), or discard the few first (x,f) pairs, and begin only at the pair whose number corresponds to <a class="wikilink codevar-wikilink" href="../rlx#restartxf">restartxf</a>. <br />
Works only for <a class="wikilink codevar-wikilink" href="../rlx#ionmov">ionmov</a>=2 (Broyden) and when an input wavefunction file is
specified, thanks to the appropriate values of <a class="wikilink codevar-wikilink" href="../files#irdwfk">irdwfk</a> or <a class="wikilink codevar-wikilink" href="../files#getwfk">getwfk</a>.  </p>
<p>NOTES :<br />
* The input wf file must have been produced by a run that exited cleanly. It cannot be one of the temporary wf files that exist when a job crashed. <br />
* One cannot restart a calculation with a non-zero <a class="wikilink codevar-wikilink" href="../rlx#optcell">optcell</a> value from the (x,f) history of another run with a different non-zero <a class="wikilink codevar-wikilink" href="../rlx#optcell">optcell</a> value. Starting a non-zero <a class="wikilink codevar-wikilink" href="../rlx#optcell">optcell</a> run from a zero <a class="wikilink codevar-wikilink" href="../rlx#optcell">optcell</a> run should work. <br />
* Deprecated, the use of the new options (-1 and -2) is preferred.   </p>
<p>** restartxf=0 (Default) ** : No restart procedure is enable and will start a Molecular dynamics or structural optimization from scratch.   </p>
<p>** restartxf=-1 (New) ** : Use the HIST file to reconstruct a partial calculation. It will reconstruct the different configurations using the forces and stress store in the HIST file, instead of calling the SCF procedure. <br />
Enable ** restartxf=-1 ** from the beginning is harmless. The only condition
is to keep the input file the same in such a way that the same predictor is
used and it will predict the same structure recorded in the HIST file.<br />
This option will always compute extra <a class="wikilink codevar-wikilink" href="../rlx#ntime">ntime</a> iterations independent of the
number of iterations recovered previously.  </p>
<p>** restartxf=-2 (New) ** :Read the HIST file and select the atomic positions and cell parameters with the lowest energy. Forget all the history and start the calculation using those values. The original atomic coordinates and cell parameters are irrelevant in that case.   </p>
<p>NOTES:<br />
* You can use ** restartxf=-1 or -2 ** for all predictiors that make no use of random numbers. <br />
* You can use ** restartxf=-1 or -2 ** to restart a calculation that was not completed. The HIST file is written on each iteration. So you always have something to recover from. <br />
* You can take advantage of the appropriate values of <a class="wikilink codevar-wikilink" href="../files#irdwfk">irdwfk</a> or <a class="wikilink codevar-wikilink" href="../files#getwfk">getwfk</a> to get a good wave function to continue your job. </p>
<h2 id="signperm"><strong>signperm</strong><a class="headerlink" href="#signperm" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> SIGN of PERMutation potential<br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-content="How to perform a molecular dynamics calculation" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../topics/MolecularDynamics">topic_MolecularDynamics</a><br />
<em>Variable type:</em> integer<br />
<em>Dimensions:</em> scalar<br />
<em>Default value:</em> 1  </p>
<details class="note"><summary>Test list (click to open). Rarely used, [2/920] in all abinit tests, [0/116] in abinit tutorials</summary><ul><li>v2:  <a class="wikilink abifile-wikilink" data-content="Ge liquid. Test of Nose and Langevin dynamics. 2 atoms in a cell. Allows 10 time steps." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v2/Input/t87.in" target="_blank">t87.in</a>, <a class="wikilink abifile-wikilink" data-content="Basic test of geometry optimization for water First computing the ground state of a close configuration Uses 5 different methods to find the relaxed positions" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v2/Input/t88.in" target="_blank">t88.in</a></li></ul></details><p>+1 favors alternation of species -1 favors segregation</p>
<h2 id="strfact"><strong>strfact</strong><a class="headerlink" href="#strfact" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> STRess FACTor<br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-content="How to perform a geometry optimization" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../topics/GeoOpt">topic_GeoOpt</a><br />
<em>Variable type:</em> real<br />
<em>Dimensions:</em> scalar<br />
<em>Default value:</em> 100  </p>
<details class="note"><summary>Test list (click to open). Rarely used, [9/920] in all abinit tests, [2/116] in abinit tutorials</summary><ul><li>seq:  <a class="wikilink abifile-wikilink" data-content="PbTiO3 in the tetragonal geometry. Test berryopt=14 (finite reduced electric field calculation, relaxing cell parameters,  e.g. optcell=2) (XG120616 : tranferred to tests/seq)" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/seq/Input/tsv6_121.in" target="_blank">tsv6_121.in</a>, <a class="wikilink abifile-wikilink" data-content="PbTiO3 in the tetragonal geometry. Test berryopt=16 (finite reduced electric displacement field calculation, relaxing cell parameters,  e.g. optcell=2) (XG120616 : tranferred to tests/seq)" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/seq/Input/tsv6_122.in" target="_blank">tsv6_122.in</a>, <a class="wikilink abifile-wikilink" data-content="PbTiO3 in the tetragonal geometry. Test berryopt=4 (finite electric field calculation,  relaxing cell parameters,  e.g. optcell=2) (XG120616 : tranferred to tests/seq)" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/seq/Input/tsv6_123.in" target="_blank">tsv6_123.in</a>, <a class="wikilink abifile-wikilink" data-content="PbTiO3 in the tetragonal geometry. Test berryopt=6 (finite electric displacement field calculation, relaxing cell perameters,  e.g. optcell=2) (XG120616 : tranferred to tests/seq)" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/seq/Input/tsv6_124.in" target="_blank">tsv6_124.in</a>, <a class="wikilink abifile-wikilink" data-content="Test berryopt=17 (mixed finite electric field and electric displacement field boundary condition,  relaxing cell parameters,  e.g. optcell=2) (XG120616 : tranferred to tests/seq)" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/seq/Input/tsv6_125.in" target="_blank">tsv6_125.in</a></li><li>tutorespfn:  <a class="wikilink abifile-wikilink" data-content="AlAs in hypothetical wurzite (hexagonal) structure Structural optimization run" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/tutorespfn/Input/telast_1.in" target="_blank">telast_1.in</a>, <a class="wikilink abifile-wikilink" data-content="Al fcc metal - elastic constant calculation" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/tutorespfn/Input/telast_6.in" target="_blank">telast_6.in</a></li><li>v1:  <a class="wikilink abifile-wikilink" data-content="2 special k point Si in 2-atom diamond unit cell. Optimization of unit cell volume (optcell=1). Use 2 datasets,  and the variable getcell." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v1/Input/t78.in" target="_blank">t78.in</a></li><li>v5:  <a class="wikilink abifile-wikilink" data-content="Make a simple geometry relaxation and validate the XML output (testing prtxml)." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v5/Input/t79.in" target="_blank">t79.in</a></li></ul></details><p>The stresses multiplied by <a class="wikilink codevar-wikilink" href="../rlx#strfact">strfact</a> will be treated like forces in the
process of optimization (<a class="wikilink codevar-wikilink" href="../rlx#ionmov">ionmov</a>=2, non-zero <a class="wikilink codevar-wikilink" href="../rlx#optcell">optcell</a>).<br />
For example, the stopping criterion defined by <a class="wikilink codevar-wikilink" href="../rlx#tolmxf">tolmxf</a> relates to these
scaled stresses.</p>
<h2 id="string_algo"><strong>string_algo</strong><a class="headerlink" href="#string_algo" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> STRING method ALGOrithm<br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-content="How to calculate transition paths" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../topics/TransPath">topic_TransPath</a><br />
<em>Variable type:</em> integer<br />
<em>Dimensions:</em> scalar<br />
<em>Default value:</em> 1  </p>
<details class="note"><summary>Test list (click to open). Rarely used, [1/920] in all abinit tests, [0/116] in abinit tutorials</summary><ul><li>v6:  <a class="wikilink abifile-wikilink" data-content="Hydrogen diatomic molecule in a cell,  close to BCC Test the string method : 6 images,  exploring the transition path. Similar to test 22 (but with string)." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v6/Input/t24.in" target="_blank">t24.in</a></li></ul></details><p>Relevant only when <a class="wikilink codevar-wikilink" href="../rlx#imgmov">imgmov</a>=2 (String Method).<br />
Gives the variant of the String Method method used.<br />
Possible values can be:  </p>
<ul>
<li>
<p>0=&gt; ** Original String Method ** . <br />
NOT YET IMPLEMENTED<br />
_ See: Phys. Rev. B 66, 052301 (2002) _</p>
</li>
<li>
<p>1=&gt; ** Simplified String Method ** with parametrization by ** equal arc length ** . <br />
Instead of using the normal force (wr the band), the full force is used; the
reparametrization is enforced by keeping the points of the string equally
spaced.<br />
_ See: J. Chem. Phys. 126, 164103 (2007) _</p>
</li>
<li>
<p>2=&gt; ** Simplified String Method ** with parametrization by ** energy-weighted arc length ** . <br />
A variant of the Simplified String Method (like 2-); the reparametrization is
done by using energy-weight arc-lengths, giving a finer distribution near the
saddle point..<br />
_ See: J. Chem. Phys. 126, 164103 (2007) and J. Chem. Phys. 130, 244108 (2009)
_</p>
</li>
</ul>
<h2 id="strprecon"><strong>strprecon</strong><a class="headerlink" href="#strprecon" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> STRess PRECONditioner<br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-content="How to tune the computation of forces and stresses" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../topics/ForcesStresses">topic_ForcesStresses</a>, <a class="wikilink topic-wikilink" data-content="How to perform a geometry optimization" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../topics/GeoOpt">topic_GeoOpt</a><br />
<em>Variable type:</em> real<br />
<em>Dimensions:</em> scalar<br />
<em>Default value:</em> 1.0  </p>
<details class="note"><summary>Test list (click to open). Rarely used, [3/920] in all abinit tests, [0/116] in abinit tutorials</summary><ul><li>bigdft:  <a class="wikilink abifile-wikilink" data-content="Hypothetical NaTi molecule. Test the diagonalisation scheme with wavelets" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/bigdft/Input/t14.in" target="_blank">t14.in</a></li><li>v5:  <a class="wikilink abifile-wikilink" data-content="White tin,  body-centered tetragonal. This test checks that in an optcell 2 run,  the symmetry of the body-centered case is not broken. This test is run for two values of strprecon also, to test the strprecon variable. Coded and contributed by J. Zwanziger." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v5/Input/t77.in" target="_blank">t77.in</a></li><li>v6:  <a class="wikilink abifile-wikilink" data-content="HI 8 molecule per cell,  the cell is triclinic. Prepare the density for the next Bader analysis." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v6/Input/t18.in" target="_blank">t18.in</a></li></ul></details><p>This is a scaling factor to initialize the part of the Hessian related to the
treatment of the stresses (optimisation of the unit cell). In case there is an
instability, decrease the default value, e.g. set it to 0.1 .</p>
<h2 id="strtarget"><strong>strtarget</strong><a class="headerlink" href="#strtarget" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> STRess TARGET<br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-content="How to tune the computation of forces and stresses" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../topics/ForcesStresses">topic_ForcesStresses</a>, <a class="wikilink topic-wikilink" data-content="How to perform a geometry optimization" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../topics/GeoOpt">topic_GeoOpt</a><br />
<em>Variable type:</em> real<br />
<em>Dimensions:</em> (6)<br />
<em>Default value:</em> [0.0, 0.0, 0.0, 0.0, 0.0, 0.0]  </p>
<details class="note"><summary>Test list (click to open). Rarely used, [3/920] in all abinit tests, [0/116] in abinit tutorials</summary><ul><li>v1:  <a class="wikilink abifile-wikilink" data-content="Mg crystal,  1 atom per unit cell. Start with deformed FCC lattice, optimize the lattice towards FCC (optcell=2). Also test the output of the HIST.nc file (restartxf==-1)" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v1/Input/t80.in" target="_blank">t80.in</a></li><li>v5:  <a class="wikilink abifile-wikilink" data-content="This is a test,  in the norm conserving case,  of the Molecular Dynamics flag ionmov==13. With the values of optcell=0,  1,  2 (isothermal (0),  isothermal/isenthalpic with homogeneous cell deformations (1) or full cell deformations (2).) This is the reversible integrator of Martyna,  Tuckerman,  Tobias and Klein. (Mol. Phys.,  1996,  p 1117). XG20090912 : Why is the result of dataset 1 and dataset 4 equivalent,  while the bmass varies by a factor of 2 ?" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v5/Input/t01.in" target="_blank">t01.in</a></li><li>v6:  <a class="wikilink abifile-wikilink" data-content="HI 8 molecule per cell,  the cell is triclinic. Prepare the density for the next Bader analysis." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v6/Input/t18.in" target="_blank">t18.in</a></li></ul></details><p>The components of the stress tensor must be stored according to : (1,1)-&gt;1
; (2,2)-&gt;2 ; (3,3)-&gt;3 ; (2,3)-&gt;4 ; (3,1)-&gt;5 ; (1,2)-&gt;6. The
conversion factor between Ha/Bohr**3 and GPa is : 1 Ha/Bohr**3 = 29421.033d0
GPa.<br />
Not used if <a class="wikilink codevar-wikilink" href="../rlx#optcell">optcell</a>==0.</p>
<h2 id="tolimg"><strong>tolimg</strong><a class="headerlink" href="#tolimg" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> TOLerance on the mean total energy for IMaGes<br />
<em>Characteristics:</em> <a class="wikilink" href="../../user-guide/abinit#32-more-about-abinit-input-variables">ENERGY</a><br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-content="How to calculate transition paths" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../topics/TransPath">topic_TransPath</a><br />
<em>Variable type:</em> real<br />
<em>Dimensions:</em> scalar<br />
<em>Default value:</em> 5e-05  </p>
<details class="note"><summary>Test list (click to open). Rarely used, [7/920] in all abinit tests, [0/116] in abinit tutorials</summary><ul><li>paral:  <a class="wikilink abifile-wikilink" data-content="Test the string method within parallelization over images Inspired by test v6#22. Hydrogen diatomic molecule in a cell,  close to BCC 7 images,  exploring the transition path. Three datasets,  testing each value of prtvolimg. Processors distribution automatically determined:" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/paral/Input/t08.in" target="_blank">t08.in</a>, <a class="wikilink abifile-wikilink" data-content="Test the string method within parallelization over images Inspired by test v6#22. Hydrogen diatomic molecule in a cell,  close to BCC 7 images,  exploring the transition path. Three datasets,  testing each value of prtvolimg. Processors distribution automatically determined:" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/paral/Input/t08.in" target="_blank">t08.in</a>, <a class="wikilink abifile-wikilink" data-content="Test the string method within parallelization over images Inspired by test v6#22. Hydrogen diatomic molecule in a cell,  close to BCC 7 images,  exploring the transition path. Three datasets,  testing each value of prtvolimg. Processors distribution automatically determined:" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/paral/Input/t08.in" target="_blank">t08.in</a>, <a class="wikilink abifile-wikilink" data-content="Test the string method within parallelization over images Inspired by test v6#22. Hydrogen diatomic molecule in a cell,  close to BCC 7 images,  exploring the transition path. Three datasets,  testing each value of prtvolimg. Processors distribution automatically determined:" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/paral/Input/t08.in" target="_blank">t08.in</a></li><li>v6:  <a class="wikilink abifile-wikilink" data-content="Bismuth,  rhombohedral,  two atoms per unit cell Scalar relativistic. Relax four different images,  using the steepest descent algorithm." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v6/Input/t21.in" target="_blank">t21.in</a>, <a class="wikilink abifile-wikilink" data-content="Hydrogen diatomic molecule in a cell,  close to BCC Test the string method : 6 images,  exploring the transition path. Similar to test 22 (but with string)." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v6/Input/t24.in" target="_blank">t24.in</a>, <a class="wikilink abifile-wikilink" data-content="Hydrogen diatomic molecule in a cell,  close to BCC Test the string method : 6 images,  exploring the transition path but keeping X coordinated fixed. Similar to test 24" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v6/Input/t25.in" target="_blank">t25.in</a></li></ul></details><p>Sets a maximal absolute energy tolerance (in hartree, averaged over dynamic
images) below which iterations on images (the one governed by the
<a class="wikilink codevar-wikilink" href="../rlx#ntimimage">ntimimage</a> input variable) will stop.<br />
This is to be used when trying to optimize a population of structures to their
lowest energy configuration, taking into account the particular algorithm
defined by <a class="wikilink codevar-wikilink" href="../rlx#imgmov">imgmov</a><br />
A value of about 5.0d-5 hartree or smaller is suggested (this corresponds to
about 3.7d-7 eV).<br />
No meaning for RF calculations.</p>
<h2 id="tolmxde"><strong>tolmxde</strong><a class="headerlink" href="#tolmxde" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> TOLerance on the MaXimal Difference in Energy<br />
<em>Characteristics:</em> <a class="wikilink" href="../../user-guide/abinit#32-more-about-abinit-input-variables">ENERGY</a><br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-content="How to perform a geometry optimization" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../topics/GeoOpt">topic_GeoOpt</a><br />
<em>Variable type:</em> real<br />
<em>Dimensions:</em> scalar<br />
<em>Default value:</em> 0.0  </p>
<details class="note"><summary>Test list (click to open). Rarely used, [1/920] in all abinit tests, [0/116] in abinit tutorials</summary><ul><li>v7:  <a class="wikilink abifile-wikilink" data-content="Test tolmxde input variable. For silicon supercell with atomic positions randomly displaced" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v7/Input/t15.in" target="_blank">t15.in</a></li></ul></details><p>Sets a maximal difference in energy with respect to the two previous steps
below which BFGS structural relaxation iterations will stop.<br />
A value of about 0.0005 eV/atom or smaller is suggested.<br />
In order to use tolmxde, you should explicitly set tolmxf to 0.0.<br />
No meaning for RF calculations.</p>
<h2 id="tolmxf"><strong>tolmxf</strong><a class="headerlink" href="#tolmxf" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> TOLerance on the MaXimal Force<br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-content="How to perform a geometry optimization" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../topics/GeoOpt">topic_GeoOpt</a><br />
<em>Variable type:</em> real<br />
<em>Dimensions:</em> scalar<br />
<em>Default value:</em> 5e-05  </p>
<details class="note"><summary>Test list (click to open). Moderately used, [76/920] in all abinit tests, [14/116] in abinit tutorials</summary><ul><li>bigdft:  <a class="wikilink abifile-wikilink" data-content="H2O in isolated boundary conditions (plane waves,  NC),  try tu use Damiens Poisson cutoff. Water molecule with HGH pseudo-potentials. This checks that a negative value in rhor will not make the Poisson solver crashes on XC computation." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/bigdft/Input/t22.in" target="_blank">t22.in</a> &hellip;</li><li>built-in:  <a class="wikilink abifile-wikilink" data-content="H2 molecule,  determination of the H-H distance by the Broyden algorithm." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/built-in/Input/testin_fast.in" target="_blank">testin_fast.in</a> &hellip;</li><li>fast:  <a class="wikilink abifile-wikilink" data-content="H2 molecule,  determination of the H-H distance by the Broyden algorithm. Very quick built-in test,  to check that ABINIT is working." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/fast/Input/t00.in" target="_blank">t00.in</a>, <a class="wikilink abifile-wikilink" data-content="H2 diatomic molecule,  static,  to check accurate forces." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/fast/Input/t02.in" target="_blank">t02.in</a>, <a class="wikilink abifile-wikilink" data-content="2 special k point Si in 2-atom diamond unit cell.  Uses symmetry. mkmem=mkpt,  mffmem=1 (&amp;quot;in core&amp;quot; solution), This calculation is also iterated to convergence.  ecut too small. Uses original Teter extended norm conserving psp 14si.psp." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/fast/Input/t03.in" target="_blank">t03.in</a> &hellip;</li><li>gpu:  <a class="wikilink abifile-wikilink" data-content="Silicon - diamond structure - PAW - volume relaxation Test the use of GPU within Projector Augmented-Wave formalism. Test structure relaxation (multiple calls to gstate routine)." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/gpu/Input/t04.in" target="_blank">t04.in</a> &hellip;</li><li>mpiio:  <a class="wikilink abifile-wikilink" data-content="Si,  Bulk,  2 atoms,  paral_kgb. Test of ground state with different occs(7 and 0), and also ionmov 2. Only with 0 and 4 procs." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/mpiio/Input/t21.in" target="_blank">t21.in</a>, <a class="wikilink abifile-wikilink" data-content="C-diamond,  Bulk,  2 atoms,  paral_kgb,  with PAW. Test of ground state with different occs(7 and 0), and also ionmov 2. Only with 0 and 4 procs." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/mpiio/Input/t22.in" target="_blank">t22.in</a>, <a class="wikilink abifile-wikilink" data-content="C-diamond,  Bulk,  2 atoms,  paral_kgb,  with PAW. Test the triple parallelisation. We cannot check the distribution npband*npfft*npkpt=2*2*2=8 processors, this number being not allowed in the test procedure. Here,  we only test the parallelisation over FFT and kpoints: npband*npfft*npkpt=1*2*2=4 processors. In test tY.in we check other distributions for guarantee." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/mpiio/Input/t24.in" target="_blank">t24.in</a> &hellip;</li><li>paral:  <a class="wikilink abifile-wikilink" data-content="Si,  Bulk,  2 atoms,  paral_kgb. Test of ground state with different occs(7 and 0), and also ionmov 2. Only with 0 and 4 procs." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/paral/Input/t21.in" target="_blank">t21.in</a>, <a class="wikilink abifile-wikilink" data-content="C-diamond,  Bulk,  2 atoms,  paral_kgb,  with PAW. Test of ground state with different occs(7 and 0), and also ionmov 2. Only with 0 and 4 procs." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/paral/Input/t22.in" target="_blank">t22.in</a>, <a class="wikilink abifile-wikilink" data-content="C-diamond,  Bulk,  2 atoms,  paral_kgb,  with PAW. Test the triple parallelisation. We cannot check the distribution npband*npfft*npkpt=2*2*2=8 processors, this number being not allowed in the test procedure. Here,  we only test the parallelisation over FFT and kpoints: npband*npfft*npkpt=1*2*2=4 processors. In test tY.in we check other distributions for guarantee." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/paral/Input/t24.in" target="_blank">t24.in</a> &hellip;</li><li>seq:  <a class="wikilink abifile-wikilink" data-content="PbTiO3 in the tetragonal geometry. Test berryopt=14 (finite reduced electric field calculation, relaxing cell parameters,  e.g. optcell=2) (XG120616 : tranferred to tests/seq)" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/seq/Input/tsv6_121.in" target="_blank">tsv6_121.in</a>, <a class="wikilink abifile-wikilink" data-content="PbTiO3 in the tetragonal geometry. Test berryopt=16 (finite reduced electric displacement field calculation, relaxing cell parameters,  e.g. optcell=2) (XG120616 : tranferred to tests/seq)" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/seq/Input/tsv6_122.in" target="_blank">tsv6_122.in</a>, <a class="wikilink abifile-wikilink" data-content="PbTiO3 in the tetragonal geometry. Test berryopt=4 (finite electric field calculation,  relaxing cell parameters,  e.g. optcell=2) (XG120616 : tranferred to tests/seq)" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/seq/Input/tsv6_123.in" target="_blank">tsv6_123.in</a> &hellip;</li><li>tutoparal:  <a class="wikilink abifile-wikilink" data-content="Hydronium ion + NH3 molecule Ground state calculation keeping O and H atoms fixed" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/tutoparal/Input/tstring_01.in" target="_blank">tstring_01.in</a> &hellip;</li><li>tutorespfn:  <a class="wikilink abifile-wikilink" data-content="AlAs in hypothetical wurzite (hexagonal) structure Structural optimization run" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/tutorespfn/Input/telast_1.in" target="_blank">telast_1.in</a>, <a class="wikilink abifile-wikilink" data-content="Al fcc metal - elastic constant calculation" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/tutorespfn/Input/telast_6.in" target="_blank">telast_6.in</a>, <a class="wikilink abifile-wikilink" data-content="Finite difference calculation of the clamped-ion piezoelectric constants of AlAs" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/tutorespfn/Input/tffield_5.in" target="_blank">tffield_5.in</a> &hellip;</li><li>tutorial:  <a class="wikilink abifile-wikilink" data-content="H2 molecule in a big box This file will optimize automatically the interatomic distance" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/tutorial/Input/tbase1_3.in" target="_blank">tbase1_3.in</a>, <a class="wikilink abifile-wikilink" data-content="H2 molecule in a big box This file to optimize the H2 bond length,  compute the associated total energy,  then to compute the total energy of the isolated H atom." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/tutorial/Input/tbase2_1.in" target="_blank">tbase2_1.in</a>, <a class="wikilink abifile-wikilink" data-content="H2 molecule in a big box This file to optimize the H2 bond length,  compute the associated total energy,  then to compute the total energy of the isolated H atom. Here,  a double loop has been used." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/tutorial/Input/tbase2_2.in" target="_blank">tbase2_2.in</a> &hellip;</li><li>v1:  <a class="wikilink abifile-wikilink" data-content="Mo surface,  with 10 atom per cell. Not very large,  but the number of bands make it a medium-size problem : 36 bands." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v1/Input/t41.in" target="_blank">t41.in</a>, <a class="wikilink abifile-wikilink" data-content="Test the option ionmov=4. 8-atom Si cube,  non spin-polarized. Same test as number 22a of fast,  except for the change of ionmov,  no reading of wavefunctions,  iscf to 2,  and different tolerance criteria." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v1/Input/t44.in" target="_blank">t44.in</a>, <a class="wikilink abifile-wikilink" data-content="Test the option ionmov=4. Si2 molecule,  spin-polarized. Same test as number 16 of fast,  except for the change of ionmov,  no reading of wavefunctions,  iscf to 2,  and different tolerance criteria. Cannot be executed in parallel: --- !ERROR message: | the number of bands in the spin up case must be equal to the number of bands in the spin down case. This is not the case for the k point number : 1 The number of bands spin up and down are : 5 3 Action : change nband,  or use the sequential version of ABINIT. src_file: chkinp.F90 src_line: 1181 ..." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v1/Input/t45.in" target="_blank">t45.in</a> &hellip;</li><li>v2:  <a class="wikilink abifile-wikilink" data-content="Arsenic in rhombohedral structure. Optimization of unit cell size and shape. With 4x4x4 k point grid,  ecut 3.0,  occopt 4 and tsmear 0.06d0, one get acell 3*7.325  angdeg 3*57.36 and xred 0.2277 These input parameters are too small,  but still give a realistic geometry,  since with the much better parameters 12x12x12 k point grid,  ecut 12.0,  occopt 4 and tsmear 0.04d0, one gets acell 3*7.633  angdeg 3*54.95 and xred 0.2296,  while the experimental values are acell 3*7.751   angdeg 3*54.554 and xred 0.2276" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v2/Input/t44.in" target="_blank">t44.in</a>, <a class="wikilink abifile-wikilink" data-content="H2 molecule in a big box. Comparison of the modified Broyden algorithm (ionmov=3) with the original one (ionmov=2). Start with different values of xcart,  from 0.6 to 1.1,  by step of 0.1 . The number of Broyden steps needed to reach acceptable residual forces with the ionmov=3 algorithm are : 3,  2,  2,  3,  3,  4,  while with the ionmov=2 algorithms,  one get : 4,  3,  2,  5,  and then, either the algorithm does not converge within 8 steps,  or it converges to a saddle point of the energy ! This test was hard to make portable. This is why the tolerance for fldiff is very large." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v2/Input/t48.in" target="_blank">t48.in</a> &hellip;</li><li>v3:  <a class="wikilink abifile-wikilink" data-content="8 atoms of Si in an elongated box. 4 of them are fixed,  while atoms 5 and 6 and 7 and 8 are grouped by pair,  such as the sum of their relative separations along y is constrained to 8.0 . This is a crazy constraint,  just to show that it can be done !" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v3/Input/t40.in" target="_blank">t40.in</a>, <a class="wikilink abifile-wikilink" data-content="Magnesium HCP. Optimize the cell volume and shape using different constraints. Checks whether the stresses are symmetric,  and whether the optimisation algorithm does not break the symmetry." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v3/Input/t42.in" target="_blank">t42.in</a>, <a class="wikilink abifile-wikilink" data-content="H2 molecule : examine the rotational freedom. The present test produces the following vibrational frequencies (with degeneracies indicated): 56.89 i cm-1 (2) 0.41   cm-1 (2) 1.05   cm-1 3800    cm-1 The large frequency corresponds to the stretching mode,  and has the right order of magnitude. The frequencies close to 1 cm-1 corresponds to translation modes,  and are small enough for usual applications. The  56.89 i cm-1 mode corresponds to rotation of the H2 molecule. The magnitude of this frequency might seem quite large. Here are the results of tests made to understand this phenomenon. First,  note that ecut 12  acell 3*14 Increasing the value of ecut to 25 decreases the magnitude of the frequency to 36.8 cm-1 . However,  in order to continue to make it smaller, the cell size must be increased,  and an oscillatory behaviour is observed : 3*16   45.6 i cm-1 3*18   22.7   cm-1 3*20   19.1 i cm-1 3*22   15.7 i cm-1 3*24   13.7   cm-1 Many other tests have been set up. In particular, it was observed that the frequency of the oscillatory behaviour changes with the ecut,  and also that using the Gamma point,  instead of the 1/4 1/4 1/4 k point (used in this test) degrades the convergence. The overall picture is as follows. There are different reasons for the translation and rotation modes to acquire a non-zero frequency when plane waves and supercells are used. Still,  as concerns translations,  only the existence of a discretization of the XC grid is important. For rotations,  supercell effects are also present : - alignement of dipole or quadrupoles - interaction between tails of wavefunctions,  accross cells Since the convergence in supercell size is oscillatory,  we infer that the breaking of the rotational invariance is mostly due to interaction between wavefunction tails. This will be checked by confining the system in a spherical well,  in a forthcoming test." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v3/Input/t80.in" target="_blank">t80.in</a> &hellip;</li><li>v5:  <a class="wikilink abifile-wikilink" data-content="Test of geometry optimization using using ionmov=1, 4, 5 and 7 Hydrogen molecule inside a box of 12 Bohr This test was created to test ionmov=5" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v5/Input/t02.in" target="_blank">t02.in</a>, <a class="wikilink abifile-wikilink" data-content="Al2 molecule in a big box. Treat 8 bands,  occupied using occopt=4. Compare tolrff and toldff stopping criteria" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v5/Input/t41.in" target="_blank">t41.in</a>, <a class="wikilink abifile-wikilink" data-content="White tin,  body-centered tetragonal. This test checks that in an optcell 2 run,  the symmetry of the body-centered case is not broken. This test is run for two values of strprecon also, to test the strprecon variable. Coded and contributed by J. Zwanziger." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v5/Input/t77.in" target="_blank">t77.in</a> &hellip;</li><li>v6:  <a class="wikilink abifile-wikilink" data-content="Hydrogen diatomic molecule in a cell,  close to BCC. One hydrogen atom is at the origin. The other lies initially along the 1 1 1 direction,  at a distance close to the one of the free dimer distance. The final position is the mirror with respect to the plane perpendicular to x,  passing at 1/2 0 0 . The cell parameter is less than twice the H2 interatomic distance,  so that the transition path come close to the x axis,  although it never reaches it (the closest being in the mirror plane). Test the images : 6 images,  exploring the transition path by a simple algorithm. MG: Does it work in parallel. I tried 2 CPUs and the run get stuck!" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v6/Input/t22.in" target="_blank">t22.in</a>, <a class="wikilink abifile-wikilink" data-content="Delocalized internals test. Si diamond bulk with displaced atom Unconverged in all variables" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v6/Input/t23.in" target="_blank">t23.in</a>, <a class="wikilink abifile-wikilink" data-content="H2 Molecule in a small box Second-order eigenvalue calculation using a small number of bands,  testing bdeigrf." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v6/Input/t61.in" target="_blank">t61.in</a> &hellip;</li><li>v7:  <a class="wikilink abifile-wikilink" data-content="Test tolmxde input variable. For silicon supercell with atomic positions randomly displaced" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v7/Input/t15.in" target="_blank">t15.in</a>, <a class="wikilink abifile-wikilink" data-content="Relaxation of FCC Argon crystal with DFT-D3 dispersion correction Both the pair-wise term and the 3-body term are included In this test the following properties are thus tested: contribution to the energy and to the stress of DFT-D3 including the 3-body term" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v7/Input/t97.in" target="_blank">t97.in</a> &hellip;</li><li>v8:  <a class="wikilink abifile-wikilink" data-content="LiH molecule test the use of spatial chemical potential. The molecule starts aligned along the x direction, while the Li chemical potential pulls the Li nucleus toward -z,  and the H chemical potential pushes the H nucleus toward +z. See whether the optimization runs properly." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v8/Input/t31.in" target="_blank">t31.in</a> &hellip;</li></ul></details><p>Sets a maximal absolute force tolerance (in hartree/Bohr) below which BFGS
structural relaxation iterations will stop.<br />
Can also control tolerance on stresses, when <a class="wikilink codevar-wikilink" href="../rlx#optcell">optcell</a> /=0, using the
conversion factor <a class="wikilink codevar-wikilink" href="../rlx#strfact">strfact</a>. This tolerance applies to any particular
cartesian component of any atom, excluding fixed ones. See the parameter
<a class="wikilink codevar-wikilink" href="../rlx#ionmov">ionmov</a>.<br />
This is to be used when trying to equilibrate a structure to its lowest energy
configuration ( <a class="wikilink codevar-wikilink" href="../rlx#ionmov">ionmov</a> =2).<br />
A value of about 5.0d-5 hartree/Bohr or smaller is suggested (this corresponds
to about 2.5d-3 eV/Angstrom).<br />
No meaning for RF calculations.</p>
<h2 id="vel"><strong>vel</strong><a class="headerlink" href="#vel" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> VELocity<br />
<em>Characteristics:</em> <a class="wikilink" href="../../user-guide/abinit#32-more-about-abinit-input-variables">EVOLVING</a><br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-content="How to perform a PIMD calculation" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../topics/PIMD">topic_PIMD</a>, <a class="wikilink topic-wikilink" data-content="How to perform a molecular dynamics calculation" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../topics/MolecularDynamics">topic_MolecularDynamics</a><br />
<em>Variable type:</em> real<br />
<em>Dimensions:</em> (3,<a class="wikilink codevar-wikilink" href="../basic#natom">natom</a>)<br />
<em>Commentdims:</em> It is represented internally as <a class="wikilink codevar-wikilink" href="../rlx#vel">vel</a>(3,<a class="wikilink codevar-wikilink" href="../basic#natom">natom</a>,<a class="wikilink codevar-wikilink" href="../rlx#nimage">nimage</a>)<br />
<em>Default value:</em> <em>0<br />
*Only relevant if:</em> <a class="wikilink codevar-wikilink" href="../rlx#ionmov">ionmov</a> &gt; 0  </p>
<details class="note"><summary>Test list (click to open). Moderately used, [82/920] in all abinit tests, [5/116] in abinit tutorials</summary><ul><li>bigdft:  <a class="wikilink abifile-wikilink" data-content="H2 molecule with default parameters. No forces are computed,  with steepest decent for the direct minimisation of the wavefunctions." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/bigdft/Input/t01.in" target="_blank">t01.in</a>, <a class="wikilink abifile-wikilink" data-content="H2 molecule with default parameters. No forces are computed,  with DIIS for the direct minimisation of the wavefunctions." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/bigdft/Input/t02.in" target="_blank">t02.in</a>, <a class="wikilink abifile-wikilink" data-content="H2 molecule with default parameters. Forces are computed." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/bigdft/Input/t03.in" target="_blank">t03.in</a> &hellip;</li><li>bigdft_paral:  <a class="wikilink abifile-wikilink" data-content="O2 in a box (NC). ABINIT routines are used (wvl_bigdft_comp=0) Test: DIIS,  potential/density mixing and parallelism" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/bigdft_paral/Input/t01.in" target="_blank">t01.in</a>, <a class="wikilink abifile-wikilink" data-content="O2 in a box (NC). ABINIT routines are used (wvl_bigdft_comp=0) Test: DIIS,  potential/density mixing and parallelism" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/bigdft_paral/Input/t01.in" target="_blank">t01.in</a>, <a class="wikilink abifile-wikilink" data-content="CO2 in a box (PAW). ABINIT routines are used (wvl_bigdft_comp=0) Test: DIIS,  density mixing" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/bigdft_paral/Input/t02.in" target="_blank">t02.in</a> &hellip;</li><li>libxc:  <a class="wikilink abifile-wikilink" data-content="Isolated Carbon atom Test the mGGA functionals generating directly XC potentials. Similar to test 08,  but with smaller acell and ecut. Without the use of a &amp;quot;large&amp;quot; xc_denpos,  Tran-Blaha is hard to converge. Hypothesis : there is a strong non-linear region,  preventing the Anderson method to work effectively. Note that the default iscf (=7) does not allow to converge with the default xc_denpos ! By contrast,  things are much better behaved with xc_denpos on the order of 1.0e-7 ." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/libxc/Input/t13.in" target="_blank">t13.in</a> &hellip;</li><li>paral:  <a class="wikilink abifile-wikilink" data-content="C-diamond,  Bulk,  2 atoms,  paral_kgb,  with PAW. Test of ground state with different occs(7 and 0), and also ionmov 2. Only with 0 and 4 procs." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/paral/Input/t22.in" target="_blank">t22.in</a> &hellip;</li><li>seq:  <a class="wikilink abifile-wikilink" data-content="81. Nitrogen atom in a big box,  spin-polarized. Compute Berry phase for atom at different positions." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/seq/Input/tsv2_81.in" target="_blank">tsv2_81.in</a> &hellip;</li><li>tutoparal:  <a class="wikilink abifile-wikilink" data-content="BaTiO3 linear response calculation" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/tutoparal/Input/tdfpt_04.in" target="_blank">tdfpt_04.in</a> &hellip;</li><li>tutorespfn:  <a class="wikilink abifile-wikilink" data-content="Temperature dependence calculation of diamond." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/tutorespfn/Input/tdepes_1.in" target="_blank">tdepes_1.in</a>, <a class="wikilink abifile-wikilink" data-content="Temperature dependence calculation of diamond." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/tutorespfn/Input/tdepes_3.in" target="_blank">tdepes_3.in</a>, <a class="wikilink abifile-wikilink" data-content="Temperature dependence calculation of diamond." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/tutorespfn/Input/tdepes_4.in" target="_blank">tdepes_4.in</a> &hellip;</li><li>tutorial:  <a class="wikilink abifile-wikilink" data-content="H2 molecule in a big box This file to optimize the H2 bond length,  compute the associated total energy,  then to compute the total energy of the isolated H atom. Here,  the ecut and acell are fixed : the double loop reduces effectively to a single loop." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/tutorial/Input/tbase2_4.in" target="_blank">tbase2_4.in</a> &hellip;</li><li>v1:  <a class="wikilink abifile-wikilink" data-content="O2 molecule,  treated like a metal (there is a degeneracy at the Fermi level), occopt=4 and tsmear=0.04,  with nsppol=2 . Use spinat to polarize the molecule at start. Gives total energy of -32.09317 Hartree This test can be used to check the accuracy of stresses in the spin-polarized case. Setting acell(1:2) to 7.005 and 6.995 gives ETOT -32.090846193972 and -32.095501753750, ucvol  441.63023 and 440.37023 so that the estimation of stress by d(Etot)/d(Vol) gives 3.6948887E-03, to be compared with sigma(1)=3.6949713786E-03 and sigma(2)=3.6949681083E-03 from the present test (the degeneracy is slightly lifted because of incomplete convergence, using tolvrs 1.0d-12 instead of toldfe gives at least 8 identical digits for both sigma values)." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v1/Input/t08.in" target="_blank">t08.in</a>, <a class="wikilink abifile-wikilink" data-content="Helium FCC solid,  using bare potential,  but a relatively low cut-off. Cell parameter is 6.505 Bohr,  2 k points are used.  PBE GGA. Combined with test 29,  it is used to test the accuracy of the stress calculation. Output of the code are as follows : Energy=-2.835212586 Ha,  volume=68.814809 Bohr^3, stress=6.03381838E-04 Ha/Bohr^3" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v1/Input/t28.in" target="_blank">t28.in</a>, <a class="wikilink abifile-wikilink" data-content="Isolated tin atom,  with 50sn.pspnc pseudopotential (Troullier-Martins), to be compared with the result of test 32. The size of the box is 12x12x12,  One k-point (1/4 1/4 1/4) is used. 8 elements of symmetry are present. The cut-off is 14 Hartree. The energy levels (eV) are found at -7.162,  -0.273 and -0.228 (two-fold deg) The difference between these energy levels is 6.889 and 0.045. Going to larger unit cells decrease the latter difference (14x14x14 makes it 0.010)." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v1/Input/t31.in" target="_blank">t31.in</a> &hellip;</li><li>v2:  <a class="wikilink abifile-wikilink" data-content="H2 molecule in a big box : compute VERY accurately the derivatives of the energy,  by both symmetric finite-differences and direct computation of forces and 2DTE. Also test the interplay between istwfk/=1 in the GS calculation and istwfk==1 in the RF calculation (istwfk/=1 is not yet-991020- allowed for RF,  which is a shame) 1) Computation of the first-order derivative of the total energy With delta(xred)=0.0002,  one gets delta(etot)/delta(xred)=-3.145846551 With delta(xred)=0.0001,  one gets delta(etot)/delta(xred)=-3.145836932 The combination of both results,  in a higher-order finite difference formula gives -3.145833726 . The direct computation of forces at the target geometry gives -3.145833725869 . The agreement is perfect, taking into account the &amp;quot;limited&amp;quot; number of digits (10) of the finite-difference result. 2) Computation of the second-order derivative of the total energy With delta(xred)=0.0002,  one gets delta(dedt)/delta(xred)=188.73875 With delta(xred)=0.0001,  one gets delta(dedt)/delta(xred)=188.73837 The combination of both results,  in a higher-order finite difference formula gives 188.73824613 . The direct computation of 2DTE at the target geometry gives 188.73824613046 . The agreement at the level of 11 digits is also perfect." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v2/Input/t33.in" target="_blank">t33.in</a>, <a class="wikilink abifile-wikilink" data-content="Aluminum in FCC structure. At fixed tsmear (=0.04),  check the convergence with respect to the number of k points,  for different occopt. The grids that are tested have respectively 10,  28,  and 60 k points in the IZB (ngkpt values are 4 4 4 4,  6 6 6 4 and 8 8 8 4 For occopt=4,  the total energy (or the enthalpy) is -2.079770 Ha,  -2.080534 Ha,  -2.080672 Ha; for occopt=5,  it is -2.079787 Ha,  -2.080563 Ha,  -2.080693 Ha; for occopt=7,  it is -2.084211 Ha,  -2.084861 Ha,  -2.084916 Ha. The latter values can be corrected by computing the mean of the total energy and the internal energy,  as described in Marzari's thesis,  for example. Thanks to this procedure,  one gets : -2.079732 Ha,  -2.080470 Ha,  and -2.080626 Ha. For information,  with a 12 12 12 4 grid (182 k points) and occopt=4,  one gets -2.080648 Ha." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v2/Input/t46.in" target="_blank">t46.in</a>, <a class="wikilink abifile-wikilink" data-content="Ta,  single atom in a box (1k point). HGH pseudopotential Ecut and acell are NOT realistic. One would need 20 Ha and a 20x20x20 box,  while here we have 5 Ha and a 12x12x12 box. The small box size changes the degeneracy of levels. The all-electron values derived from another code,  for comparison,  are level  degeneracy       energy (Ha)  spin-orbit splitting 6s      2               -0.194573 5d3/2   4               -0.141779 5d5/2   6               -0.119933       0.0218 6p1/2   2               -0.050542 6p3/2   4               -0.031281       0.0193" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v2/Input/t84.in" target="_blank">t84.in</a> &hellip;</li><li>v3:  <a class="wikilink abifile-wikilink" data-content="Fe,  one atom in a big box. Test spin-polarized GS and RF calculation,  at fixed occupation number. Check acoustic sum rule. The ecut is 18 Hartree. It seems quite low,  but gives phonon frequency of 3.83i cm-1,  which is quite small, while increasing it to 22 Ha or 26 Ha,  respectively, gives 4.73i cm-1 and 1.32i cm-1,  respectively." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v3/Input/t19.in" target="_blank">t19.in</a>, <a class="wikilink abifile-wikilink" data-content="O2 molecule,  treated like a metal (there is a degeneracy at the Fermi level),  occopt=4 and tsmear=0.04,  with nsppol=2 . Use spinat to polarize the molecule at start. Same as test v1 #8,  except use different energy and length units. Test the use of dimensional and logical input variables." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v3/Input/t53.in" target="_blank">t53.in</a>, <a class="wikilink abifile-wikilink" data-content="Test on splitting screening calculation using nqptdm and qptdm Si in 2-atom diamond unit cell (nkpt 2;  ecut 6) In dataset 1,  ABINIT finds the ground state (tolwfr 1.0d-16) the eigenvectors necessary for the GW calculation being stored in _WFK. In datasets 2-4,  ABINIT computes the eps^-1 matrix for 1 qpt,  2 qpts,  3qpts respectively" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v3/Input/t87.in" target="_blank">t87.in</a> &hellip;</li><li>v4:  <a class="wikilink abifile-wikilink" data-content="Ta,  single atom in a box (1k point). Same as test v2 #84 HGH pseudopotential Ecut and acell are NOT realistic. One would need 20 Ha and a 20x20x20 box,  while here we have 5 Ha and a 12x12x12 box. Prepare the analysis of next test level  degeneracy       energy (Ha)  spin-orbit splitting 6s      2               -0.194573 5d3/2   4               -0.141779 5d5/2   6               -0.119933       0.0218 6p1/2   2               -0.050542 6p3/2   4               -0.031281       0.0193" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v4/Input/t36.in" target="_blank">t36.in</a>, <a class="wikilink abifile-wikilink" data-content="Test of the strain perturbation with the GGA Note that this test and the following one,  59, address more advanced features of strain perturbation than the ones going with number 60 to 79,  and were developed later,  but were placed here for convenience). AlAs in the zincblende structure,  with a choice of pseudopotenials that include model core charges.  In the LDA,  the strain perturbation simply introduces perturbing charge densities leading to perturbing first-order potentials in a straightforward manner.  (This perturbation does act differently on the valence and model core charges.)   The GGA is a non-trivial extension, since strain perturbs the gradient operator itself.  This test is a symmetrized and stripped-down version of a test case used in development which yielded agreement between such RF calculations and numerical derivatives of GS stresses and forces to 1.e-8 to 1.e-9 atomic units for the elastic tensor and intenal strain coupling parameters.  Acheiving this agreement does require forcing a considerably more dense fft grid than the default which is accepted here. (ngfft= 50 50 50 was used.)  While only the PBE GGA is tested here,  all forms present in Abinit should give comparably accurate results." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v4/Input/t58.in" target="_blank">t58.in</a>, <a class="wikilink abifile-wikilink" data-content="Test of the strain perturbation. Si in the usual diamond structrure using a Troullier-Martins potential including a model core charge.  The datasets include an initial ground state run,  2 pairs of 5 ground state runs with incrementally strained lattice vectors rprim (xx and yz strains -0.00004,  -0.00002,  0,  0.00002,  0.00004), and one response function run for the strain 2nd derivatives. This set is illustrative of the kind of testing used extensively in developing the strain perturbation portions of the code. The numerical 2nd derivatives of the energy wrt strains were computed by applying a 5-point derivative formula to ucvol*sigma(i, j), where the sigmas are the stresses calculated in the series of ground state calculations.  These agree with the analytic 2nd derivatives calculated in the response function run to ~1.E-7 for the data given here.  The cartesian internal strain (mixed second derivative wrt strain and atomic coordinate) agrees to ~1E-8. Note that such numerical tests of the internal strain are only valid if the forces are zero (relaxed or by symmetry).  Otherwise,  the reduced-coordinate 2nd derivatives must be used for such tests because the conversion to cartesian coordinates is itself strain- dependent and will introduce changes in the computed derivatives. In the actual development tests,  stricter converence criteria were used (tolvrs=1.E-24) as well as larger cutoffs,  and agreement was obtained at the 1.E-9 level.  The model core charge contributes the largest component of the error. ADDITIONAL NOTE : It should be noted that numerical derivative tests of the internal strain should be done by comparing the &amp;quot;2nd-order matrix&amp;quot; with numerical strain derivatives of the gradients wrt reduced atomic coordinates. These are not normally part of the GS output,  but can be obtained by uncommenting the DEBUG section at the end of prtxf.F90. Comparisons between numerical strain derivatives of the cartesian forces and the cartesian internal strain tensor will be incorrect unless the coordinates have been relaxed and the forces are zero within a tight tolerance." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v4/Input/t61.in" target="_blank">t61.in</a> &hellip;</li><li>v5:  <a class="wikilink abifile-wikilink" data-content="uses ionmov 14 : a 14-step time integration by Blanes and Moan." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v5/Input/t03.in" target="_blank">t03.in</a>, <a class="wikilink abifile-wikilink" data-content="PAW: test of some variables reserved to experienced users or developers BCC iron,  ferromagnetic,  GGA,  PAW. Also tests maxnsym keyword." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v5/Input/t06.in" target="_blank">t06.in</a>, <a class="wikilink abifile-wikilink" data-content="Ni atom in a big box. Test the use of the atvshift variable : change the potential of the d orbitals." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v5/Input/t14.in" target="_blank">t14.in</a> &hellip;</li><li>v6:  <a class="wikilink abifile-wikilink" data-content="Chain of Silicon diatomic molecules (4 Si2 molecules in the cell) Freeze oscillatory perturbations with different wavelengths and intensities, thanks to the qprtrb and vprtrb input variables. This should be linked with the computation of the dielectric constant, test v2#05,  that uses directly the RF capabilities of ABINIT, for one diatomic molecule. For dataset 1,  one reproduces the results obtained in Tv2#05,  multiplied by 4. The total energy is consistent up to more than 10 digits : -6.6499924738006 Ha for Tv2#05,  -26.599969895203 Ha for the present calculation. For dataset 2,  the perturbation qprtrb 0 0 1 is frozen in,  with vprtrb 100. The total energy is -26.600317638775 Ha. The difference wrt the unperturbed situation is 0.000348743572 Ha. For dataset 3,  a much smaller perturbation (10 times smaller) is taken, giving total energy -26.599973367786 Ha. The difference wrt the unperturbed situation is 0.3472583 microHa. For dataset 4,  an even smaller perturbation (100 times smaller) is taken, giving total energy -26.599969929928 Ha. The difference wrt the unperturbed situation is 0.000034725 microHa. With datasets 3 and 4,  we are in the linear regime. The previous amplitude is better for such studies. Dataset 5 is the same as 3,  with reversed amplitude. Results are similar to dataset 3. I had no sufficient time to analyze these data correctly and make the connection with the results of Tv2#05,  unfortunately. The following (also test 02 below) gives some more data,  and raise questions. There might be some problem with the use of qprtrb and vprtrb. For dataset 2,  the group of the four lowest eigenenergies (each corresponding to a different molecule) is : -0.47198  -0.46381  -0.46091  -0.45266,  whose spread is 0.01932 Ha. One might think that the maximum and minimum of the potential are separated roughly by 0.02 Ha. The value vprtrb 100 corresponds to a cosine wave whose amplitude is 100,  divided by the volume of the cell,  that is 5000 Bohr^3 : 0.02 Ha. The maximum and minimum of the potential should thus be separated by 0.04 Ha. There seems to be a factor of 2 off." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v6/Input/t01.in" target="_blank">t01.in</a>, <a class="wikilink abifile-wikilink" data-content="Chain of Silicon diatomic molecules (4 Si2 molecules in the cell) Freeze oscillatory perturbations with different wavelengths and intensities, thanks to the qprtrb and vprtrb input variables. Compute the dielectric constant. Similar to test v6#01, but uses a more symmetric geometry,  to examine invariance of the response with respect to shifts of potential,  and also a shorter wavelength. I do not understand why vprtrb 0 10.0 leads to no response. Such a sine wave should cause similar response as for the cosine wave. No time presently to investigate this problem (XG090909)" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v6/Input/t02.in" target="_blank">t02.in</a>, <a class="wikilink abifile-wikilink" data-content="MgB2,  space group P6/m m m (#191);  Bravais hP (primitive hexag.) GS calculation followed by a band structure calculation in which the k-path is automatically defined via the ndivsm input variable. The third dataset tests the calculation and the output of the Fermi surface (prtfsurf=1) in the Xcrysden format." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v6/Input/t14.in" target="_blank">t14.in</a> &hellip;</li><li>v67mbpt:  <a class="wikilink abifile-wikilink" data-content="Isolated hydrogen atom Calculation of the HF levels with different ways to integrate the Coulomb divergence" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v67mbpt/Input/t09.in" target="_blank">t09.in</a>, <a class="wikilink abifile-wikilink" data-content="Silicon: Solution of the Bethe-Salpeter equation (BSE) with norm-conserving pseudopotentials. W is calculated at the RPA level while the scissors operator is used to open the gap by 0.8 eV. First the BSE is solved with the direct diagonalization of the two-particle Hamiltonian,  then the Haydock iterative method is employed to calculate the macroscopic dielectric function. The last dataset solves the BSE problem including the coupling between resonant and anti-resonant transition via brute force diagonalization. Shiva is disabled because the coupling part is unstable on this machine" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v67mbpt/Input/t11.in" target="_blank">t11.in</a>, <a class="wikilink abifile-wikilink" data-content="Molecular SiH4 in GWLS (G0W0 with Lanczos basis and Sternheimer equations)" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v67mbpt/Input/t15.in" target="_blank">t15.in</a> &hellip;</li><li>v7:  <a class="wikilink abifile-wikilink" data-content="Random stopping power within RPA and TD-LDA" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v7/Input/t16.in" target="_blank">t16.in</a>, <a class="wikilink abifile-wikilink" data-content="EuO rock salt structure. 20-electron PAW pseudo (4f in the core,  but 4d10 as semi-core : 4d10 5s2 5p6 (5d0) 6s2 ) Test of computation of dielectric tensor,  Born effective charges,  dynamical matrix. To keep the CPU time to a reasonable level,  the responses are computed with minimal parameters (nstep and the kpoint grid are too small, ecut might be increased,  and also pawecutdg). Note: Charge neutrality is not achieved with the present dataset for Born Effective Charges, but can be easily reached by increasing some parameters;  for instance: (ngkpt 6 6 6,   ecut 40. pawecutdg 40.) gives Z(Eu)=2.5085,  Z(O)=-2.4982" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v7/Input/t43.in" target="_blank">t43.in</a>, <a class="wikilink abifile-wikilink" data-content="Diamond. Test the getddb variable that cancels the divergence in el-ph for small q-points Such divergence is induced by a residual electric field. Note: The Fan terms in DATASET 6 should be different (renormalized) from the one of DATASET 5." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v7/Input/t58.in" target="_blank">t58.in</a> &hellip;</li><li>v8:  <a class="wikilink abifile-wikilink" data-content="Test generation of k-point mesh with kptrlatt and shiftk. version 8.0.8b was stopping at the level of the parser." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v8/Input/t03.in" target="_blank">t03.in</a>, <a class="wikilink abifile-wikilink" data-content="Simple test of velocity verlet integrator" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v8/Input/t12.in" target="_blank">t12.in</a>, <a class="wikilink abifile-wikilink" data-content="WARNING: This code is still under development. Use at your own risk! Preparatory run for the calculation of the Fan-Migdal + Debye-Waller self-energy C in diamond structure. Very rough 2x2x2 q-point grid (3 qpoints);  low ecut." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v8/Input/t41.in" target="_blank">t41.in</a> &hellip;</li></ul></details><p>Gives the starting velocities of atoms, in cartesian coordinates, in
Bohr/atomic time units (atomic time units given where <a class="wikilink codevar-wikilink" href="../rlx#dtion">dtion</a> is described).<br />
For <a class="wikilink codevar-wikilink" href="../rlx#ionmov">ionmov</a>=8 (Nose thermostat), if <a class="wikilink codevar-wikilink" href="../rlx#vel">vel</a> is not initialized, a random
initial velocity giving the right kinetic energy will be generated.<br />
If the atom manipulator is used, <a class="wikilink codevar-wikilink" href="../rlx#vel">vel</a> will be related to the preprocessed
set of atoms, generated by the atom manipulator. The user must thus foresee
the effect of this atom manipulator (see <a class="wikilink codevar-wikilink" href="../geo#objarf">objarf</a>).<br />
Velocities evolve is <a class="wikilink codevar-wikilink" href="../rlx#ionmov">ionmov</a>==1.</p>
<h2 id="vel_cell"><strong>vel_cell</strong><a class="headerlink" href="#vel_cell" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> VELocity of the CELL parameters<br />
<em>Characteristics:</em> <a class="wikilink" href="../../user-guide/abinit#32-more-about-abinit-input-variables">EVOLVING</a><br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-content="How to perform a PIMD calculation" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../topics/PIMD">topic_PIMD</a><br />
<em>Variable type:</em> real<br />
<em>Dimensions:</em> (3,3)<br />
<em>Commentdims:</em> It is represented internally as <a class="wikilink codevar-wikilink" href="../rlx#vel_cell">vel_cell</a>(3,3,<a class="wikilink codevar-wikilink" href="../rlx#nimage">nimage</a>) <br />
<em>Default value:</em> <em>3<br />
*Only relevant if:</em> <a class="wikilink codevar-wikilink" href="../rlx#imgmov">imgmov</a> in [9,13] and <a class="wikilink codevar-wikilink" href="../rlx#optcell">optcell</a> &gt; 0
(Path-Integral Molecular Dynamics
with NPT algorithm)  </p>
<details class="note"><summary>Test list (click to open). Rarely used, [1/920] in all abinit tests, [0/116] in abinit tutorials</summary><ul><li>v6:  <a class="wikilink abifile-wikilink" data-content="(TO BE MODIFIED BY GREGORY) Hydrogen diatomic molecule in a cell,  close to BCC Test the string method : 6 images,  exploring the transition path but keeping X coordinated fixed. Similar to test 24" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v6/Input/t26.in" target="_blank">t26.in</a></li></ul></details><p>Irrelevant unless <a class="wikilink codevar-wikilink" href="../rlx#imgmov">imgmov</a>=9 or 13 and <a class="wikilink codevar-wikilink" href="../rlx#optcell">optcell</a>&gt;0 (Path-Integral
Molecular Dynamics with NPT algorithm).<br />
Gives the starting velocities of the dimensional cell parameters in
Bohr/atomic time units (atomic time units given where <a class="wikilink codevar-wikilink" href="../rlx#dtion">dtion</a> is described).</p>
<h2 id="vis"><strong>vis</strong><a class="headerlink" href="#vis" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> VIScosity<br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-content="How to perform a PIMD calculation" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../topics/PIMD">topic_PIMD</a>, <a class="wikilink topic-wikilink" data-content="How to perform a molecular dynamics calculation" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../topics/MolecularDynamics">topic_MolecularDynamics</a><br />
<em>Variable type:</em> real<br />
<em>Dimensions:</em> scalar<br />
<em>Default value:</em> 100  </p>
<details class="note"><summary>Test list (click to open). Moderately used, [20/920] in all abinit tests, [2/116] in abinit tutorials</summary><ul><li>etsf_io:  <a class="wikilink abifile-wikilink" data-content="Bulk water ice 1h,  made-up with 8 water molecules. Density is exported using the ETSF I/O file format. Should be a nice system to look at with visualisation tools." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/etsf_io/Input/t01.in" target="_blank">t01.in</a></li><li>fast:  <a class="wikilink abifile-wikilink" data-content="21. Conduct same relaxation but use ionmov=1 (molecular dynamics)." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/fast/Input/t21.in" target="_blank">t21.in</a>, <a class="wikilink abifile-wikilink" data-content="move with natfix. Output also GEO files" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/fast/Input/t29.in" target="_blank">t29.in</a></li><li>libxc:  <a class="wikilink abifile-wikilink" data-content="Isolated Helium atom Compare the results using the native XC library and the corresponding ones of LibXC,  in the non-spin-polarized case : 5 LDAs (Teter93,  Perdew-Zunger,  Wigner,  Hedin-Ludqvist,  PW92) 7 GGAs (PBE,  RPBE,  HCTH93,  HCTH120,  HCTH147,  HCTH407,  WuCohen) Results are excellent (better than 10 microHa)." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/libxc/Input/t01.in" target="_blank">t01.in</a></li><li>tutorial:  <a class="wikilink abifile-wikilink" data-content="Crystalline silicon Computation of the band structure. First,  a SCF density computation,  then a non-SCF band structure calculation." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/tutorial/Input/tbase3_5.in" target="_blank">tbase3_5.in</a>, <a class="wikilink abifile-wikilink" data-content="H6 Supercell Generates a 6 atom Hydrogen supercell with the multiplicity of 1:2:3 (x:y:z) Produce WFK file to be analyzed with fold2bloch" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/tutorial/Input/tfold2bloch_1.in" target="_blank">tfold2bloch_1.in</a></li><li>v1:  <a class="wikilink abifile-wikilink" data-content="Isolated Helium atom (see above). Treated with LDA,  Hedin-Lundqvist functional (ixc=5). The total energy is -2.8335 Ha. For speed-up,  it begins with the wavefunctions from test 11." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v1/Input/t15.in" target="_blank">t15.in</a></li><li>v2:  <a class="wikilink abifile-wikilink" data-content="Basic test of geometry optimization for water First computing the ground state of a close configuration Uses 5 different methods to find the relaxed positions" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v2/Input/t88.in" target="_blank">t88.in</a></li><li>v3:  <a class="wikilink abifile-wikilink" data-content="8 atoms of Si in an elongated box. 4 of them are fixed,  while atoms 5 and 6 and 7 and 8 are grouped by pair,  such as the sum of their relative separations along y is constrained to 8.0 . This is a crazy constraint,  just to show that it can be done !" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v3/Input/t40.in" target="_blank">t40.in</a></li><li>v5:  <a class="wikilink abifile-wikilink" data-content="Test of geometry optimization using using ionmov=1, 4, 5 and 7 Hydrogen molecule inside a box of 12 Bohr This test was created to test ionmov=5" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v5/Input/t02.in" target="_blank">t02.in</a>, <a class="wikilink abifile-wikilink" data-content="Generate both DEN and PAWDEN files for silicon. Note that in the PAW case,  DEN files include the valence density and compensation charge,  while PAWDEN files include the valence and on-site corrections,  but not the compensation charge. Therefore,  DEN files are needed for further computations,  while PAWDEN files are useful for visualization,  AIM,  and other post-processing of the density. PAW AE DEN not compatible with parallelization over atoms." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v5/Input/t11.in" target="_blank">t11.in</a>, <a class="wikilink abifile-wikilink" data-content="Crystalline Silicon to test the CP2K pseudopotential implementation with the new pseudo type: pspcod 10 (same formalism as HGH pseudos,  pspcod 3). Same input file as t35.in of tests/tutorial,  the only difference is that in the test the used pseudo is taken from the CP2K library. (http://cvs.berlios.de/cgi-bin/viewcvs.cgi/cp2k/potentials/Goedecker/abinit/) The bandstructure is in agreement with that calculated in the tutorial example with the pspcod=1 pseudopotential." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v5/Input/t54.in" target="_blank">t54.in</a></li><li>v6:  <a class="wikilink abifile-wikilink" data-content="MgB2,  space group P6/m m m (#191);  Bravais hP (primitive hexag.) GS calculation followed by a band structure calculation in which the k-path is automatically defined via the ndivsm input variable. The third dataset tests the calculation and the output of the Fermi surface (prtfsurf=1) in the Xcrysden format." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v6/Input/t14.in" target="_blank">t14.in</a>, <a class="wikilink abifile-wikilink" data-content="Bismuth,  rhombohedral,  two atoms per unit cell Scalar relativistic. Relax four different images,  using the steepest descent algorithm." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v6/Input/t21.in" target="_blank">t21.in</a>, <a class="wikilink abifile-wikilink" data-content="Crystalline silicon. Diamond structure. Quick test of the use of datasets with a numbering beyond 1000 (the last one is 1022). (Tests 35-40 are for response-function)" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v6/Input/t34.in" target="_blank">t34.in</a></li><li>v7:  <a class="wikilink abifile-wikilink" data-content="Test of the extension of the pspcod=8 input format to include spin-orbit coupling with multiple non-local projectors.  The band structure of fcc Pt is computed at Gamma,  X,  and L. The pseudopotential used in the test is generated by the open-source ONCVPSP code,  which is available at www.mat-simresearch.com,  and described in D. R. Hamann,  Phys. Rev. B 88,  085177 (2013). The relativistic extension is documented in the code package. The Pt 5s and 5p semi-cores are treated as valence,  and the band structure is well-converged at the ecut used in the test (20 Ha). A complete description of the input format is avaiable in doc/psp_infos/psp8_info.txt." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v7/Input/t07.in" target="_blank">t07.in</a>, <a class="wikilink abifile-wikilink" data-content="Hydrogen molecule: Test the Path-Integral Molecular Dynamics (PIMD) implementation in (N, V, T) ensemble. THe two thermostats (Nose-Hoover chains and Langevin) are tested together with the three coordinate systems (primitive,  normal mode,  staging). NOTE: MPI run with 2 procs is ok but fldiff comparison fails due to an extra section with --- Pseudopotential description --" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v7/Input/t08.in" target="_blank">t08.in</a>, <a class="wikilink abifile-wikilink" data-content="Test of the extension of the pspcod=8 input format to include spin-orbit coupling with multiple non-local projectors and pseudized valence charge density The relativistic extension is documented in the code package. The Pt 5s and 5p semi-cores are treated as valence. A complete description of the input format is avaiable in  doc/psp_infos/psp8_info.txt." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v7/Input/t17.in" target="_blank">t17.in</a></li><li>v8:  <a class="wikilink abifile-wikilink" data-content="H2 molecule: PIMD simulation with Langevin thermostat. Test of restart with HIST.nc file. Test of linear constraint." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v8/Input/t05.in" target="_blank">t05.in</a></li><li>wannier90:  <a class="wikilink abifile-wikilink" data-content="Cannot be executed in parallel (mlwfovl_qp) Si fcc,  in primitive cell (2 atoms/cell) Test of self-consistent model GW (2 iterations) following Faleev et al., [PRL 93,  126406 (2004)] followed by construction of quasiparticle maximally-localized wannier functions [Hamann &amp;amp; Vanderbilt, arXiv:0810.3616v1 (cond-mat.mtrl-sci)].  Cutoffs are set for test- acceptable speed,  and the results are poorly converged.  The input file is sufficiently annotated to serve as a model.  Note that well- converged GW calculations are extremely time consuming,  and in general it is advisable to run the SCGW part separately on a parallel system, and then run a separate serial job modeled on the last dataset, substituting &amp;quot;irdwfk&amp;quot; and &amp;quot;irdqps&amp;quot; for &amp;quot;getwfk&amp;quot; and &amp;quot;getqps, &amp;quot; with appropriate links to the files produced in the serial run.  Note that the _DEN file from the first dataset is also needed as input,  although the discontinued &amp;quot;irdden&amp;quot; input variable is not needed or supported. Note that acceptable names for the secondary input file needed by the wannier90 library are wannier90.win,  w90.win,  abo_DSn_w90.win (ndtset &amp;gt;0) and abo_w90.win (ndtset=0),  where abo is the 4th line of the .files file and n is the wannier dataset." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/wannier90/Input/t03.in" target="_blank">t03.in</a></li></ul></details><p>The equation of motion is :<br />
M  I  d  2  R  I  /dt  2  = F  I  - <a class="wikilink codevar-wikilink" href="../rlx#vis">vis</a> dR  I  /dt  </p>
<p>The atomic unit of viscosity is hartrees*(atomic time units)/Bohr  2  . Units
are not critical as this is a fictitious damping used to relax structures. A
typical value for silicon is 400 with <a class="wikilink codevar-wikilink" href="../rlx#dtion">dtion</a> of 350 and atomic mass 28
<a class="wikilink codevar-wikilink" href="../rlx#amu">amu</a>. Critical damping is most desirable and is found only by optimizing
<a class="wikilink codevar-wikilink" href="../rlx#vis">vis</a> for a given situation.  </p>
<p>In the case of Path-Integral Molecular Dynamics using the Langevin Thermostat
(<a class="wikilink codevar-wikilink" href="../rlx#imgmov">imgmov</a>=9), <a class="wikilink codevar-wikilink" href="../rlx#vis">vis</a> defines the friction coefficient, in atomic units.
Typical value range is 0.00001-0.001.</p>
<h2 id="wtatcon"><strong>wtatcon</strong><a class="headerlink" href="#wtatcon" title="Permanent link">&para;</a></h2>
<p><em>Mnemonics:</em> WeighTs for AToms in CONstraint equations<br />
<em>Characteristics:</em> <a class="wikilink" href="../../user-guide/abinit#32-more-about-abinit-input-variables">NO_MULTI</a><br />
<em>Mentioned in topic(s):</em> <a class="wikilink topic-wikilink" data-content="How to constaint the geometry of the system in geometry optimization, molecular
dynamics or searches" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../topics/GeoConstraints">topic_GeoConstraints</a><br />
<em>Variable type:</em> real<br />
<em>Dimensions:</em> (3,<a class="wikilink codevar-wikilink" href="../rlx#natcon">natcon</a>,<a class="wikilink codevar-wikilink" href="../rlx#nconeq">nconeq</a>)<br />
<em>Default value:</em> 0  </p>
<details class="note"><summary>Test list (click to open). Rarely used, [3/920] in all abinit tests, [0/116] in abinit tutorials</summary><ul><li>v3:  <a class="wikilink abifile-wikilink" data-content="8 atoms of Si in an elongated box. 4 of them are fixed,  while atoms 5 and 6 and 7 and 8 are grouped by pair,  such as the sum of their relative separations along y is constrained to 8.0 . This is a crazy constraint,  just to show that it can be done !" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v3/Input/t40.in" target="_blank">t40.in</a></li><li>v6:  <a class="wikilink abifile-wikilink" data-content="Hydrogen diatomic molecule in a cell,  close to BCC. One hydrogen atom is at the origin. The other lies initially along the 1 1 1 direction,  at a distance close to the one of the free dimer distance. The final position is the mirror with respect to the plane perpendicular to x,  passing at 1/2 0 0 . The cell parameter is less than twice the H2 interatomic distance,  so that the transition path come close to the x axis,  although it never reaches it (the closest being in the mirror plane). Test the images : 6 images,  exploring the transition path by a simple algorithm. MG: Does it work in parallel. I tried 2 CPUs and the run get stuck!" data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v6/Input/t22.in" target="_blank">t22.in</a></li><li>v8:  <a class="wikilink abifile-wikilink" data-content="H2 molecule: PIMD simulation with Langevin thermostat. Test of restart with HIST.nc file. Test of linear constraint." data-placement="auto right" data-toggle="popover" data-trigger="hover focus" href="../../tests/v8/Input/t05.in" target="_blank">t05.in</a></li></ul></details><p>Gives the weights determining how the motion of atoms is constrained during
structural optimization or molecular dynamics (see <a class="wikilink codevar-wikilink" href="../rlx#nconeq">nconeq</a> , <a class="wikilink codevar-wikilink" href="../rlx#natcon">natcon</a>,
and <a class="wikilink codevar-wikilink" href="../rlx#iatcon">iatcon</a>). For each of the <a class="wikilink codevar-wikilink" href="../rlx#nconeq">nconeq</a> independent constraint equations,
wtatcon is a 3*<a class="wikilink codevar-wikilink" href="../rlx#natcon">natcon</a> array giving weights, W  I  , for the x, y, and z
components of each of the atoms (labeled by I) in the list of indices
<a class="wikilink codevar-wikilink" href="../rlx#iatcon">iatcon</a>. Prior to taking an atomic step, the calculated forces, F  I  , are
replaced by projected forces, F&rsquo;  I  , which satisfy the set of constraint
equations  </p>
<p>Sum  mu=x,y,z; I=1,natcon  : W  mu,I  * F&rsquo;  mu,I  = 0 for each of the
<a class="wikilink codevar-wikilink" href="../rlx#nconeq">nconeq</a> arrays W  I  .  </p>
<p>Different types of motion constraints can be implemented this way. For
example,  </p>
<p>nconeq 1 natcon 2 iatcon 1 2 wtatcon 0 0 +1 0 0 -1  </p>
<p>could be used to constrain the relative height difference of two adsorbate
atoms on a surface (assuming their masses are equal), since F&rsquo;  z,1  - F&rsquo;
z,2  = 0 implies z  1  - z  2  = constant.</p>
                
                  
                
              

<!-- 
add dropdown button with toolbar (based on https://bootsnipp.com/snippets/featured/inbox-by-gmail)
and return to Top button (based on https://codepen.io/rdallaire/pen/apoyx)
-->
<div class="md-container">
  <div class="row">
    <div id="inbox">
      <div class="fab btn-group show-on-hover dropup">
        <div data-toggle="tooltip" data-placement="left" title="Menu" style="margin-left: 42px;">
          <button id="return-to-top" type="button" class="btn btn-primary btn-io dropdown-toggle" data-toggle="dropdown">
            <span class="fa-stack fa-2x">
                <i class="fa fa-circle fa-stack-2x fab-backdrop"></i>
                <i class="fa fa-chevron-up fa-stack-1x fa-inverse fab-primary"></i>
                <i class="fa fa-map-signs fa-stack-1x fa-inverse fab-secondary"></i>
            </span>
          </button>
        </div>
        <ul class="dropdown-menu dropdown-menu-right" role="menu">
        
        
          <li><a href="../../about/release-notes/" data-toggle="tooltip" data-placement="left" 
                  title="About"><i class="fa fa-info-circle"></i></a></li>
        
          <li><a href="../../developers/abimkdocs/" data-toggle="tooltip" data-placement="left" 
                  title="Developers"><i class="fa fa-terminal"></i></a></li>
        
          <li><a href="../../theory/wavefunctions/" data-toggle="tooltip" data-placement="left" 
                  title="Theory"><i class="fa fa-book"></i></a></li>
        
          <li><a href="../../tutorials/welcome/" data-toggle="tooltip" data-placement="left" 
                  title="Lessons"><i class="fa fa-graduation-cap"></i></a></li>
        
          <li><a href="../" data-toggle="tooltip" data-placement="left" 
                  title="Variables"><i class="fa fa-file-text-o"></i></a></li>
        
          <li><a href="../../user-guide/features/" data-toggle="tooltip" data-placement="left" 
                  title="Topics"><i class="fa fa-question"></i></a></li>
        
          <li><a href="../.." data-toggle="tooltip" data-placement="left" 
                  title="User Guide"><i class="fa fa-user-circle-o"></i></a></li>
        
        <!--
        Example: 
        <li><a href="/index.html" data-toggle="tooltip" data-placement="left" title="User-Guide">
              <i class="fa fa-user-circle-o"></i></a></li>
        -->
        </ul>
      </div>
    </div>
  </div>
</div>

              
                
              
            </article>
          </div>
        </div>
      </main>
      
    <!-- Add tabs to footer -->
    
        

  

<nav class="md-tabs md-tabs--active" data-md-component="tabs">
  <div class="md-tabs__inner md-grid">
    <ul class="md-tabs__list">
      
        
  
  
    <li class="md-tabs__item">
      
        <a href="../.." title="User Guide" class="md-tabs__link">
          User Guide
        </a>
      
    </li>
  

      
        
  
  
    <li class="md-tabs__item">
      
        <a href="../../user-guide/features/" title="Topics" class="md-tabs__link">
          Topics
        </a>
      
    </li>
  

      
        
  
  
    <li class="md-tabs__item">
      
        <a href="../" title="Variables" class="md-tabs__link md-tabs__link--active">
          Variables
        </a>
      
    </li>
  

      
        
  
  
    <li class="md-tabs__item">
      
        <a href="../../tutorials/welcome/" title="Lessons" class="md-tabs__link">
          Lessons
        </a>
      
    </li>
  

      
        
  
  
    <li class="md-tabs__item">
      
        <a href="../../theory/wavefunctions/" title="Theory" class="md-tabs__link">
          Theory
        </a>
      
    </li>
  

      
        
  
  
    <li class="md-tabs__item">
      
        <a href="../../developers/abimkdocs/" title="Developers" class="md-tabs__link">
          Developers
        </a>
      
    </li>
  

      
        
  
  
    <li class="md-tabs__item">
      
        <a href="../../about/release-notes/" title="About" class="md-tabs__link">
          About
        </a>
      
    </li>
  

      
    </ul>
  </div>
</nav>
    
    
        
<footer class="md-footer">
  
    <div class="md-footer-nav">
      <nav class="md-footer-nav__inner md-grid">
        
          <a href="../dfpt/" title="DFPT" class="md-flex md-footer-nav__link md-footer-nav__link--prev" rel="prev">
            <div class="md-flex__cell md-flex__cell--shrink">
              <i class="md-icon md-icon--arrow-back md-footer-nav__button"></i>
            </div>
            <div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title">
              <span class="md-flex__ellipsis">
                <span class="md-footer-nav__direction">
                  Previous
                </span>
                DFPT
              </span>
            </div>
          </a>
        
        
          <a href="../vdw/" title="VdW" class="md-flex md-footer-nav__link md-footer-nav__link--next" rel="next">
            <div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title">
              <span class="md-flex__ellipsis">
                <span class="md-footer-nav__direction">
                  Next
                </span>
                VdW
              </span>
            </div>
            <div class="md-flex__cell md-flex__cell--shrink">
              <i class="md-icon md-icon--arrow-forward md-footer-nav__button"></i>
            </div>
          </a>
        
      </nav>
    </div>
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-footer-copyright">
        
            <center>Abinit version 8.5.4</center>
        
        
          <div class="md-footer-copyright__highlight">
            Copyright &copy; 2017 The Abinit Group
          </div>
        

        powered by
        <a href="http://www.mkdocs.org" title="MkDocs">MkDocs</a>
        and
        <a href="http://squidfunk.github.io/mkdocs-material/" title="Material for MkDocs">
          Material for MkDocs</a>
      </div>
      
        
  <div class="md-footer-social">
    
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css">
    
      <a href="https://github.com/abinit/abinit" class="md-footer-social__link fa fa-github-alt"></a>
    
      <a href="https://twitter.com/hashtag/abinit" class="md-footer-social__link fa fa-twitter"></a>
    
      <a href="http://www.facebook.com/abinit.org" class="md-footer-social__link fa fa-facebook"></a>
    
      <a href="https://www.youtube.com/watch?v=gcbfb_Mteo4" class="md-footer-social__link fa fa-youtube"></a>
    
  </div>

      
    </div>
  </div>
</footer>
      

    </div>
    
      <script src="../../assets/javascripts/application-f3ab9e5ff8.js"></script>
      
      
      <script>app.initialize({url:{base:"../.."}})</script>
      
        <script src="https://cdnjs.cloudflare.com/ajax/libs/ace/1.2.8/ace.js"></script>
      
        <script src="../../extra_javascript/jquery.dialogextend.min.js"></script>
      
        <script src="../../extra_javascript/abidocs.js"></script>
      
    
    
      
    
  </body>
</html>